// @egovernments/digit-ui-module-core/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var reactQuery = require('react-query');
var reactRedux = require('react-redux');
var reactRouterDom = require('react-router-dom');
var reactI18next = require('react-i18next');
var digitUiReactComponents = require('@egovernments/digit-ui-react-components');
var redux = require('redux');
var thunk = _interopDefault(require('redux-thunk'));

var CitizenHome = function CitizenHome(_ref) {
  var modules = _ref.modules;
  var ComponentProvider = Digit.Contexts.ComponentProvider;
  var registry = React.useContext(ComponentProvider);
  var paymentModule = modules.filter(function (_ref2) {
    var code = _ref2.code;
    return code === "Payment";
  })[0];
  var moduleArr = modules.filter(function (_ref3) {
    var code = _ref3.code;
    return code !== "Payment";
  });
  var moduleArray = [paymentModule].concat(moduleArr);
  var showQuickPay = moduleArr.some(function (module) {
    return module.code === "QuickPayLinks";
  });
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, null), /*#__PURE__*/React__default.createElement("div", {
    className: "citizenAllServiceGrid"
  }, moduleArray.filter(function (mod) {
    return mod;
  }).map(function (_ref4, index) {
    var code = _ref4.code;
    var Links = Digit.ComponentRegistryService.getComponent(code + "Links") || function () {
      return /*#__PURE__*/React__default.createElement(React__default.Fragment, null);
    };
    if (code === "Payment" && !showQuickPay) {
      Links = function Links() {
        return /*#__PURE__*/React__default.createElement(React__default.Fragment, null);
      };
    }
    return /*#__PURE__*/React__default.createElement(Links, {
      key: index,
      matchPath: "/digit-ui/citizen/" + code.toLowerCase(),
      userType: "citizen"
    });
  })));
};
var EmployeeHome = function EmployeeHome(_ref5) {
  var modules = _ref5.modules;
  return /*#__PURE__*/React__default.createElement("div", {
    className: "employee-app-container"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "ground-container moduleCardWrapper gridModuleWrapper"
  }, modules.map(function (_ref6, index) {
    var code = _ref6.code;
    var Card = Digit.ComponentRegistryService.getComponent(code + "Card") || function () {
      return /*#__PURE__*/React__default.createElement(React__default.Fragment, null);
    };
    return /*#__PURE__*/React__default.createElement(Card, {
      key: index
    });
  })));
};
var AppHome = function AppHome(_ref7) {
  var userType = _ref7.userType,
    modules = _ref7.modules;
  if (userType === "citizen") {
    return /*#__PURE__*/React__default.createElement(CitizenHome, {
      modules: modules
    });
  }
  return /*#__PURE__*/React__default.createElement(EmployeeHome, {
    modules: modules
  });
};

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
function _inheritsLoose(subClass, superClass) {
  subClass.prototype = Object.create(superClass.prototype);
  subClass.prototype.constructor = subClass;
  _setPrototypeOf(subClass, superClass);
}
function _setPrototypeOf(o, p) {
  _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) {
    o.__proto__ = p;
    return o;
  };
  return _setPrototypeOf(o, p);
}
function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;
  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }
  return target;
}

var _iteratorSymbol = /*#__PURE__*/typeof Symbol !== "undefined" ? Symbol.iterator || (Symbol.iterator = Symbol("Symbol.iterator")) : "@@iterator";
var _asyncIteratorSymbol = /*#__PURE__*/typeof Symbol !== "undefined" ? Symbol.asyncIterator || (Symbol.asyncIterator = Symbol("Symbol.asyncIterator")) : "@@asyncIterator";

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }
  if (result && result.then) {
    return result.then(void 0, recover);
  }
  return result;
}

var loginSteps = [{
  texts: {
    header: "PROVIDE MOBILE NUMBER",
    cardText: "LOGIN",
    nextText: "CS_COMMONS_NEXT",
    submitBarLabel: "NEXT"
  },
  inputs: [{
    label: "MOBILE NUMBER",
    type: "text",
    name: "mobileNumber",
    error: "ERR_HRMS_INVALID_MOB_NO",
    validation: {
      required: true,
      minLength: 10,
      maxLength: 10
    }
  }]
}, {
  texts: {
    header: "LOGIN OTP",
    cardText: "ENTER OTP",
    nextText: "CS_COMMONS_NEXT",
    submitBarLabel: "NEXT"
  }
}, {
  texts: {
    header: "CS_LOGIN_PROVIDE_NAME",
    cardText: "CS_LOGIN_NAME_TEXT",
    nextText: "CS_COMMONS_NEXT",
    submitBarLabel: "CS_COMMONS_NEXT"
  },
  inputs: [{
    label: "CORE_COMMON_NAME",
    type: "text",
    name: "name",
    error: "CORE_COMMON_NAME_VALIDMSG",
    validation: {
      required: true,
      minLength: 1,
      pattern: /^[^{0-9}^\$\"<>?\\\\~!@#$%^()+={}\[\]*,/_:;“”‘’]{1,50}$/i
    }
  }]
}];

var SelectMobileNumber = function SelectMobileNumber(_ref) {
  var t = _ref.t,
    onSelect = _ref.onSelect,
    mobileNumber = _ref.mobileNumber,
    onMobileChange = _ref.onMobileChange,
    config = _ref.config;
  return /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    isDisabled: mobileNumber.length !== 10,
    onSelect: onSelect,
    config: config,
    t: t,
    componentInFront: "+91",
    onChange: onMobileChange,
    value: mobileNumber
  });
};

function useInterval(callback, delay) {
  var savedCallback = React.useRef();
  React.useEffect(function () {
    savedCallback.current = callback;
  }, [callback]);
  React.useEffect(function () {
    function tick() {
      savedCallback.current();
    }
    if (delay !== null) {
      var timer = setInterval(tick, delay);
      return function () {
        return clearInterval(timer);
      };
    }
  }, [delay]);
}

var SelectOtp = function SelectOtp(_ref) {
  var config = _ref.config,
    otp = _ref.otp,
    onOtpChange = _ref.onOtpChange,
    onResend = _ref.onResend,
    onSelect = _ref.onSelect,
    t = _ref.t,
    error = _ref.error,
    _ref$userType = _ref.userType,
    userType = _ref$userType === void 0 ? "citizen" : _ref$userType;
  var _useState = React.useState(30),
    timeLeft = _useState[0],
    setTimeLeft = _useState[1];
  useInterval(function () {
    setTimeLeft(timeLeft - 1);
  }, timeLeft > 0 ? 1000 : null);
  var handleResendOtp = function handleResendOtp() {
    onResend();
    setTimeLeft(2);
  };
  if (userType === "employee") {
    return /*#__PURE__*/React__default.createElement(React.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.OTPInput, {
      length: 6,
      onChange: onOtpChange,
      value: otp
    }), timeLeft > 0 ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardText, null, t("CS_RESEND_ANOTHER_OTP") + " " + timeLeft + " " + t("CS_RESEND_SECONDS")) : /*#__PURE__*/React__default.createElement("p", {
      className: "card-text-button",
      onClick: handleResendOtp
    }, t("CS_RESEND_OTP")), !error && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, null, t("CS_INVALID_OTP")));
  }
  return /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    onSelect: onSelect,
    config: config,
    t: t,
    isDisabled: (otp === null || otp === void 0 ? void 0 : otp.length) !== 6
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.OTPInput, {
    length: 6,
    onChange: onOtpChange,
    value: otp
  }), timeLeft > 0 ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardText, null, t("RESEND ANOTHER OTP IN ") + " " + timeLeft + " " + t("SECONDS")) : /*#__PURE__*/React__default.createElement("p", {
    className: "card-text-button",
    onClick: handleResendOtp
  }, t("RESEND OTP")), !error && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, null, t("INVALID OTP")));
};

var SelectName = function SelectName(_ref) {
  var config = _ref.config,
    onSelect = _ref.onSelect,
    t = _ref.t;
  return /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    config: config,
    onSelect: onSelect,
    t: t
  });
};

var _excluded = ["ResponseInfo", "UserRequest"],
  _excluded2 = ["ResponseInfo", "UserRequest"];
var TYPE_REGISTER = {
  type: "register"
};
var TYPE_LOGIN = {
  type: "login"
};
var DEFAULT_REDIRECT_URL = "/digit-ui/citizen";

var setCitizenDetail = function setCitizenDetail(userObject, token, tenantId) {
  var _JSON$parse, _JSON$parse$value;
  var locale = (_JSON$parse = JSON.parse(sessionStorage.getItem("Digit.initData"))) === null || _JSON$parse === void 0 ? void 0 : (_JSON$parse$value = _JSON$parse.value) === null || _JSON$parse$value === void 0 ? void 0 : _JSON$parse$value.selectedLanguage;
  localStorage.setItem("Citizen.tenant-id", tenantId);
  localStorage.setItem("tenant-id", tenantId);
  localStorage.setItem("citizen.userRequestObject", JSON.stringify(userObject));
  localStorage.setItem("locale", locale);
  localStorage.setItem("Citizen.locale", locale);
  localStorage.setItem("token", token);
  localStorage.setItem("Citizen.token", token);
  localStorage.setItem("user-info", JSON.stringify(userObject));
  localStorage.setItem("Citizen.user-info", JSON.stringify(userObject));
};
var getFromLocation = function getFromLocation(state, searchParams) {
  return (state === null || state === void 0 ? void 0 : state.from) || (searchParams === null || searchParams === void 0 ? void 0 : searchParams.from) || DEFAULT_REDIRECT_URL;
};
var Login = function Login(_ref) {
  var _location$state, _location$state7;
  var stateCode = _ref.stateCode,
    _ref$isUserRegistered = _ref.isUserRegistered,
    isUserRegistered = _ref$isUserRegistered === void 0 ? true : _ref$isUserRegistered;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var location = reactRouterDom.useLocation();
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var history = reactRouterDom.useHistory();
  var _useState = React.useState(null),
    user = _useState[0],
    setUser = _useState[1];
  var _useState2 = React.useState(null),
    error = _useState2[0],
    setError = _useState2[1];
  var _useState3 = React.useState(true),
    isOtpValid = _useState3[0],
    setIsOtpValid = _useState3[1];
  var _useState4 = React.useState(null);
  var _useState5 = React.useState(isUserRegistered ? {} : location === null || location === void 0 ? void 0 : (_location$state = location.state) === null || _location$state === void 0 ? void 0 : _location$state.data),
    params = _useState5[0],
    setParmas = _useState5[1];
  var _useState6 = React.useState(null),
    errorTO = _useState6[0],
    setErrorTO = _useState6[1];
  var searchParams = Digit.Hooks.useQueryParams();
  React.useEffect(function () {
    var errorTimeout;
    if (error) {
      if (errorTO) {
        clearTimeout(errorTO);
        setErrorTO(null);
      }
      errorTimeout = setTimeout(function () {
        setError("");
      }, 5000);
      setErrorTO(errorTimeout);
    }
    return function () {
      errorTimeout && clearTimeout(errorTimeout);
    };
  }, [error]);
  React.useEffect(function () {
    var _location$state2;
    if (!user) {
      return;
    }
    Digit.SessionStorage.set("citizen.userRequestObject", user);
    Digit.UserService.setUser(user);
    setCitizenDetail(user === null || user === void 0 ? void 0 : user.info, user === null || user === void 0 ? void 0 : user.access_token, stateCode);
    var redirectPath = ((_location$state2 = location.state) === null || _location$state2 === void 0 ? void 0 : _location$state2.from) || DEFAULT_REDIRECT_URL;
    history.replace(redirectPath);
  }, [user]);
  var stepItems = React.useMemo(function () {
    return loginSteps.map(function (step) {
      var texts = {};
      for (var key in step.texts) {
        texts[key] = t(step.texts[key]);
      }
      return _extends({}, step, {
        texts: texts
      });
    }, [loginSteps]);
  });
  var getUserType = function getUserType() {
    return Digit.UserService.getType();
  };
  var handleOtpChange = function handleOtpChange(otp) {
    setParmas(_extends({}, params, {
      otp: otp
    }));
  };
  var handleMobileChange = function handleMobileChange(event) {
    var value = event.target.value;
    setParmas(_extends({}, params, {
      mobileNumber: value
    }));
  };
  var selectMobileNumber = function selectMobileNumber(mobileNumber) {
    try {
      setParmas(_extends({}, params, mobileNumber));
      var data = _extends({}, mobileNumber, {
        tenantId: stateCode,
        userType: getUserType()
      });
      return Promise.resolve(function () {
        if (isUserRegistered) {
          return Promise.resolve(sendOtp({
            otp: _extends({}, data, TYPE_LOGIN)
          })).then(function (_ref2) {
            var _location$state4;
            var res = _ref2[0],
              err = _ref2[1];
            if (!err) {
              var _location$state3;
              history.replace(path + "/otp", {
                from: getFromLocation(location.state, searchParams),
                role: (_location$state3 = location.state) === null || _location$state3 === void 0 ? void 0 : _location$state3.role
              });
              return;
            } else {
              if (!(location.state && location.state.role === 'FSM_DSO')) {
                history.push("/digit-ui/citizen/register/name", {
                  from: getFromLocation(location.state, searchParams),
                  data: data
                });
              }
            }
            if ((_location$state4 = location.state) !== null && _location$state4 !== void 0 && _location$state4.role) {
              var _location$state5;
              setError(((_location$state5 = location.state) === null || _location$state5 === void 0 ? void 0 : _location$state5.role) === "FSM_DSO" ? t("ES_ERROR_DSO_LOGIN") : "User not registered.");
            }
          });
        } else {
          return Promise.resolve(sendOtp({
            otp: _extends({}, data, TYPE_REGISTER)
          })).then(function (_ref3) {
            var res = _ref3[0],
              err = _ref3[1];
            if (!err) {
              history.replace(path + "/otp", {
                from: getFromLocation(location.state, searchParams)
              });
            }
          });
        }
      }());
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var selectName = function selectName(name) {
    try {
      var data = _extends({}, params, {
        tenantId: stateCode,
        userType: getUserType()
      }, name);
      setParmas(_extends({}, params, name));
      return Promise.resolve(sendOtp({
        otp: _extends({}, data, TYPE_REGISTER)
      })).then(function (_ref4) {
        var res = _ref4[0],
          err = _ref4[1];
        if (res) {
          history.replace(path + "/otp", {
            from: getFromLocation(location.state, searchParams)
          });
        }
      });
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var selectOtp = function selectOtp() {
    try {
      return Promise.resolve(_catch(function () {
        setIsOtpValid(true);
        var mobileNumber = params.mobileNumber,
          otp = params.otp,
          name = params.name;
        return function () {
          if (isUserRegistered) {
            var requestData = {
              username: mobileNumber,
              password: otp,
              tenantId: stateCode,
              userType: getUserType()
            };
            return Promise.resolve(Digit.UserService.authenticate(requestData)).then(function (_ref5) {
              var _location$state6, _window, _window$globalConfigs;
              var ResponseInfo = _ref5.ResponseInfo,
                info = _ref5.UserRequest,
                tokens = _objectWithoutPropertiesLoose(_ref5, _excluded);
              if ((_location$state6 = location.state) !== null && _location$state6 !== void 0 && _location$state6.role) {
                var roleInfo = info.roles.find(function (userRole) {
                  return userRole.code === location.state.role;
                });
                if (!roleInfo || !roleInfo.code) {
                  setError(t("ES_ERROR_USER_NOT_PERMITTED"));
                  setTimeout(function () {
                    return history.replace(DEFAULT_REDIRECT_URL);
                  }, 5000);
                  return;
                }
              }
              if ((_window = window) !== null && _window !== void 0 && (_window$globalConfigs = _window.globalConfigs) !== null && _window$globalConfigs !== void 0 && _window$globalConfigs.getConfig("ENABLE_SINGLEINSTANCE")) {
                info.tenantId = Digit.ULBService.getStateId();
              }
              setUser(_extends({
                info: info
              }, tokens));
            });
          } else {
            var _temp2 = function () {
              if (!isUserRegistered) {
                var _requestData = {
                  name: name,
                  username: mobileNumber,
                  otpReference: otp,
                  tenantId: stateCode
                };
                return Promise.resolve(Digit.UserService.registerUser(_requestData, stateCode)).then(function (_ref6) {
                  var _window2, _window2$globalConfig;
                  var ResponseInfo = _ref6.ResponseInfo,
                    info = _ref6.UserRequest,
                    tokens = _objectWithoutPropertiesLoose(_ref6, _excluded2);
                  if ((_window2 = window) !== null && _window2 !== void 0 && (_window2$globalConfig = _window2.globalConfigs) !== null && _window2$globalConfig !== void 0 && _window2$globalConfig.getConfig("ENABLE_SINGLEINSTANCE")) {
                    info.tenantId = Digit.ULBService.getStateId();
                  }
                  setUser(_extends({
                    info: info
                  }, tokens));
                });
              }
            }();
            if (_temp2 && _temp2.then) return _temp2.then(function () {});
          }
        }();
      }, function () {
        setIsOtpValid(false);
      }));
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var resendOtp = function resendOtp() {
    try {
      var mobileNumber = params.mobileNumber;
      var data = {
        mobileNumber: mobileNumber,
        tenantId: stateCode,
        userType: getUserType()
      };
      var _temp5 = function () {
        if (!isUserRegistered) {
          return Promise.resolve(sendOtp({
            otp: _extends({}, data, TYPE_REGISTER)
          })).then(function () {});
        } else {
          var _temp6 = function () {
            if (isUserRegistered) {
              return Promise.resolve(sendOtp({
                otp: _extends({}, data, TYPE_LOGIN)
              })).then(function () {});
            }
          }();
          if (_temp6 && _temp6.then) return _temp6.then(function () {});
        }
      }();
      return Promise.resolve(_temp5 && _temp5.then ? _temp5.then(function () {}) : void 0);
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var sendOtp = function sendOtp(data) {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(Digit.UserService.sendOtp(data, stateCode)).then(function (res) {
        return [res, null];
      });
    }, function (err) {
      return [null, err];
    }));
  };
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.AppContainer, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, null), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "" + path,
    exact: true
  }, /*#__PURE__*/React__default.createElement(SelectMobileNumber, {
    onSelect: selectMobileNumber,
    config: stepItems[0],
    mobileNumber: params.mobileNumber || "",
    onMobileChange: handleMobileChange,
    showRegisterLink: isUserRegistered && !((_location$state7 = location.state) !== null && _location$state7 !== void 0 && _location$state7.role),
    t: t
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/otp"
  }, /*#__PURE__*/React__default.createElement(SelectOtp, {
    config: _extends({}, stepItems[1], {
      texts: _extends({}, stepItems[1].texts, {
        cardText: stepItems[1].texts.cardText + " " + (params.mobileNumber || "")
      })
    }),
    onOtpChange: handleOtpChange,
    onResend: resendOtp,
    onSelect: selectOtp,
    otp: params.otp,
    error: isOtpValid,
    t: t
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/name"
  }, /*#__PURE__*/React__default.createElement(SelectName, {
    config: stepItems[2],
    onSelect: selectName,
    t: t
  })), error && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    error: true,
    label: error,
    onClose: function onClose() {
      return setError(null);
    }
  })));
};

var loginConfig = [{
  texts: {
    header: "CORE_COMMON_LOGIN",
    submitButtonLabel: "CORE_COMMON_CONTINUE",
    secondaryButtonLabel: "CORE_COMMON_FORGOT_PASSWORD"
  },
  inputs: [{
    label: "CORE_LOGIN_USERNAME",
    type: "text",
    name: "username",
    error: "ERR_HRMS_INVALID_USER_ID"
  }, {
    label: "CORE_LOGIN_PASSWORD",
    type: "password",
    name: "password",
    error: "ERR_HRMS_WRONG_PASSWORD"
  }, {
    label: "CORE_COMMON_CITY",
    type: "custom",
    name: "city",
    error: "ERR_HRMS_INVALID_CITY"
  }]
}];

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var b = "function" === typeof Symbol && Symbol.for,
  c = b ? Symbol.for("react.element") : 60103,
  d = b ? Symbol.for("react.portal") : 60106,
  e = b ? Symbol.for("react.fragment") : 60107,
  f = b ? Symbol.for("react.strict_mode") : 60108,
  g = b ? Symbol.for("react.profiler") : 60114,
  h = b ? Symbol.for("react.provider") : 60109,
  k = b ? Symbol.for("react.context") : 60110,
  l = b ? Symbol.for("react.async_mode") : 60111,
  m = b ? Symbol.for("react.concurrent_mode") : 60111,
  n = b ? Symbol.for("react.forward_ref") : 60112,
  p = b ? Symbol.for("react.suspense") : 60113,
  q = b ? Symbol.for("react.suspense_list") : 60120,
  r = b ? Symbol.for("react.memo") : 60115,
  t = b ? Symbol.for("react.lazy") : 60116,
  v = b ? Symbol.for("react.block") : 60121,
  w = b ? Symbol.for("react.fundamental") : 60117,
  x = b ? Symbol.for("react.responder") : 60118,
  y = b ? Symbol.for("react.scope") : 60119;
function z(a) {
  if ("object" === typeof a && null !== a) {
    var u = a.$$typeof;
    switch (u) {
      case c:
        switch (a = a.type, a) {
          case l:
          case m:
          case e:
          case g:
          case f:
          case p:
            return a;
          default:
            switch (a = a && a.$$typeof, a) {
              case k:
              case n:
              case t:
              case r:
              case h:
                return a;
              default:
                return u;
            }
        }
      case d:
        return u;
    }
  }
}
function A(a) {
  return z(a) === m;
}
var AsyncMode = l;
var ConcurrentMode = m;
var ContextConsumer = k;
var ContextProvider = h;
var Element = c;
var ForwardRef = n;
var Fragment = e;
var Lazy = t;
var Memo = r;
var Portal = d;
var Profiler = g;
var StrictMode = f;
var Suspense = p;
var isAsyncMode = function isAsyncMode(a) {
  return A(a) || z(a) === l;
};
var isConcurrentMode = A;
var isContextConsumer = function isContextConsumer(a) {
  return z(a) === k;
};
var isContextProvider = function isContextProvider(a) {
  return z(a) === h;
};
var isElement = function isElement(a) {
  return "object" === typeof a && null !== a && a.$$typeof === c;
};
var isForwardRef = function isForwardRef(a) {
  return z(a) === n;
};
var isFragment = function isFragment(a) {
  return z(a) === e;
};
var isLazy = function isLazy(a) {
  return z(a) === t;
};
var isMemo = function isMemo(a) {
  return z(a) === r;
};
var isPortal = function isPortal(a) {
  return z(a) === d;
};
var isProfiler = function isProfiler(a) {
  return z(a) === g;
};
var isStrictMode = function isStrictMode(a) {
  return z(a) === f;
};
var isSuspense = function isSuspense(a) {
  return z(a) === p;
};
var isValidElementType = function isValidElementType(a) {
  return "string" === typeof a || "function" === typeof a || a === e || a === m || a === g || a === f || a === p || a === q || "object" === typeof a && null !== a && (a.$$typeof === t || a.$$typeof === r || a.$$typeof === h || a.$$typeof === k || a.$$typeof === n || a.$$typeof === w || a.$$typeof === x || a.$$typeof === y || a.$$typeof === v);
};
var typeOf = z;
var reactIs_production_min = {
  AsyncMode: AsyncMode,
  ConcurrentMode: ConcurrentMode,
  ContextConsumer: ContextConsumer,
  ContextProvider: ContextProvider,
  Element: Element,
  ForwardRef: ForwardRef,
  Fragment: Fragment,
  Lazy: Lazy,
  Memo: Memo,
  Portal: Portal,
  Profiler: Profiler,
  StrictMode: StrictMode,
  Suspense: Suspense,
  isAsyncMode: isAsyncMode,
  isConcurrentMode: isConcurrentMode,
  isContextConsumer: isContextConsumer,
  isContextProvider: isContextProvider,
  isElement: isElement,
  isForwardRef: isForwardRef,
  isFragment: isFragment,
  isLazy: isLazy,
  isMemo: isMemo,
  isPortal: isPortal,
  isProfiler: isProfiler,
  isStrictMode: isStrictMode,
  isSuspense: isSuspense,
  isValidElementType: isValidElementType,
  typeOf: typeOf
};

var reactIs_development = createCommonjsModule(function (module, exports) {

  if (process.env.NODE_ENV !== "production") {
    (function () {

      var hasSymbol = typeof Symbol === 'function' && Symbol.for;
      var REACT_ELEMENT_TYPE = hasSymbol ? Symbol.for('react.element') : 0xeac7;
      var REACT_PORTAL_TYPE = hasSymbol ? Symbol.for('react.portal') : 0xeaca;
      var REACT_FRAGMENT_TYPE = hasSymbol ? Symbol.for('react.fragment') : 0xeacb;
      var REACT_STRICT_MODE_TYPE = hasSymbol ? Symbol.for('react.strict_mode') : 0xeacc;
      var REACT_PROFILER_TYPE = hasSymbol ? Symbol.for('react.profiler') : 0xead2;
      var REACT_PROVIDER_TYPE = hasSymbol ? Symbol.for('react.provider') : 0xeacd;
      var REACT_CONTEXT_TYPE = hasSymbol ? Symbol.for('react.context') : 0xeace;

      var REACT_ASYNC_MODE_TYPE = hasSymbol ? Symbol.for('react.async_mode') : 0xeacf;
      var REACT_CONCURRENT_MODE_TYPE = hasSymbol ? Symbol.for('react.concurrent_mode') : 0xeacf;
      var REACT_FORWARD_REF_TYPE = hasSymbol ? Symbol.for('react.forward_ref') : 0xead0;
      var REACT_SUSPENSE_TYPE = hasSymbol ? Symbol.for('react.suspense') : 0xead1;
      var REACT_SUSPENSE_LIST_TYPE = hasSymbol ? Symbol.for('react.suspense_list') : 0xead8;
      var REACT_MEMO_TYPE = hasSymbol ? Symbol.for('react.memo') : 0xead3;
      var REACT_LAZY_TYPE = hasSymbol ? Symbol.for('react.lazy') : 0xead4;
      var REACT_BLOCK_TYPE = hasSymbol ? Symbol.for('react.block') : 0xead9;
      var REACT_FUNDAMENTAL_TYPE = hasSymbol ? Symbol.for('react.fundamental') : 0xead5;
      var REACT_RESPONDER_TYPE = hasSymbol ? Symbol.for('react.responder') : 0xead6;
      var REACT_SCOPE_TYPE = hasSymbol ? Symbol.for('react.scope') : 0xead7;
      function isValidElementType(type) {
        return typeof type === 'string' || typeof type === 'function' ||
        type === REACT_FRAGMENT_TYPE || type === REACT_CONCURRENT_MODE_TYPE || type === REACT_PROFILER_TYPE || type === REACT_STRICT_MODE_TYPE || type === REACT_SUSPENSE_TYPE || type === REACT_SUSPENSE_LIST_TYPE || typeof type === 'object' && type !== null && (type.$$typeof === REACT_LAZY_TYPE || type.$$typeof === REACT_MEMO_TYPE || type.$$typeof === REACT_PROVIDER_TYPE || type.$$typeof === REACT_CONTEXT_TYPE || type.$$typeof === REACT_FORWARD_REF_TYPE || type.$$typeof === REACT_FUNDAMENTAL_TYPE || type.$$typeof === REACT_RESPONDER_TYPE || type.$$typeof === REACT_SCOPE_TYPE || type.$$typeof === REACT_BLOCK_TYPE);
      }
      function typeOf(object) {
        if (typeof object === 'object' && object !== null) {
          var $$typeof = object.$$typeof;
          switch ($$typeof) {
            case REACT_ELEMENT_TYPE:
              var type = object.type;
              switch (type) {
                case REACT_ASYNC_MODE_TYPE:
                case REACT_CONCURRENT_MODE_TYPE:
                case REACT_FRAGMENT_TYPE:
                case REACT_PROFILER_TYPE:
                case REACT_STRICT_MODE_TYPE:
                case REACT_SUSPENSE_TYPE:
                  return type;
                default:
                  var $$typeofType = type && type.$$typeof;
                  switch ($$typeofType) {
                    case REACT_CONTEXT_TYPE:
                    case REACT_FORWARD_REF_TYPE:
                    case REACT_LAZY_TYPE:
                    case REACT_MEMO_TYPE:
                    case REACT_PROVIDER_TYPE:
                      return $$typeofType;
                    default:
                      return $$typeof;
                  }
              }
            case REACT_PORTAL_TYPE:
              return $$typeof;
          }
        }
        return undefined;
      }

      var AsyncMode = REACT_ASYNC_MODE_TYPE;
      var ConcurrentMode = REACT_CONCURRENT_MODE_TYPE;
      var ContextConsumer = REACT_CONTEXT_TYPE;
      var ContextProvider = REACT_PROVIDER_TYPE;
      var Element = REACT_ELEMENT_TYPE;
      var ForwardRef = REACT_FORWARD_REF_TYPE;
      var Fragment = REACT_FRAGMENT_TYPE;
      var Lazy = REACT_LAZY_TYPE;
      var Memo = REACT_MEMO_TYPE;
      var Portal = REACT_PORTAL_TYPE;
      var Profiler = REACT_PROFILER_TYPE;
      var StrictMode = REACT_STRICT_MODE_TYPE;
      var Suspense = REACT_SUSPENSE_TYPE;
      var hasWarnedAboutDeprecatedIsAsyncMode = false;

      function isAsyncMode(object) {
        {
          if (!hasWarnedAboutDeprecatedIsAsyncMode) {
            hasWarnedAboutDeprecatedIsAsyncMode = true;

            console['warn']('The ReactIs.isAsyncMode() alias has been deprecated, ' + 'and will be removed in React 17+. Update your code to use ' + 'ReactIs.isConcurrentMode() instead. It has the exact same API.');
          }
        }
        return isConcurrentMode(object) || typeOf(object) === REACT_ASYNC_MODE_TYPE;
      }
      function isConcurrentMode(object) {
        return typeOf(object) === REACT_CONCURRENT_MODE_TYPE;
      }
      function isContextConsumer(object) {
        return typeOf(object) === REACT_CONTEXT_TYPE;
      }
      function isContextProvider(object) {
        return typeOf(object) === REACT_PROVIDER_TYPE;
      }
      function isElement(object) {
        return typeof object === 'object' && object !== null && object.$$typeof === REACT_ELEMENT_TYPE;
      }
      function isForwardRef(object) {
        return typeOf(object) === REACT_FORWARD_REF_TYPE;
      }
      function isFragment(object) {
        return typeOf(object) === REACT_FRAGMENT_TYPE;
      }
      function isLazy(object) {
        return typeOf(object) === REACT_LAZY_TYPE;
      }
      function isMemo(object) {
        return typeOf(object) === REACT_MEMO_TYPE;
      }
      function isPortal(object) {
        return typeOf(object) === REACT_PORTAL_TYPE;
      }
      function isProfiler(object) {
        return typeOf(object) === REACT_PROFILER_TYPE;
      }
      function isStrictMode(object) {
        return typeOf(object) === REACT_STRICT_MODE_TYPE;
      }
      function isSuspense(object) {
        return typeOf(object) === REACT_SUSPENSE_TYPE;
      }
      exports.AsyncMode = AsyncMode;
      exports.ConcurrentMode = ConcurrentMode;
      exports.ContextConsumer = ContextConsumer;
      exports.ContextProvider = ContextProvider;
      exports.Element = Element;
      exports.ForwardRef = ForwardRef;
      exports.Fragment = Fragment;
      exports.Lazy = Lazy;
      exports.Memo = Memo;
      exports.Portal = Portal;
      exports.Profiler = Profiler;
      exports.StrictMode = StrictMode;
      exports.Suspense = Suspense;
      exports.isAsyncMode = isAsyncMode;
      exports.isConcurrentMode = isConcurrentMode;
      exports.isContextConsumer = isContextConsumer;
      exports.isContextProvider = isContextProvider;
      exports.isElement = isElement;
      exports.isForwardRef = isForwardRef;
      exports.isFragment = isFragment;
      exports.isLazy = isLazy;
      exports.isMemo = isMemo;
      exports.isPortal = isPortal;
      exports.isProfiler = isProfiler;
      exports.isStrictMode = isStrictMode;
      exports.isSuspense = isSuspense;
      exports.isValidElementType = isValidElementType;
      exports.typeOf = typeOf;
    })();
  }
});

var reactIs = createCommonjsModule(function (module) {

  if (process.env.NODE_ENV === 'production') {
    module.exports = reactIs_production_min;
  } else {
    module.exports = reactIs_development;
  }
});

var getOwnPropertySymbols = Object.getOwnPropertySymbols;
var hasOwnProperty = Object.prototype.hasOwnProperty;
var propIsEnumerable = Object.prototype.propertyIsEnumerable;
function toObject(val) {
  if (val === null || val === undefined) {
    throw new TypeError('Object.assign cannot be called with null or undefined');
  }
  return Object(val);
}
function shouldUseNative() {
  try {
    if (!Object.assign) {
      return false;
    }

    var test1 = new String('abc');
    test1[5] = 'de';
    if (Object.getOwnPropertyNames(test1)[0] === '5') {
      return false;
    }

    var test2 = {};
    for (var i = 0; i < 10; i++) {
      test2['_' + String.fromCharCode(i)] = i;
    }
    var order2 = Object.getOwnPropertyNames(test2).map(function (n) {
      return test2[n];
    });
    if (order2.join('') !== '0123456789') {
      return false;
    }

    var test3 = {};
    'abcdefghijklmnopqrst'.split('').forEach(function (letter) {
      test3[letter] = letter;
    });
    if (Object.keys(Object.assign({}, test3)).join('') !== 'abcdefghijklmnopqrst') {
      return false;
    }
    return true;
  } catch (err) {
    return false;
  }
}
var objectAssign = shouldUseNative() ? Object.assign : function (target, source) {
  var from;
  var to = toObject(target);
  var symbols;
  for (var s = 1; s < arguments.length; s++) {
    from = Object(arguments[s]);
    for (var key in from) {
      if (hasOwnProperty.call(from, key)) {
        to[key] = from[key];
      }
    }
    if (getOwnPropertySymbols) {
      symbols = getOwnPropertySymbols(from);
      for (var i = 0; i < symbols.length; i++) {
        if (propIsEnumerable.call(from, symbols[i])) {
          to[symbols[i]] = from[symbols[i]];
        }
      }
    }
  }
  return to;
};

var ReactPropTypesSecret = 'SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED';
var ReactPropTypesSecret_1 = ReactPropTypesSecret;

var has = Function.call.bind(Object.prototype.hasOwnProperty);

var printWarning = function printWarning() {};
if (process.env.NODE_ENV !== 'production') {
  var ReactPropTypesSecret$1 = ReactPropTypesSecret_1;
  var loggedTypeFailures = {};
  var has$1 = has;
  printWarning = function printWarning(text) {
    var message = 'Warning: ' + text;
    if (typeof console !== 'undefined') {
      console.error(message);
    }
    try {
      throw new Error(message);
    } catch (x) {}
  };
}

function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
  if (process.env.NODE_ENV !== 'production') {
    for (var typeSpecName in typeSpecs) {
      if (has$1(typeSpecs, typeSpecName)) {
        var error;
        try {
          if (typeof typeSpecs[typeSpecName] !== 'function') {
            var err = Error((componentName || 'React class') + ': ' + location + ' type `' + typeSpecName + '` is invalid; ' + 'it must be a function, usually from the `prop-types` package, but received `' + typeof typeSpecs[typeSpecName] + '`.' + 'This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.');
            err.name = 'Invariant Violation';
            throw err;
          }
          error = typeSpecs[typeSpecName](values, typeSpecName, componentName, location, null, ReactPropTypesSecret$1);
        } catch (ex) {
          error = ex;
        }
        if (error && !(error instanceof Error)) {
          printWarning((componentName || 'React class') + ': type specification of ' + location + ' `' + typeSpecName + '` is invalid; the type checker ' + 'function must return `null` or an `Error` but returned a ' + typeof error + '. ' + 'You may have forgotten to pass an argument to the type checker ' + 'creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and ' + 'shape all require an argument).');
        }
        if (error instanceof Error && !(error.message in loggedTypeFailures)) {
          loggedTypeFailures[error.message] = true;
          var stack = getStack ? getStack() : '';
          printWarning('Failed ' + location + ' type: ' + error.message + (stack != null ? stack : ''));
        }
      }
    }
  }
}

checkPropTypes.resetWarningCache = function () {
  if (process.env.NODE_ENV !== 'production') {
    loggedTypeFailures = {};
  }
};
var checkPropTypes_1 = checkPropTypes;

var printWarning$1 = function printWarning() {};
if (process.env.NODE_ENV !== 'production') {
  printWarning$1 = function printWarning(text) {
    var message = 'Warning: ' + text;
    if (typeof console !== 'undefined') {
      console.error(message);
    }
    try {
      throw new Error(message);
    } catch (x) {}
  };
}
function emptyFunctionThatReturnsNull() {
  return null;
}
var factoryWithTypeCheckers = function factoryWithTypeCheckers(isValidElement, throwOnDirectAccess) {
  var ITERATOR_SYMBOL = typeof Symbol === 'function' && Symbol.iterator;
  var FAUX_ITERATOR_SYMBOL = '@@iterator';

  function getIteratorFn(maybeIterable) {
    var iteratorFn = maybeIterable && (ITERATOR_SYMBOL && maybeIterable[ITERATOR_SYMBOL] || maybeIterable[FAUX_ITERATOR_SYMBOL]);
    if (typeof iteratorFn === 'function') {
      return iteratorFn;
    }
  }

  var ANONYMOUS = '<<anonymous>>';

  var ReactPropTypes = {
    array: createPrimitiveTypeChecker('array'),
    bigint: createPrimitiveTypeChecker('bigint'),
    bool: createPrimitiveTypeChecker('boolean'),
    func: createPrimitiveTypeChecker('function'),
    number: createPrimitiveTypeChecker('number'),
    object: createPrimitiveTypeChecker('object'),
    string: createPrimitiveTypeChecker('string'),
    symbol: createPrimitiveTypeChecker('symbol'),
    any: createAnyTypeChecker(),
    arrayOf: createArrayOfTypeChecker,
    element: createElementTypeChecker(),
    elementType: createElementTypeTypeChecker(),
    instanceOf: createInstanceTypeChecker,
    node: createNodeChecker(),
    objectOf: createObjectOfTypeChecker,
    oneOf: createEnumTypeChecker,
    oneOfType: createUnionTypeChecker,
    shape: createShapeTypeChecker,
    exact: createStrictShapeTypeChecker
  };

  function is(x, y) {
    if (x === y) {
      return x !== 0 || 1 / x === 1 / y;
    } else {
      return x !== x && y !== y;
    }
  }

  function PropTypeError(message, data) {
    this.message = message;
    this.data = data && typeof data === 'object' ? data : {};
    this.stack = '';
  }
  PropTypeError.prototype = Error.prototype;
  function createChainableTypeChecker(validate) {
    if (process.env.NODE_ENV !== 'production') {
      var manualPropTypeCallCache = {};
      var manualPropTypeWarningCount = 0;
    }
    function checkType(isRequired, props, propName, componentName, location, propFullName, secret) {
      componentName = componentName || ANONYMOUS;
      propFullName = propFullName || propName;
      if (secret !== ReactPropTypesSecret_1) {
        if (throwOnDirectAccess) {
          var err = new Error('Calling PropTypes validators directly is not supported by the `prop-types` package. ' + 'Use `PropTypes.checkPropTypes()` to call them. ' + 'Read more at http://fb.me/use-check-prop-types');
          err.name = 'Invariant Violation';
          throw err;
        } else if (process.env.NODE_ENV !== 'production' && typeof console !== 'undefined') {
          var cacheKey = componentName + ':' + propName;
          if (!manualPropTypeCallCache[cacheKey] &&
          manualPropTypeWarningCount < 3) {
            printWarning$1('You are manually calling a React.PropTypes validation ' + 'function for the `' + propFullName + '` prop on `' + componentName + '`. This is deprecated ' + 'and will throw in the standalone `prop-types` package. ' + 'You may be seeing this warning due to a third-party PropTypes ' + 'library. See https://fb.me/react-warning-dont-call-proptypes ' + 'for details.');
            manualPropTypeCallCache[cacheKey] = true;
            manualPropTypeWarningCount++;
          }
        }
      }
      if (props[propName] == null) {
        if (isRequired) {
          if (props[propName] === null) {
            return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required ' + ('in `' + componentName + '`, but its value is `null`.'));
          }
          return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required in ' + ('`' + componentName + '`, but its value is `undefined`.'));
        }
        return null;
      } else {
        return validate(props, propName, componentName, location, propFullName);
      }
    }
    var chainedCheckType = checkType.bind(null, false);
    chainedCheckType.isRequired = checkType.bind(null, true);
    return chainedCheckType;
  }
  function createPrimitiveTypeChecker(expectedType) {
    function validate(props, propName, componentName, location, propFullName, secret) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== expectedType) {
        var preciseType = getPreciseType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + preciseType + '` supplied to `' + componentName + '`, expected ') + ('`' + expectedType + '`.'), {
          expectedType: expectedType
        });
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createAnyTypeChecker() {
    return createChainableTypeChecker(emptyFunctionThatReturnsNull);
  }
  function createArrayOfTypeChecker(typeChecker) {
    function validate(props, propName, componentName, location, propFullName) {
      if (typeof typeChecker !== 'function') {
        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside arrayOf.');
      }
      var propValue = props[propName];
      if (!Array.isArray(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an array.'));
      }
      for (var i = 0; i < propValue.length; i++) {
        var error = typeChecker(propValue, i, componentName, location, propFullName + '[' + i + ']', ReactPropTypesSecret_1);
        if (error instanceof Error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createElementTypeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      if (!isValidElement(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createElementTypeTypeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      if (!reactIs.isValidElementType(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement type.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createInstanceTypeChecker(expectedClass) {
    function validate(props, propName, componentName, location, propFullName) {
      if (!(props[propName] instanceof expectedClass)) {
        var expectedClassName = expectedClass.name || ANONYMOUS;
        var actualClassName = getClassName(props[propName]);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + actualClassName + '` supplied to `' + componentName + '`, expected ') + ('instance of `' + expectedClassName + '`.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createEnumTypeChecker(expectedValues) {
    if (!Array.isArray(expectedValues)) {
      if (process.env.NODE_ENV !== 'production') {
        if (arguments.length > 1) {
          printWarning$1('Invalid arguments supplied to oneOf, expected an array, got ' + arguments.length + ' arguments. ' + 'A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z]).');
        } else {
          printWarning$1('Invalid argument supplied to oneOf, expected an array.');
        }
      }
      return emptyFunctionThatReturnsNull;
    }
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      for (var i = 0; i < expectedValues.length; i++) {
        if (is(propValue, expectedValues[i])) {
          return null;
        }
      }
      var valuesString = JSON.stringify(expectedValues, function replacer(key, value) {
        var type = getPreciseType(value);
        if (type === 'symbol') {
          return String(value);
        }
        return value;
      });
      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of value `' + String(propValue) + '` ' + ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.'));
    }
    return createChainableTypeChecker(validate);
  }
  function createObjectOfTypeChecker(typeChecker) {
    function validate(props, propName, componentName, location, propFullName) {
      if (typeof typeChecker !== 'function') {
        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside objectOf.');
      }
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an object.'));
      }
      for (var key in propValue) {
        if (has(propValue, key)) {
          var error = typeChecker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
          if (error instanceof Error) {
            return error;
          }
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createUnionTypeChecker(arrayOfTypeCheckers) {
    if (!Array.isArray(arrayOfTypeCheckers)) {
      process.env.NODE_ENV !== 'production' ? printWarning$1('Invalid argument supplied to oneOfType, expected an instance of array.') : void 0;
      return emptyFunctionThatReturnsNull;
    }
    for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
      var checker = arrayOfTypeCheckers[i];
      if (typeof checker !== 'function') {
        printWarning$1('Invalid argument supplied to oneOfType. Expected an array of check functions, but ' + 'received ' + getPostfixForTypeWarning(checker) + ' at index ' + i + '.');
        return emptyFunctionThatReturnsNull;
      }
    }
    function validate(props, propName, componentName, location, propFullName) {
      var expectedTypes = [];
      for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
        var checker = arrayOfTypeCheckers[i];
        var checkerResult = checker(props, propName, componentName, location, propFullName, ReactPropTypesSecret_1);
        if (checkerResult == null) {
          return null;
        }
        if (checkerResult.data && has(checkerResult.data, 'expectedType')) {
          expectedTypes.push(checkerResult.data.expectedType);
        }
      }
      var expectedTypesMessage = expectedTypes.length > 0 ? ', expected one of type [' + expectedTypes.join(', ') + ']' : '';
      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`' + expectedTypesMessage + '.'));
    }
    return createChainableTypeChecker(validate);
  }
  function createNodeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      if (!isNode(props[propName])) {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`, expected a ReactNode.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function invalidValidatorError(componentName, location, propFullName, key, type) {
    return new PropTypeError((componentName || 'React class') + ': ' + location + ' type `' + propFullName + '.' + key + '` is invalid; ' + 'it must be a function, usually from the `prop-types` package, but received `' + type + '`.');
  }
  function createShapeTypeChecker(shapeTypes) {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
      }
      for (var key in shapeTypes) {
        var checker = shapeTypes[key];
        if (typeof checker !== 'function') {
          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
        }
        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
        if (error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createStrictShapeTypeChecker(shapeTypes) {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
      }
      var allKeys = objectAssign({}, props[propName], shapeTypes);
      for (var key in allKeys) {
        var checker = shapeTypes[key];
        if (has(shapeTypes, key) && typeof checker !== 'function') {
          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
        }
        if (!checker) {
          return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` key `' + key + '` supplied to `' + componentName + '`.' + '\nBad object: ' + JSON.stringify(props[propName], null, '  ') + '\nValid keys: ' + JSON.stringify(Object.keys(shapeTypes), null, '  '));
        }
        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
        if (error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function isNode(propValue) {
    switch (typeof propValue) {
      case 'number':
      case 'string':
      case 'undefined':
        return true;
      case 'boolean':
        return !propValue;
      case 'object':
        if (Array.isArray(propValue)) {
          return propValue.every(isNode);
        }
        if (propValue === null || isValidElement(propValue)) {
          return true;
        }
        var iteratorFn = getIteratorFn(propValue);
        if (iteratorFn) {
          var iterator = iteratorFn.call(propValue);
          var step;
          if (iteratorFn !== propValue.entries) {
            while (!(step = iterator.next()).done) {
              if (!isNode(step.value)) {
                return false;
              }
            }
          } else {
            while (!(step = iterator.next()).done) {
              var entry = step.value;
              if (entry) {
                if (!isNode(entry[1])) {
                  return false;
                }
              }
            }
          }
        } else {
          return false;
        }
        return true;
      default:
        return false;
    }
  }
  function isSymbol(propType, propValue) {
    if (propType === 'symbol') {
      return true;
    }

    if (!propValue) {
      return false;
    }

    if (propValue['@@toStringTag'] === 'Symbol') {
      return true;
    }

    if (typeof Symbol === 'function' && propValue instanceof Symbol) {
      return true;
    }
    return false;
  }

  function getPropType(propValue) {
    var propType = typeof propValue;
    if (Array.isArray(propValue)) {
      return 'array';
    }
    if (propValue instanceof RegExp) {
      return 'object';
    }
    if (isSymbol(propType, propValue)) {
      return 'symbol';
    }
    return propType;
  }

  function getPreciseType(propValue) {
    if (typeof propValue === 'undefined' || propValue === null) {
      return '' + propValue;
    }
    var propType = getPropType(propValue);
    if (propType === 'object') {
      if (propValue instanceof Date) {
        return 'date';
      } else if (propValue instanceof RegExp) {
        return 'regexp';
      }
    }
    return propType;
  }

  function getPostfixForTypeWarning(value) {
    var type = getPreciseType(value);
    switch (type) {
      case 'array':
      case 'object':
        return 'an ' + type;
      case 'boolean':
      case 'date':
      case 'regexp':
        return 'a ' + type;
      default:
        return type;
    }
  }

  function getClassName(propValue) {
    if (!propValue.constructor || !propValue.constructor.name) {
      return ANONYMOUS;
    }
    return propValue.constructor.name;
  }
  ReactPropTypes.checkPropTypes = checkPropTypes_1;
  ReactPropTypes.resetWarningCache = checkPropTypes_1.resetWarningCache;
  ReactPropTypes.PropTypes = ReactPropTypes;
  return ReactPropTypes;
};

function emptyFunction() {}
function emptyFunctionWithReset() {}
emptyFunctionWithReset.resetWarningCache = emptyFunction;
var factoryWithThrowingShims = function factoryWithThrowingShims() {
  function shim(props, propName, componentName, location, propFullName, secret) {
    if (secret === ReactPropTypesSecret_1) {
      return;
    }
    var err = new Error('Calling PropTypes validators directly is not supported by the `prop-types` package. ' + 'Use PropTypes.checkPropTypes() to call them. ' + 'Read more at http://fb.me/use-check-prop-types');
    err.name = 'Invariant Violation';
    throw err;
  }
  shim.isRequired = shim;
  function getShim() {
    return shim;
  }
  var ReactPropTypes = {
    array: shim,
    bigint: shim,
    bool: shim,
    func: shim,
    number: shim,
    object: shim,
    string: shim,
    symbol: shim,
    any: shim,
    arrayOf: getShim,
    element: shim,
    elementType: shim,
    instanceOf: getShim,
    node: shim,
    objectOf: getShim,
    oneOf: getShim,
    oneOfType: getShim,
    shape: getShim,
    exact: getShim,
    checkPropTypes: emptyFunctionWithReset,
    resetWarningCache: emptyFunction
  };
  ReactPropTypes.PropTypes = ReactPropTypes;
  return ReactPropTypes;
};

var propTypes = createCommonjsModule(function (module) {

  if (process.env.NODE_ENV !== 'production') {
    var ReactIs = reactIs;

    var throwOnDirectAccess = true;
    module.exports = factoryWithTypeCheckers(ReactIs.isElement, throwOnDirectAccess);
  } else {
    module.exports = factoryWithThrowingShims();
  }
});

var Background = function Background(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/React__default.createElement("div", {
    className: "banner"
  }, children);
};

var Header = function Header() {
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    storeData = _Digit$Hooks$useStore.data,
    isLoading = _Digit$Hooks$useStore.isLoading;
  var _ref = storeData || {},
    stateInfo = _ref.stateInfo;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  if (isLoading) return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null);
  return /*#__PURE__*/React__default.createElement("div", {
    className: "bannerHeader"
  }, /*#__PURE__*/React__default.createElement("img", {
    className: "bannerLogo",
    src: stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.logoUrl
  }), /*#__PURE__*/React__default.createElement("p", null, t("TENANT_TENANTS_" + (stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.code.toUpperCase()))));
};

var _excluded$1 = ["UserRequest"];

var setEmployeeDetail = function setEmployeeDetail(userObject, token) {
  var _JSON$parse;
  var locale = ((_JSON$parse = JSON.parse(sessionStorage.getItem("Digit.locale"))) === null || _JSON$parse === void 0 ? void 0 : _JSON$parse.value) || "en_IN";
  localStorage.setItem("Employee.tenant-id", userObject === null || userObject === void 0 ? void 0 : userObject.tenantId);
  localStorage.setItem("tenant-id", userObject === null || userObject === void 0 ? void 0 : userObject.tenantId);
  localStorage.setItem("citizen.userRequestObject", JSON.stringify(userObject));
  localStorage.setItem("locale", locale);
  localStorage.setItem("Employee.locale", locale);
  localStorage.setItem("token", token);
  localStorage.setItem("Employee.token", token);
  localStorage.setItem("user-info", JSON.stringify(userObject));
  localStorage.setItem("Employee.user-info", JSON.stringify(userObject));
};
var Login$1 = function Login(_ref) {
  var propsConfig = _ref.config,
    t = _ref.t,
    isDisabled = _ref.isDisabled;
  var _Digit$Hooks$useTenan = Digit.Hooks.useTenants(),
    cities = _Digit$Hooks$useTenan.data,
    isLoading = _Digit$Hooks$useTenan.isLoading;
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    isStoreLoading = _Digit$Hooks$useStore.isLoading;
  var _useState = React.useState(null),
    user = _useState[0],
    setUser = _useState[1];
  var _useState2 = React.useState(null),
    showToast = _useState2[0],
    setShowToast = _useState2[1];
  var _useState3 = React.useState(false),
    disable = _useState3[0],
    setDisable = _useState3[1];
  var history = reactRouterDom.useHistory();
  var sourceUrl = "https://s3.ap-south-1.amazonaws.com/egov-qa-assets";
  var pdfUrl = "https://pg-egov-assets.s3.ap-south-1.amazonaws.com/Upyog+Code+and+Copyright+License_v1.pdf";
  React.useEffect(function () {
    var _user$info, _user$info$roles, _user$info2, _user$info2$roles, _window, _window$location, _window$location$href, _user$info3, _user$info4, _user$info4$roles, _user$info5, _user$info6, _user$info6$roles;
    if (!user) {
      return;
    }
    Digit.SessionStorage.set("citizen.userRequestObject", user);
    var filteredRoles = user === null || user === void 0 ? void 0 : (_user$info = user.info) === null || _user$info === void 0 ? void 0 : (_user$info$roles = _user$info.roles) === null || _user$info$roles === void 0 ? void 0 : _user$info$roles.filter(function (role) {
      return role.tenantId === Digit.SessionStorage.get("Employee.tenantId");
    });
    if ((user === null || user === void 0 ? void 0 : (_user$info2 = user.info) === null || _user$info2 === void 0 ? void 0 : (_user$info2$roles = _user$info2.roles) === null || _user$info2$roles === void 0 ? void 0 : _user$info2$roles.length) > 0) user.info.roles = filteredRoles;
    Digit.UserService.setUser(user);
    setEmployeeDetail(user === null || user === void 0 ? void 0 : user.info, user === null || user === void 0 ? void 0 : user.access_token);
    var redirectPath = "/digit-ui/employee";

    if ((_window = window) !== null && _window !== void 0 && (_window$location = _window.location) !== null && _window$location !== void 0 && (_window$location$href = _window$location.href) !== null && _window$location$href !== void 0 && _window$location$href.includes("from=")) {
      var _window2, _window2$location, _window2$location$hre, _window2$location$hre2;
      redirectPath = decodeURIComponent((_window2 = window) === null || _window2 === void 0 ? void 0 : (_window2$location = _window2.location) === null || _window2$location === void 0 ? void 0 : (_window2$location$hre = _window2$location.href) === null || _window2$location$hre === void 0 ? void 0 : (_window2$location$hre2 = _window2$location$hre.split("from=")) === null || _window2$location$hre2 === void 0 ? void 0 : _window2$location$hre2[1]) || "/digit-ui/employee";
    }

    if (user !== null && user !== void 0 && (_user$info3 = user.info) !== null && _user$info3 !== void 0 && _user$info3.roles && user !== null && user !== void 0 && (_user$info4 = user.info) !== null && _user$info4 !== void 0 && (_user$info4$roles = _user$info4.roles) !== null && _user$info4$roles !== void 0 && _user$info4$roles.every(function (e) {
      return e.code === "NATADMIN";
    })) {
      redirectPath = "/digit-ui/employee/payment/integration/dss/NURT_DASHBOARD";
    }
    if (user !== null && user !== void 0 && (_user$info5 = user.info) !== null && _user$info5 !== void 0 && _user$info5.roles && user !== null && user !== void 0 && (_user$info6 = user.info) !== null && _user$info6 !== void 0 && (_user$info6$roles = _user$info6.roles) !== null && _user$info6$roles !== void 0 && _user$info6$roles.every(function (e) {
      return e.code === "STADMIN";
    })) {
      redirectPath = "/digit-ui/employee/payment/integration/dss/home";
    }
    history.replace(redirectPath);
  }, [user]);
  var onLogin = function onLogin(data) {
    try {
      var _temp3 = function _temp3() {
        setDisable(false);
      };
      if (!data.city) {
        alert("Please Select City!");
        return Promise.resolve();
      }
      setDisable(true);
      var requestData = _extends({}, data, {
        userType: "EMPLOYEE"
      });
      requestData.tenantId = data.city.code;
      delete requestData.city;
      var _temp4 = _catch(function () {
        return Promise.resolve(Digit.UserService.authenticate(requestData)).then(function (_ref3) {
          var info = _ref3.UserRequest,
            tokens = _objectWithoutPropertiesLoose(_ref3, _excluded$1);
          Digit.SessionStorage.set("Employee.tenantId", info === null || info === void 0 ? void 0 : info.tenantId);
          setUser(_extends({
            info: info
          }, tokens));
        });
      }, function (err) {
        var _err$response, _err$response$data;
        setShowToast((err === null || err === void 0 ? void 0 : (_err$response = err.response) === null || _err$response === void 0 ? void 0 : (_err$response$data = _err$response.data) === null || _err$response$data === void 0 ? void 0 : _err$response$data.error_description) || "Invalid login credentials!");
        setTimeout(closeToast, 5000);
      });
      return Promise.resolve(_temp4 && _temp4.then ? _temp4.then(_temp3) : _temp3(_temp4));
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var closeToast = function closeToast() {
    setShowToast(null);
  };
  var onForgotPassword = function onForgotPassword() {
    history.push("/digit-ui/employee/user/forgot-password");
  };
  var _propsConfig$inputs = propsConfig.inputs,
    userId = _propsConfig$inputs[0],
    password = _propsConfig$inputs[1],
    city = _propsConfig$inputs[2];
  var config = [{
    body: [{
      label: t(userId.label),
      type: userId.type,
      populators: {
        name: userId.name
      },
      isMandatory: true
    }, {
      label: t(password.label),
      type: password.type,
      populators: {
        name: password.name
      },
      isMandatory: true
    }, {
      label: t(city.label),
      type: city.type,
      populators: {
        name: city.name,
        customProps: {},
        component: function component(props, customProps) {
          return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, _extends({
            option: cities,
            className: "login-city-dd",
            optionKey: "i18nKey",
            select: function select(d) {
              props.onChange(d);
            },
            t: t
          }, customProps));
        }
      },
      isMandatory: true
    }]
  }];
  return isLoading || isStoreLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null) : /*#__PURE__*/React__default.createElement(Background, null, /*#__PURE__*/React__default.createElement("div", {
    className: "employeeBackbuttonAlign"
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, {
    variant: "white",
    style: {
      borderBottom: "none"
    }
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormComposer, {
    onSubmit: onLogin,
    isDisabled: isDisabled || disable,
    noBoxShadow: true,
    inline: true,
    submitInForm: true,
    config: config,
    label: propsConfig.texts.submitButtonLabel,
    secondaryActionLabel: propsConfig.texts.secondaryButtonLabel,
    onSecondayActionClick: onForgotPassword,
    heading: propsConfig.texts.header,
    headingStyle: {
      textAlign: "center"
    },
    cardStyle: {
      margin: "auto",
      minWidth: "408px"
    },
    className: "loginFormStyleEmployee",
    buttonStyle: {
      maxWidth: "100%",
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(Header, null)), showToast && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    error: true,
    label: t(showToast),
    onClose: closeToast
  }), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: '100%',
      position: 'fixed',
      bottom: 0
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center',
      color: "white"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      cursor: "pointer",
      display: "inline-flex",
      height: '1.4em'
    },
    alt: "Powered by DIGIT",
    src: sourceUrl + "/digit-footer-bw.png",
    onError: "this.src='./../digit-footer.png'",
    onClick: function onClick() {
      window.open('https://www.digit.org/', '_blank').focus();
    }
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    onClick: function onClick() {
      window.open('https://niua.in/', '_blank').focus();
    }
  }, "Copyright \xA9 2022 National Institute of Urban Affairs"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("a", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    href: pdfUrl,
    target: "_blank"
  }, "UPYOG License"))));
};
Login$1.propTypes = {
  loginParams: propTypes.any
};
Login$1.defaultProps = {
  loginParams: null
};

var EmployeeLogin = function EmployeeLogin() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var loginParams = React.useMemo(function () {
    return loginConfig.map(function (step) {
      var texts = {};
      for (var key in step.texts) {
        texts[key] = t(step.texts[key]);
      }
      return _extends({}, step, {
        texts: texts
      });
    }, [loginConfig]);
  });
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "" + path,
    exact: true
  }, /*#__PURE__*/React__default.createElement(Login$1, {
    config: loginParams[0],
    t: t
  })));
};

var config = [{
  texts: {
    header: "CORE_COMMON_RESET_PASSWORD_LABEL",
    submitButtonLabel: "CORE_COMMON_CHANGE_PASSWORD"
  },
  inputs: [{
    label: "CORE_LOGIN_USERNAME",
    type: "text",
    name: "userName",
    error: "ERR_HRMS_INVALID_USERNAME"
  }, {
    label: "CORE_LOGIN_NEW_PASSWORD",
    type: "password",
    name: "newPassword",
    error: "CORE_COMMON_REQUIRED_ERRMSG"
  }, {
    label: "CORE_LOGIN_CONFIRM_NEW_PASSWORD",
    type: "password",
    name: "confirmPassword",
    error: "CORE_COMMON_REQUIRED_ERRMSG"
  }]
}];

var ChangePasswordComponent = function ChangePasswordComponent(_ref) {
  var propsConfig = _ref.config,
    t = _ref.t;
  var _useState = React.useState(null),
    user = _useState[0];
  var _Digit$Hooks$useQuery = Digit.Hooks.useQueryParams(),
    mobileNumber = _Digit$Hooks$useQuery.mobile_number,
    tenantId = _Digit$Hooks$useQuery.tenantId;
  var history = reactRouterDom.useHistory();
  var _useState2 = React.useState(""),
    otp = _useState2[0],
    setOtp = _useState2[1];
  var _useState3 = React.useState(true),
    isOtpValid = _useState3[0];
  var _useState4 = React.useState(null),
    showToast = _useState4[0],
    setShowToast = _useState4[1];
  var getUserType = function getUserType() {
    return Digit.UserService.getType();
  };
  var sourceUrl = "https://s3.ap-south-1.amazonaws.com/egov-qa-assets";
  var pdfUrl = "https://pg-egov-assets.s3.ap-south-1.amazonaws.com/Upyog+Code+and+Copyright+License_v1.pdf";
  React.useEffect(function () {
    var _location$state;
    if (!user) {
      Digit.UserService.setType("employee");
      return;
    }
    Digit.UserService.setUser(user);
    var redirectPath = ((_location$state = location.state) === null || _location$state === void 0 ? void 0 : _location$state.from) || "/digit-ui/employee";
    history.replace(redirectPath);
  }, [user]);
  var closeToast = function closeToast() {
    setShowToast(null);
  };
  var onResendOTP = function onResendOTP() {
    try {
      var _temp3 = function _temp3() {
        setTimeout(closeToast, 5000);
      };
      var requestData = {
        otp: {
          mobileNumber: mobileNumber,
          userType: getUserType().toUpperCase(),
          type: "passwordreset",
          tenantId: tenantId
        }
      };
      var _temp4 = _catch(function () {
        return Promise.resolve(Digit.UserService.sendOtp(requestData, tenantId)).then(function () {
          setShowToast(t("ES_OTP_RESEND"));
        });
      }, function (err) {
        var _err$response, _err$response$data;
        setShowToast((err === null || err === void 0 ? void 0 : (_err$response = err.response) === null || _err$response === void 0 ? void 0 : (_err$response$data = _err$response.data) === null || _err$response$data === void 0 ? void 0 : _err$response$data.error_description) || t("ES_INVALID_LOGIN_CREDENTIALS"));
      });
      return Promise.resolve(_temp4 && _temp4.then ? _temp4.then(_temp3) : _temp3(_temp4));
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var onChangePassword = function onChangePassword(data) {
    try {
      return Promise.resolve(_catch(function () {
        if (data.newPassword !== data.confirmPassword) {
          return setShowToast(t("ERR_PASSWORD_DO_NOT_MATCH"));
        }
        var requestData = _extends({}, data, {
          otpReference: otp,
          tenantId: tenantId,
          type: getUserType().toUpperCase()
        });
        return Promise.resolve(Digit.UserService.changePassword(requestData, tenantId)).then(function (response) {
          navigateToLogin();
        });
      }, function (err) {
        var _err$response2, _err$response2$data, _err$response2$data$e, _err$response2$data$e2, _err$response2$data$e3;
        setShowToast((err === null || err === void 0 ? void 0 : (_err$response2 = err.response) === null || _err$response2 === void 0 ? void 0 : (_err$response2$data = _err$response2.data) === null || _err$response2$data === void 0 ? void 0 : (_err$response2$data$e = _err$response2$data.error) === null || _err$response2$data$e === void 0 ? void 0 : (_err$response2$data$e2 = _err$response2$data$e.fields) === null || _err$response2$data$e2 === void 0 ? void 0 : (_err$response2$data$e3 = _err$response2$data$e2[0]) === null || _err$response2$data$e3 === void 0 ? void 0 : _err$response2$data$e3.message) || t("ES_SOMETHING_WRONG"));
        setTimeout(closeToast, 5000);
      }));
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var navigateToLogin = function navigateToLogin() {
    history.replace("/digit-ui/employee/user/login");
  };
  var _propsConfig$inputs = propsConfig.inputs,
    username = _propsConfig$inputs[0],
    password = _propsConfig$inputs[1],
    confirmPassword = _propsConfig$inputs[2];
  var config = [{
    body: [{
      label: t(username.label),
      type: username.type,
      populators: {
        name: username.name
      },
      isMandatory: true
    }, {
      label: t(password.label),
      type: password.type,
      populators: {
        name: password.name
      },
      isMandatory: true
    }, {
      label: t(confirmPassword.label),
      type: confirmPassword.type,
      populators: {
        name: confirmPassword.name
      },
      isMandatory: true
    }]
  }];
  return /*#__PURE__*/React__default.createElement(Background, null, /*#__PURE__*/React__default.createElement("div", {
    className: "employeeBackbuttonAlign"
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, {
    variant: "white",
    style: {
      borderBottom: "none"
    }
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormComposer, {
    onSubmit: onChangePassword,
    noBoxShadow: true,
    inline: true,
    submitInForm: true,
    config: config,
    label: propsConfig.texts.submitButtonLabel,
    cardStyle: {
      maxWidth: "408px",
      margin: "auto"
    },
    className: "employeeChangePassword"
  }, /*#__PURE__*/React__default.createElement(Header, null), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardSubHeader, {
    style: {
      textAlign: "center"
    }
  }, " ", propsConfig.texts.header, " "), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardText, null, t("CS_LOGIN_OTP_TEXT") + " ", /*#__PURE__*/React__default.createElement("b", null, " ", "" + t("+ 91 - "), " ", mobileNumber)), /*#__PURE__*/React__default.createElement(SelectOtp, {
    t: t,
    userType: "employee",
    otp: otp,
    onOtpChange: setOtp,
    error: isOtpValid,
    onResend: onResendOTP
  })), showToast && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    error: true,
    label: t(showToast),
    onClose: closeToast
  }), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: '100%',
      position: "absolute",
      bottom: 0
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center',
      color: "white"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      cursor: "pointer",
      display: "inline-flex",
      height: '1.4em'
    },
    alt: "Powered by DIGIT",
    src: sourceUrl + "/digit-footer-bw.png",
    onError: "this.src='./../digit-footer.png'",
    onClick: function onClick() {
      window.open('https://www.digit.org/', '_blank').focus();
    }
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    onClick: function onClick() {
      window.open('https://niua.in/', '_blank').focus();
    }
  }, "Copyright \xA9 2022 National Institute of Urban Affairs"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("a", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    href: pdfUrl,
    target: "_blank"
  }, "UPYOG License"))));
};
ChangePasswordComponent.propTypes = {
  loginParams: propTypes.any
};
ChangePasswordComponent.defaultProps = {
  loginParams: null
};

var EmployeeChangePassword = function EmployeeChangePassword() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var params = React.useMemo(function () {
    return config.map(function (step) {
      var texts = {};
      for (var key in step.texts) {
        texts[key] = t(step.texts[key]);
      }
      return _extends({}, step, {
        texts: texts
      });
    }, [config]);
  });
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "" + path,
    exact: true
  }, /*#__PURE__*/React__default.createElement(ChangePasswordComponent, {
    config: params[0],
    t: t
  })));
};

var loginConfig$1 = [{
  texts: {
    header: "CORE_COMMON_FORGOT_PASSWORD_LABEL",
    description: "ES_FORGOT_PASSWORD_DESC",
    submitButtonLabel: "CORE_COMMON_CONTINUE"
  },
  inputs: [{
    label: "CORE_COMMON_MOBILE_NUMBER",
    type: "text",
    name: "mobileNumber",
    error: "ERR_HRMS_INVALID_MOBILE_NUMBER"
  }, {
    label: "CORE_COMMON_CITY",
    type: "custom",
    name: "city",
    error: "ERR_HRMS_INVALID_CITY"
  }]
}];

var ForgotPassword = function ForgotPassword(_ref) {
  var propsConfig = _ref.config,
    t = _ref.t;
  var _Digit$Hooks$useTenan = Digit.Hooks.useTenants(),
    cities = _Digit$Hooks$useTenan.data,
    isLoading = _Digit$Hooks$useTenan.isLoading;
  var _useState = React.useState(null),
    user = _useState[0];
  var history = reactRouterDom.useHistory();
  var _useState2 = React.useState(null),
    showToast = _useState2[0],
    setShowToast = _useState2[1];
  var getUserType = function getUserType() {
    return Digit.UserService.getType();
  };
  var sourceUrl = "https://s3.ap-south-1.amazonaws.com/egov-qa-assets";
  var pdfUrl = "https://pg-egov-assets.s3.ap-south-1.amazonaws.com/Upyog+Code+and+Copyright+License_v1.pdf";
  React.useEffect(function () {
    var _location$state;
    if (!user) {
      Digit.UserService.setType("employee");
      return;
    }
    Digit.UserService.setUser(user);
    var redirectPath = ((_location$state = location.state) === null || _location$state === void 0 ? void 0 : _location$state.from) || "/digit-ui/employee";
    history.replace(redirectPath);
  }, [user]);
  var closeToast = function closeToast() {
    setShowToast(null);
  };
  var onForgotPassword = function onForgotPassword(data) {
    try {
      if (!data.city) {
        alert("Please Select City!");
        return Promise.resolve();
      }
      var requestData = {
        otp: {
          mobileNumber: data.mobileNumber,
          userType: getUserType().toUpperCase(),
          type: "passwordreset",
          tenantId: data.city.code
        }
      };
      var _temp2 = _catch(function () {
        return Promise.resolve(Digit.UserService.sendOtp(requestData, data.city.code)).then(function () {
          history.push("/digit-ui/employee/user/change-password?mobile_number=" + data.mobileNumber + "&tenantId=" + data.city.code);
        });
      }, function (err) {
        var _err$response, _err$response$data, _err$response$data$er, _err$response$data$er2, _err$response$data$er3;
        setShowToast((err === null || err === void 0 ? void 0 : (_err$response = err.response) === null || _err$response === void 0 ? void 0 : (_err$response$data = _err$response.data) === null || _err$response$data === void 0 ? void 0 : (_err$response$data$er = _err$response$data.error) === null || _err$response$data$er === void 0 ? void 0 : (_err$response$data$er2 = _err$response$data$er.fields) === null || _err$response$data$er2 === void 0 ? void 0 : (_err$response$data$er3 = _err$response$data$er2[0]) === null || _err$response$data$er3 === void 0 ? void 0 : _err$response$data$er3.message) || "Invalid login credentials!");
        setTimeout(closeToast, 5000);
      });
      return Promise.resolve(_temp2 && _temp2.then ? _temp2.then(function () {}) : void 0);
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var navigateToLogin = function navigateToLogin() {
    history.replace("/digit-ui/employee/login");
  };
  var _propsConfig$inputs = propsConfig.inputs,
    userId = _propsConfig$inputs[0],
    city = _propsConfig$inputs[1];
  var config = [{
    body: [{
      label: t(userId.label),
      type: userId.type,
      populators: {
        name: userId.name,
        componentInFront: "+91"
      },
      isMandatory: true
    }, {
      label: t(city.label),
      type: city.type,
      populators: {
        name: city.name,
        customProps: {},
        component: function component(props, customProps) {
          return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, _extends({
            option: cities,
            optionKey: "name",
            id: city.name,
            className: "login-city-dd",
            select: function select(d) {
              props.onChange(d);
            }
          }, customProps));
        }
      },
      isMandatory: true
    }]
  }];
  if (isLoading) {
    return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null);
  }
  return /*#__PURE__*/React__default.createElement(Background, null, /*#__PURE__*/React__default.createElement("div", {
    className: "employeeBackbuttonAlign"
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, {
    variant: "white",
    style: {
      borderBottom: "none"
    }
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormComposer, {
    onSubmit: onForgotPassword,
    noBoxShadow: true,
    inline: true,
    submitInForm: true,
    config: config,
    label: propsConfig.texts.submitButtonLabel,
    secondaryActionLabel: propsConfig.texts.secondaryButtonLabel,
    onSecondayActionClick: navigateToLogin,
    heading: propsConfig.texts.header,
    description: propsConfig.texts.description,
    headingStyle: {
      textAlign: "center"
    },
    cardStyle: {
      maxWidth: "408px",
      margin: "auto"
    },
    className: "employeeForgotPassword"
  }, /*#__PURE__*/React__default.createElement(Header, null)), showToast && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    error: true,
    label: t(showToast),
    onClose: closeToast
  }), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: '100%',
      position: "absolute",
      bottom: 0
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center',
      color: "white"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      cursor: "pointer",
      display: "inline-flex",
      height: '1.4em'
    },
    alt: "Powered by DIGIT",
    src: sourceUrl + "/digit-footer-bw.png",
    onError: "this.src='./../digit-footer.png'",
    onClick: function onClick() {
      window.open('https://www.digit.org/', '_blank').focus();
    }
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    onClick: function onClick() {
      window.open('https://niua.in/', '_blank').focus();
    }
  }, "Copyright \xA9 2022 National Institute of Urban Affairs"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("a", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    href: pdfUrl,
    target: "_blank"
  }, "UPYOG License"))));
};
ForgotPassword.propTypes = {
  loginParams: propTypes.any
};
ForgotPassword.defaultProps = {
  loginParams: null
};

var EmployeeForgotPassword = function EmployeeForgotPassword() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var params = React.useMemo(function () {
    return loginConfig$1.map(function (step) {
      var texts = {};
      for (var key in step.texts) {
        texts[key] = t(step.texts[key]);
      }
      return _extends({}, step, {
        texts: texts
      });
    }, [loginConfig$1]);
  });
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "" + path,
    exact: true
  }, /*#__PURE__*/React__default.createElement(ForgotPassword, {
    config: params[0],
    t: t
  })));
};

var LanguageSelection = function LanguageSelection() {
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    storeData = _Digit$Hooks$useStore.data,
    isLoading = _Digit$Hooks$useStore.isLoading;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var _ref = storeData || {},
    languages = _ref.languages,
    stateInfo = _ref.stateInfo;
  var selectedLanguage = Digit.StoreData.getCurrentLanguage();
  var _useState = React.useState(selectedLanguage),
    selected = _useState[0],
    setselected = _useState[1];
  var handleChangeLanguage = function handleChangeLanguage(language) {
    setselected(language.value);
    Digit.LocalizationService.changeLanguage(language.value, stateInfo.code);
  };
  var sourceUrl = "https://s3.ap-south-1.amazonaws.com/egov-qa-assets";
  var pdfUrl = "https://pg-egov-assets.s3.ap-south-1.amazonaws.com/Upyog+Code+and+Copyright+License_v1.pdf";
  var handleSubmit = function handleSubmit(event) {
    history.push("/digit-ui/employee/user/login");
  };
  if (isLoading) return null;
  return /*#__PURE__*/React__default.createElement(Background, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Card, {
    className: "bannerCard removeBottomMargin"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "bannerHeader"
  }, /*#__PURE__*/React__default.createElement("img", {
    className: "bannerLogo",
    src: stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.logoUrl,
    alt: "Digit"
  }), /*#__PURE__*/React__default.createElement("p", null, t("TENANT_TENANTS_" + (stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.code.toUpperCase())))), /*#__PURE__*/React__default.createElement("div", {
    className: "language-selector",
    style: {
      justifyContent: "space-around",
      marginBottom: "24px",
      padding: "0 5%"
    }
  }, languages.map(function (language, index) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "language-button-container",
      key: index
    }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CustomButton, {
      selected: language.value === selected,
      text: language.label,
      onClick: function onClick() {
        return handleChangeLanguage(language);
      }
    }));
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.SubmitBar, {
    style: {
      width: "100%"
    },
    label: t("CORE_COMMON_CONTINUE"),
    onSubmit: handleSubmit
  })), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: '100%',
      position: "absolute",
      bottom: 0
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center',
      color: "white"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      cursor: "pointer",
      display: "inline-flex",
      height: '1.4em'
    },
    alt: "Powered by DIGIT",
    src: sourceUrl + "/digit-footer-bw.png",
    onError: "this.src='./../digit-footer.png'",
    onClick: function onClick() {
      window.open('https://www.digit.org/', '_blank').focus();
    }
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    onClick: function onClick() {
      window.open('https://niua.in/', '_blank').focus();
    }
  }, "Copyright \xA9 2022 National Institute of Urban Affairs"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("a", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    href: pdfUrl,
    target: "_blank"
  }, "UPYOG License"))));
};

var getTenants = function getTenants(codes, tenants) {
  return tenants.filter(function (tenant) {
    var _codes$map;
    return codes === null || codes === void 0 ? void 0 : (_codes$map = codes.map) === null || _codes$map === void 0 ? void 0 : _codes$map.call(codes, function (item) {
      return item.code;
    }).includes(tenant.code);
  });
};
var AppModules = function AppModules(_ref) {
  var stateCode = _ref.stateCode,
    userType = _ref.userType,
    modules = _ref.modules,
    appTenants = _ref.appTenants;
  var ComponentProvider = Digit.Contexts.ComponentProvider;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var location = reactRouterDom.useLocation();
  var user = Digit.UserService.getUser();
  if (!user || !(user !== null && user !== void 0 && user.access_token) || !(user !== null && user !== void 0 && user.info)) {
    return /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
      to: {
        pathname: "/digit-ui/employee/user/login",
        state: {
          from: location.pathname + location.search
        }
      }
    });
  }
  var appRoutes = modules.map(function (_ref2, index) {
    var code = _ref2.code,
      tenants = _ref2.tenants;
    var Module = Digit.ComponentRegistryService.getComponent(code + "Module");
    return Module ? /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
      key: index,
      path: path + "/" + code.toLowerCase()
    }, /*#__PURE__*/React__default.createElement(Module, {
      stateCode: stateCode,
      moduleCode: code,
      userType: userType,
      tenants: getTenants(tenants, appTenants)
    })) : null;
  });
  return /*#__PURE__*/React__default.createElement("div", {
    className: "ground-container"
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, appRoutes, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/login"
  }, " ", /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
    to: {
      pathname: "/digit-ui/employee/user/login",
      state: {
        from: location.pathname + location.search
      }
    }
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/forgot-password"
  }, /*#__PURE__*/React__default.createElement(EmployeeForgotPassword, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/change-password"
  }, " ", /*#__PURE__*/React__default.createElement(EmployeeChangePassword, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, null, /*#__PURE__*/React__default.createElement(AppHome, {
    userType: userType,
    modules: modules
  }))));
};

var ErrorBoundary = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(ErrorBoundary, _React$Component);
  function ErrorBoundary(props) {
    var _this;
    _this = _React$Component.call(this, props) || this;
    _this.state = {
      error: null,
      errorStack: null,
      hasError: false
    };
    return _this;
  }
  ErrorBoundary.getDerivedStateFromError = function getDerivedStateFromError(error) {
    return {
      error: error === null || error === void 0 ? void 0 : error.message,
      hasError: true,
      errorStack: error === null || error === void 0 ? void 0 : error.stack
    };
  };
  var _proto = ErrorBoundary.prototype;
  _proto.componentDidCatch = function componentDidCatch(error, errorInfo) {
    this.setState({
      error: error === null || error === void 0 ? void 0 : error.message,
      hasError: true,
      errorStack: error === null || error === void 0 ? void 0 : error.stack
    });
  };
  _proto.render = function render() {
    if (this.state.hasError) {
      var _this$state, _this$state2, _this$state3;
      console.error("UI-error", (_this$state = this.state) === null || _this$state === void 0 ? void 0 : _this$state.error);
      return /*#__PURE__*/React__default.createElement("div", {
        className: "error-boundary"
      }, /*#__PURE__*/React__default.createElement("summary", null, "Something went wrong"), /*#__PURE__*/React__default.createElement("details", {
        style: {
          whiteSpace: "pre-wrap"
        }
      }, ((_this$state2 = this.state) === null || _this$state2 === void 0 ? void 0 : _this$state2.errorStack) && this.state.errorStack.toString().substring(0, 600), (_this$state3 = this.state) === null || _this$state3 === void 0 ? void 0 : _this$state3.error));
    }
    return this.props.children;
  };
  return ErrorBoundary;
}(React__default.Component);

var stringReplaceAll = function stringReplaceAll(str, searcher, replaceWith) {
  if (str === void 0) {
    str = "";
  }
  if (searcher === void 0) {
    searcher = "";
  }
  if (replaceWith === void 0) {
    replaceWith = "";
  }
  if (searcher == "") return str;
  while ((_str = str) !== null && _str !== void 0 && _str.includes(searcher)) {
    var _str, _str2;
    str = (_str2 = str) === null || _str2 === void 0 ? void 0 : _str2.replace(searcher, replaceWith);
  }
  return str;
};
var ChangeCity = function ChangeCity(prop) {
  var _stringReplaceAll2;
  var _useState = React.useState(null),
    dropDownData = _useState[0],
    setDropDownData = _useState[1];
  var _useState2 = React.useState([]),
    selectCityData = _useState2[0],
    setSelectCityData = _useState2[1];
  var _useState3 = React.useState([]);
  var history = reactRouterDom.useHistory();
  var selectedCities = [];
  var handleChangeCity = function handleChangeCity(city) {
    var _Digit$SessionStorage, _Digit$SessionStorage2, _Digit$SessionStorage3;
    var loggedInData = Digit.SessionStorage.get("citizen.userRequestObject");
    var filteredRoles = (_Digit$SessionStorage = Digit.SessionStorage.get("citizen.userRequestObject")) === null || _Digit$SessionStorage === void 0 ? void 0 : (_Digit$SessionStorage2 = _Digit$SessionStorage.info) === null || _Digit$SessionStorage2 === void 0 ? void 0 : (_Digit$SessionStorage3 = _Digit$SessionStorage2.roles) === null || _Digit$SessionStorage3 === void 0 ? void 0 : _Digit$SessionStorage3.filter(function (role) {
      return role.tenantId === city.value;
    });
    if ((filteredRoles === null || filteredRoles === void 0 ? void 0 : filteredRoles.length) > 0) {
      loggedInData.info.roles = filteredRoles;
      loggedInData.info.tenantId = city === null || city === void 0 ? void 0 : city.value;
    }
    Digit.SessionStorage.set("Employee.tenantId", city === null || city === void 0 ? void 0 : city.value);
    Digit.UserService.setUser(loggedInData);
    setDropDownData(city);
    if (window.location.href.includes("/digit-ui/employee/")) {
      var _location$state;
      var redirectPath = ((_location$state = location.state) === null || _location$state === void 0 ? void 0 : _location$state.from) || "/digit-ui/employee";
      history.replace(redirectPath);
    }
    window.location.reload();
  };
  React.useEffect(function () {
    var _userloggedValues$inf, _userloggedValues$inf2;
    var userloggedValues = Digit.SessionStorage.get("citizen.userRequestObject");
    var teantsArray = [],
      filteredArray = [];
    userloggedValues === null || userloggedValues === void 0 ? void 0 : (_userloggedValues$inf = userloggedValues.info) === null || _userloggedValues$inf === void 0 ? void 0 : (_userloggedValues$inf2 = _userloggedValues$inf.roles) === null || _userloggedValues$inf2 === void 0 ? void 0 : _userloggedValues$inf2.forEach(function (role) {
      return teantsArray.push(role.tenantId);
    });
    var unique = teantsArray.filter(function (item, i, ar) {
      return ar.indexOf(item) === i;
    });
    unique === null || unique === void 0 ? void 0 : unique.forEach(function (uniCode) {
      var _stringReplaceAll;
      filteredArray.push({
        label: prop === null || prop === void 0 ? void 0 : prop.t("TENANT_TENANTS_" + ((_stringReplaceAll = stringReplaceAll(uniCode, ".", "_")) === null || _stringReplaceAll === void 0 ? void 0 : _stringReplaceAll.toUpperCase())),
        value: uniCode
      });
    });
    selectedCities = filteredArray === null || filteredArray === void 0 ? void 0 : filteredArray.filter(function (select) {
      return select.value == Digit.SessionStorage.get("Employee.tenantId");
    });
    setSelectCityData(filteredArray);
  }, [dropDownData]);

  return /*#__PURE__*/React__default.createElement("div", {
    style: prop !== null && prop !== void 0 && prop.mobileView ? {
      color: "#767676"
    } : {}
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    option: selectCityData,
    selected: selectCityData.find(function (cityValue) {
      return cityValue.value === (dropDownData === null || dropDownData === void 0 ? void 0 : dropDownData.value);
    }),
    optionKey: "label",
    select: handleChangeCity,
    freeze: true,
    customSelector: /*#__PURE__*/React__default.createElement("label", {
      className: "cp"
    }, prop === null || prop === void 0 ? void 0 : prop.t("TENANT_TENANTS_" + ((_stringReplaceAll2 = stringReplaceAll(Digit.SessionStorage.get("Employee.tenantId"), ".", "_")) === null || _stringReplaceAll2 === void 0 ? void 0 : _stringReplaceAll2.toUpperCase())))
  }));
};

var ChangeLanguage = function ChangeLanguage(prop) {
  var isDropdown = prop.dropdown || false;
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    storeData = _Digit$Hooks$useStore.data,
    isLoading = _Digit$Hooks$useStore.isLoading;
  var _ref = storeData || {},
    languages = _ref.languages,
    stateInfo = _ref.stateInfo;
  var selectedLanguage = Digit.StoreData.getCurrentLanguage();
  var _useState = React.useState(selectedLanguage),
    selected = _useState[0],
    setselected = _useState[1];
  var handleChangeLanguage = function handleChangeLanguage(language) {
    setselected(language.value);
    Digit.LocalizationService.changeLanguage(language.value, stateInfo.code);
  };
  if (isLoading) return null;
  if (isDropdown) {
    return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
      option: languages,
      selected: languages.find(function (language) {
        return language.value === selectedLanguage;
      }),
      optionKey: "label",
      select: handleChangeLanguage,
      freeze: true,
      customSelector: /*#__PURE__*/React__default.createElement("label", {
        className: "cp"
      }, languages.find(function (language) {
        return language.value === selected;
      }).label)
    }));
  } else {
    return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
      style: {
        marginBottom: "5px"
      }
    }, "Language"), /*#__PURE__*/React__default.createElement("div", {
      className: "language-selector"
    }, languages.map(function (language, index) {
      return /*#__PURE__*/React__default.createElement("div", {
        className: "language-button-container",
        key: index
      }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CustomButton, {
        selected: language.value === selected,
        text: language.label,
        onClick: function onClick() {
          return handleChangeLanguage(language);
        }
      }));
    })));
  }
};

var TextToImg = function TextToImg(props) {
  return /*#__PURE__*/React__default.createElement("span", {
    className: "user-img-txt",
    onClick: props.toggleMenu,
    title: props.name
  }, props.name[0].toUpperCase());
};
var TopBar = function TopBar(_ref) {
  var _userDetails$info2, _cityDetails$city, _cityDetails$city2, _stateInfo$code, _userDetails$info3, _userDetails$info4, _userDetails$info4$us;
  var t = _ref.t,
    stateInfo = _ref.stateInfo,
    toggleSidebar = _ref.toggleSidebar,
    handleLogout = _ref.handleLogout,
    userDetails = _ref.userDetails,
    CITIZEN = _ref.CITIZEN,
    cityDetails = _ref.cityDetails,
    mobileView = _ref.mobileView,
    userOptions = _ref.userOptions,
    handleUserDropdownSelection = _ref.handleUserDropdownSelection,
    logoUrl = _ref.logoUrl,
    _ref$showLanguageChan = _ref.showLanguageChange,
    showLanguageChange = _ref$showLanguageChan === void 0 ? true : _ref$showLanguageChan;
  var _React$useState = React__default.useState(null),
    profilePic = _React$useState[0],
    setProfilePic = _React$useState[1];
  React__default.useEffect(function () {
    try {
      var _userDetails$info;
      var tenant = Digit.ULBService.getCurrentTenantId();
      var uuid = userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$info = userDetails.info) === null || _userDetails$info === void 0 ? void 0 : _userDetails$info.uuid;
      var _temp2 = function () {
        if (uuid) {
          return Promise.resolve(Digit.UserService.userSearch(tenant, {
            uuid: [uuid]
          }, {})).then(function (usersResponse) {
            if (usersResponse && usersResponse.user && usersResponse.user.length) {
              var _userDetails$photo;
              var _userDetails = usersResponse.user[0];
              var thumbs = _userDetails === null || _userDetails === void 0 ? void 0 : (_userDetails$photo = _userDetails.photo) === null || _userDetails$photo === void 0 ? void 0 : _userDetails$photo.split(",");
              setProfilePic(thumbs === null || thumbs === void 0 ? void 0 : thumbs.at(0));
            }
          });
        }
      }();
      return Promise.resolve(_temp2 && _temp2.then ? _temp2.then(function () {}) : void 0);
    } catch (e) {
      return Promise.reject(e);
    }
  }, [profilePic !== null, userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$info2 = userDetails.info) === null || _userDetails$info2 === void 0 ? void 0 : _userDetails$info2.uuid]);
  var CitizenHomePageTenantId = Digit.ULBService.getCitizenCurrentTenant(true);
  var history = reactRouterDom.useHistory();
  var _useLocation = reactRouterDom.useLocation(),
    pathname = _useLocation.pathname;
  var conditionsToDisableNotificationCountTrigger = function conditionsToDisableNotificationCountTrigger() {
    var _Digit$UserService, _Digit$UserService$ge, _Digit$UserService$ge2, _Digit$UserService2, _Digit$UserService2$g, _Digit$UserService2$g2;
    if (((_Digit$UserService = Digit.UserService) === null || _Digit$UserService === void 0 ? void 0 : (_Digit$UserService$ge = _Digit$UserService.getUser()) === null || _Digit$UserService$ge === void 0 ? void 0 : (_Digit$UserService$ge2 = _Digit$UserService$ge.info) === null || _Digit$UserService$ge2 === void 0 ? void 0 : _Digit$UserService$ge2.type) === "EMPLOYEE") return false;
    if (((_Digit$UserService2 = Digit.UserService) === null || _Digit$UserService2 === void 0 ? void 0 : (_Digit$UserService2$g = _Digit$UserService2.getUser()) === null || _Digit$UserService2$g === void 0 ? void 0 : (_Digit$UserService2$g2 = _Digit$UserService2$g.info) === null || _Digit$UserService2$g2 === void 0 ? void 0 : _Digit$UserService2$g2.type) === "CITIZEN") {
      if (!CitizenHomePageTenantId) return false;else return true;
    }
    return false;
  };
  var _Digit$Hooks$useNotif = Digit.Hooks.useNotificationCount({
      tenantId: CitizenHomePageTenantId,
      config: {
        enabled: conditionsToDisableNotificationCountTrigger()
      }
    }),
    _Digit$Hooks$useNotif2 = _Digit$Hooks$useNotif.data;
  _Digit$Hooks$useNotif2 = _Digit$Hooks$useNotif2 === void 0 ? {} : _Digit$Hooks$useNotif2;
  var unreadNotificationCount = _Digit$Hooks$useNotif2.unreadCount,
    notificationCountLoaded = _Digit$Hooks$useNotif.isSuccess;
  var updateSidebar = function updateSidebar() {
    if (!Digit.clikOusideFired) {
      toggleSidebar(true);
    } else {
      Digit.clikOusideFired = false;
    }
  };
  function onNotificationIconClick() {
    history.push("/digit-ui/citizen/engagement/notifications");
  }
  var urlsToDisableNotificationIcon = function urlsToDisableNotificationIcon(pathname) {
    var _Digit$UserService3, _Digit$UserService3$g;
    return !!((_Digit$UserService3 = Digit.UserService) !== null && _Digit$UserService3 !== void 0 && (_Digit$UserService3$g = _Digit$UserService3.getUser()) !== null && _Digit$UserService3$g !== void 0 && _Digit$UserService3$g.access_token) ? false : ["/digit-ui/citizen/select-language", "/digit-ui/citizen/select-location"].includes(pathname);
  };
  if (CITIZEN) {
    return /*#__PURE__*/React__default.createElement(digitUiReactComponents.TopBar, {
      img: stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.logoUrlWhite,
      isMobile: true,
      toggleSidebar: updateSidebar,
      logoUrl: stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.logoUrlWhite,
      onLogout: handleLogout,
      userDetails: userDetails,
      notificationCount: unreadNotificationCount < 99 ? unreadNotificationCount : 99,
      notificationCountLoaded: notificationCountLoaded,
      cityOfCitizenShownBesideLogo: t(CitizenHomePageTenantId),
      onNotificationIconClick: onNotificationIconClick,
      hideNotificationIconOnSomeUrlsWhenNotLoggedIn: urlsToDisableNotificationIcon(pathname)
    });
  }
  var loggedin = userDetails !== null && userDetails !== void 0 && userDetails.access_token ? true : false;
  return /*#__PURE__*/React__default.createElement("div", {
    className: "topbar"
  }, mobileView ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.Hamburger, {
    handleClick: toggleSidebar,
    color: "#9E9E9E"
  }) : null, /*#__PURE__*/React__default.createElement("img", {
    className: "city",
    src: loggedin ? cityDetails === null || cityDetails === void 0 ? void 0 : cityDetails.logoId : stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.statelogo
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      width: "100%"
    }
  }, loggedin && (cityDetails !== null && cityDetails !== void 0 && (_cityDetails$city = cityDetails.city) !== null && _cityDetails$city !== void 0 && _cityDetails$city.ulbGrade ? /*#__PURE__*/React__default.createElement("p", {
    className: "ulb",
    style: mobileView ? {
      fontSize: "14px",
      display: "inline-block"
    } : {}
  }, t(cityDetails === null || cityDetails === void 0 ? void 0 : cityDetails.i18nKey).toUpperCase(), " ", t("ULBGRADE_" + (cityDetails === null || cityDetails === void 0 ? void 0 : (_cityDetails$city2 = cityDetails.city) === null || _cityDetails$city2 === void 0 ? void 0 : _cityDetails$city2.ulbGrade.toUpperCase().replace(" ", "_").replace(".", "_"))).toUpperCase()) : /*#__PURE__*/React__default.createElement("img", {
    className: "state",
    src: logoUrl
  })), !loggedin && /*#__PURE__*/React__default.createElement("p", {
    className: "ulb",
    style: mobileView ? {
      fontSize: "14px",
      display: "inline-block"
    } : {}
  }, t("MYCITY_" + (stateInfo === null || stateInfo === void 0 ? void 0 : (_stateInfo$code = stateInfo.code) === null || _stateInfo$code === void 0 ? void 0 : _stateInfo$code.toUpperCase()) + "_LABEL"), " ", t("MYCITY_STATECODE_LABEL")), !mobileView && /*#__PURE__*/React__default.createElement("div", {
    className: mobileView ? "right" : "flex-right right w-80 column-gap-15",
    style: !loggedin ? {
      width: "80%"
    } : {}
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "left"
  }, !window.location.href.includes("employee/user/login") && !window.location.href.includes("employee/user/language-selection") && /*#__PURE__*/React__default.createElement(ChangeCity, {
    dropdown: true,
    t: t
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "left"
  }, showLanguageChange && /*#__PURE__*/React__default.createElement(ChangeLanguage, {
    dropdown: true
  })), (userDetails === null || userDetails === void 0 ? void 0 : userDetails.access_token) && /*#__PURE__*/React__default.createElement("div", {
    className: "left"
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    option: userOptions,
    optionKey: "name",
    select: handleUserDropdownSelection,
    showArrow: true,
    freeze: true,
    style: mobileView ? {
      right: 0
    } : {},
    optionCardStyles: {
      overflow: "revert"
    },
    customSelector: profilePic == null ? /*#__PURE__*/React__default.createElement(TextToImg, {
      name: (userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$info3 = userDetails.info) === null || _userDetails$info3 === void 0 ? void 0 : _userDetails$info3.name) || (userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$info4 = userDetails.info) === null || _userDetails$info4 === void 0 ? void 0 : (_userDetails$info4$us = _userDetails$info4.userInfo) === null || _userDetails$info4$us === void 0 ? void 0 : _userDetails$info4$us.name) || "Employee"
    }) : /*#__PURE__*/React__default.createElement("img", {
      src: profilePic,
      style: {
        height: "48px",
        width: "48px",
        borderRadius: "50%"
      }
    })
  })), /*#__PURE__*/React__default.createElement("img", {
    className: "state",
    src: logoUrl
  }))));
};

var SideBarMenu = function SideBarMenu(t, closeSidebar, redirectToLoginPage, isEmployee) {
  return [{
    type: "link",
    element: "HOME",
    text: t("HOME"),
    link: isEmployee ? "/digit-ui/employee/" : "/digit-ui/citizen/",
    icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.HomeIcon, {
      className: "icon"
    }),
    populators: {
      onClick: closeSidebar
    }
  }, {
    type: "component",
    element: "LANGUAGE",
    action: /*#__PURE__*/React__default.createElement(ChangeLanguage, null),
    icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.LanguageIcon, {
      className: "icon"
    })
  }, {
    id: 'login-btn',
    element: "LOGIN",
    text: t("LOGIN"),
    icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.LogoutIcon, {
      className: "icon"
    }),
    populators: {
      onClick: redirectToLoginPage
    }
  }];
};

var defaultImage = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAO4AAADUCAMAAACs0e/bAAAAM1BMVEXK0eL" + "/" + "/" + "/" + "/Dy97GzuD4+fvL0uPg5O7T2efb4OvR1+Xr7vTk5/Df4+37/P3v8fbO1eTt8PUsnq5FAAAGqElEQVR4nO2d25ajIBBFCajgvf/" + "/a0eMyZgEjcI5xgt7Hmatme507UaxuJXidiDqjmSgeVIMlB1ZR1WZAf2gbdu0QwixSYzjOJPmHurfEGEfY9XzjNGG9whQCeVAuv5xQEySLtR9hPuIcwj0EeroN5m3D1IbsbgHK0esiQ9MKs" + "qXVr8Hm/a/Pulk6wihpCIXBw3dh7bTvRBt9+dC5NfS1VH3xETdM3MxXRN1T0zUPTNR98xcS1dlV9NNfx3DhkTdM6PKqHteVBF1z0vU5f0sKdpc2zWLKutXrjJjdLvpesRmukqYonauPhXpds" + "Lb6CppmpnltsYIuY2yavi6Mi2/rzAWm1zUfF0limVLqkZyA+mDYevKBS37aGC+L1lX5e7uyU1Cv565uiua9k5LFqbqqrnu2I3m+jJ11ZoLeRtfmdB0Uw/ZDsP0VTxdn7a1VERfmq7Xl" + "Xyn5D2QWLoq8bZlPoBJumphJjVBw/Ll6CoTZGsTDs4NrGqKbqBth8ZHJUi6cn168QmleSm6GmB7Kxm+6obXlf7PoDHosCwM3QpiS2legi6ocSl3L0G3BdneDDgwQdENfeY+SfDJBkF37Z" + "B+GvwzA6/rMaafAn8143VhPZWdjMWG1oHXhdnemgPoAvLlB/iZyRTfVeF06wPoQhJmlm4bdcOAZRlRN5gcPc5SoPEQR1fDdbOo6wn+uYvXxY0QCLom6gYROKH+Aj5nvphuFXWDiLpRdxl" + "/19LFT95k6CHCrnW7pCDqBn1i1PUFvii2c11oZOJ6usWeH0RRNzC4Zs+6FTi2nevCVwCjbugnXklX5fkfTldL8PEilUB1kfNyN1u9MME2sATr4lbuB7AjfLAuvsRm1A0g6gYRdcPAjvBlje" + "2Z8brI8OC68AcRdlCkwLohx2mcZMjw9q+LzarQurjtnwPYAydX08WecECO/u6Ad0GBdYG7jO5gB4Ap+PwKcA9ZT43dn4/W9TyiPAn4OAJaF7h3uwe8StSCddFdM3jqFa2LvnnB5zzhuuBBAj" + "Y4gi50cg694gnXhTYvfMdrjtcFZhrwE9r41gUem8IXWMC3LrBzxh+a0gRd1N1LOK7M0IUUGuggvEmHoStA2/MJh7MpupiDU4TzjhxdzLAoO4ouZvqVURbFMHQlZD6SUeWHoguZsSLUGegreh" + "A+FZFowPdUWTi6iMoZlIpGGUUXkDbjj/9ZOLqAQS/+GIKl5BQOCn/ycqpzkXSDm5dU7ZWkG7wUyGlcmm7g5Ux56AqirgoaJ7BeokPTDbp9CbVunjFxPrl7+HqnkrSq1Da7JX20f3dV8yJi6v" + "oO81mX8vV0mx3qUsZCPRfTlVRdz2EvdufYGDvNQvvwqHtmXd+a1ITinwNcXc+lT6JuzdT1XDyBn/x7wtX1HCQQdW9MXc8xArGrirowfLeUEbMqqq6f7TF1lfRdOuGNiGi6SpT+WxY06xUfNN" + "2wBfyE9I4tlm7w5hvOPDNJN3yNiLMipji6gE3chKhouoCtN5x3QlF0EZt8OW/8ougitqJQlk1aii7iFC9l0MvRReyao7xNjKML2Z/PuHlzhi5mFxljiZeiC9rPTEisNEMX9KYAwo5Xhi7qaA" + "3hamboYm7dG+NVrXhdaYDv5zFaQZsYrCtbbAGnjkQDX2+J1FXCwOsqWOpKoIQNTFdqYBWydxqNqUoG0pVpCS+H8kaJaGKErlIaXj7CRRE+gRWuKwW9YZ80oVOUgbpdT0zpnSZJTIiwCtJVelv" + "Xntr4P5j6BWfPb5Wcx84C4cq3hb11lco2u2Mdwp6XdJ/Ne3wb8DWdfiRenZaXrhLwOj4e+GQeHroy3YOspS7TlU28Wle2m2QUS0mqdcbrdNW+ZHsSsyK7tBfm0q/dWcv+Z3mytVx3t7KWulq" + "Ue6ilunu8jF8pFwgv1FXp3mUt35OtRbr7eM4u4Gs6vUBXgeuHc5kfE/cbvWZtkROLm1DMtLCy80tzsu2PRj0hTI8fvrQuvsjlJkyutszq+m423wHaLTyniy/XuiGZ84LuT+m5ZfNfRxyGs7L" + "XZOvia7VujatUwVTrIt+Q/Csc7Tuhe+BOakT10b4TuoiiJjvgU9emTO42PwEfBa+cuodKkuf42DXr1D3JpXz73Hnn0j10evHKe+nufgfUm+7B84sX9FfdEzXux2DBpWuKokkCqN/5pa/8pmvn" + "L+RGKCddCGmatiPyPB/+ekO/M/q/7uvbt22kTt3zEnXPzCV13T3Gel4/6NduDu66xRvlPNkM1RjjxUdv+4WhGx6TftD19Q/dfzpwcHO+rE3fAAAAAElFTkSuQmCC";
var Profile = function Profile(_ref) {
  var info = _ref.info,
    t = _ref.t;
  var _React$useState = React__default.useState(null),
    profilePic = _React$useState[0],
    setProfilePic = _React$useState[1];
  React__default.useEffect(function () {
    try {
      var tenant = Digit.ULBService.getCurrentTenantId();
      var uuid = info === null || info === void 0 ? void 0 : info.uuid;
      var _temp2 = function () {
        if (uuid) {
          return Promise.resolve(Digit.UserService.userSearch(tenant, {
            uuid: [uuid]
          }, {})).then(function (usersResponse) {
            if (usersResponse && usersResponse.user && usersResponse.user.length) {
              var _userDetails$photo;
              var userDetails = usersResponse.user[0];
              var thumbs = userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$photo = userDetails.photo) === null || _userDetails$photo === void 0 ? void 0 : _userDetails$photo.split(",");
              setProfilePic(thumbs === null || thumbs === void 0 ? void 0 : thumbs.at(0));
            }
          });
        }
      }();
      return Promise.resolve(_temp2 && _temp2.then ? _temp2.then(function () {}) : void 0);
    } catch (e) {
      return Promise.reject(e);
    }
  }, [profilePic !== null]);
  return /*#__PURE__*/React__default.createElement("div", {
    className: "profile-section"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "imageloader imageloader-loaded"
  }, /*#__PURE__*/React__default.createElement("img", {
    className: "img-responsive img-circle img-Profile",
    src: profilePic ? profilePic : defaultImage,
    style: {
      objectFit: "cover",
      objectPosition: "center"
    }
  })), /*#__PURE__*/React__default.createElement("div", {
    id: "profile-name",
    className: "label-container name-Profile"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "label-text"
  }, " ", info === null || info === void 0 ? void 0 : info.name, " ")), /*#__PURE__*/React__default.createElement("div", {
    id: "profile-location",
    className: "label-container loc-Profile"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "label-text"
  }, " ", info === null || info === void 0 ? void 0 : info.mobileNumber, " ")), (info === null || info === void 0 ? void 0 : info.emailId) && /*#__PURE__*/React__default.createElement("div", {
    id: "profile-emailid",
    className: "label-container loc-Profile"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "label-text"
  }, " ", info.emailId, " ")), window.location.href.includes("/employee") && !window.location.href.includes("/employee/user/login") && !window.location.href.includes("employee/user/language-selection") && /*#__PURE__*/React__default.createElement(ChangeCity, {
    t: t,
    mobileView: true
  }));
};
var PoweredBy = function PoweredBy() {
  var _window, _window$globalConfigs, _window$globalConfigs2;
  return /*#__PURE__*/React__default.createElement("div", {
    className: "digit-footer"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "Powered by DIGIT",
    src: (_window = window) === null || _window === void 0 ? void 0 : (_window$globalConfigs = _window.globalConfigs) === null || _window$globalConfigs === void 0 ? void 0 : (_window$globalConfigs2 = _window$globalConfigs.getConfig) === null || _window$globalConfigs2 === void 0 ? void 0 : _window$globalConfigs2.call(_window$globalConfigs, "DIGIT_FOOTER"),
    style: {
      cursor: "pointer"
    },
    onClick: function onClick() {
      var _window2, _window2$globalConfig, _window2$globalConfig2;
      window.open((_window2 = window) === null || _window2 === void 0 ? void 0 : (_window2$globalConfig = _window2.globalConfigs) === null || _window2$globalConfig === void 0 ? void 0 : (_window2$globalConfig2 = _window2$globalConfig.getConfig) === null || _window2$globalConfig2 === void 0 ? void 0 : _window2$globalConfig2.call(_window2$globalConfig, "DIGIT_HOME_URL"), "_blank").focus();
    }
  }), " ");
};
var CitizenSideBar = function CitizenSideBar(_ref2) {
  var isOpen = _ref2.isOpen,
    toggleSidebar = _ref2.toggleSidebar,
    onLogout = _ref2.onLogout,
    _ref2$isEmployee = _ref2.isEmployee,
    isEmployee = _ref2$isEmployee === void 0 ? false : _ref2$isEmployee;
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    storeData = _Digit$Hooks$useStore.data,
    isFetched = _Digit$Hooks$useStore.isFetched;
  var _ref3 = storeData || {},
    stateInfo = _ref3.stateInfo;
  var user = Digit.UserService.getUser();
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var closeSidebar = function closeSidebar() {
    Digit.clikOusideFired = true;
    toggleSidebar(false);
  };
  var tenantId = Digit.ULBService.getCurrentTenantId();
  var showProfilePage = function showProfilePage() {
    var redirectUrl = isEmployee ? "/digit-ui/employee/user/profile" : "/digit-ui/citizen/user/profile";
    history.push(redirectUrl);
    closeSidebar();
  };
  var redirectToLoginPage = function redirectToLoginPage() {
    history.push("/digit-ui/citizen/login");
    closeSidebar();
  };
  var menuItems = [].concat(SideBarMenu(t, closeSidebar, redirectToLoginPage, isEmployee));
  var profileItem;
  if (isFetched && user && user.access_token) {
    profileItem = /*#__PURE__*/React__default.createElement(Profile, {
      info: user === null || user === void 0 ? void 0 : user.info,
      stateName: stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.name,
      t: t
    });
    menuItems = menuItems.filter(function (item) {
      return (item === null || item === void 0 ? void 0 : item.id) !== "login-btn";
    });
    menuItems = [].concat(menuItems, [{
      text: t("EDIT_PROFILE"),
      element: "PROFILE",
      icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.EditPencilIcon, {
        className: "icon edit-btn-ico",
        width: "16",
        height: "16"
      }),
      populators: {
        onClick: showProfilePage
      }
    }, {
      text: t("CORE_COMMON_LOGOUT"),
      element: "LOGOUT",
      icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.LogoutIcon, {
        className: "icon"
      }),
      populators: {
        onClick: onLogout
      }
    }, {
      text: /*#__PURE__*/React__default.createElement(React__default.Fragment, null, t("CS_COMMON_HELPLINE"), /*#__PURE__*/React__default.createElement("div", {
        className: "telephone",
        style: {
          marginTop: "-10%"
        }
      }, storeData === null || storeData === void 0 ? void 0 : storeData.tenants.map(function (i) {
      }), /*#__PURE__*/React__default.createElement("div", {
        className: "link"
      }, /*#__PURE__*/React__default.createElement("a", {
        href: "tel:" + (storeData === null || storeData === void 0 ? void 0 : storeData.tenants[0].contactNumber)
      }, storeData === null || storeData === void 0 ? void 0 : storeData.tenants[0].contactNumber)))),
      element: "Helpline",
      icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.Phone, {
        className: "icon"
      })
    }]);
  }

  if (history.location.pathname.includes("/openlink")) {
    profileItem = /*#__PURE__*/React__default.createElement("span", null);
    menuItems = menuItems.filter(function (ele) {
      return ele.element === "LANGUAGE";
    });
  }
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.NavBar, {
    open: isOpen,
    profileItem: profileItem,
    menuItems: menuItems,
    onClose: closeSidebar,
    Footer: /*#__PURE__*/React__default.createElement(PoweredBy, null)
  }));
};

var ToolTipWrapper = function ToolTipWrapper(_ref) {
  var child = _ref.child,
    label = _ref.label,
    t = _ref.t;
  return /*#__PURE__*/React__default.createElement("span", {
    className: "tooltip"
  }, child, /*#__PURE__*/React__default.createElement("span", {
    className: "tooltiptext",
    style: {
      fontSize: "14px",
      marginLeft: "0px",
      bottom: "20%"
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      background: "#555",
      width: "100%",
      padding: "5px",
      borderRadius: "6px",
      whiteSpace: "pre"
    }
  }, t(label))));
};
var EmployeeSideBar = function EmployeeSideBar() {
  var STADMIN = Digit.UserService.hasAccess("STADMIN");
  var NATADMIN = Digit.UserService.hasAccess("NATADMIN");
  function getRedirectionUrl() {
    if (NATADMIN) return "/digit-ui/employee/payment/integration/dss/NURT_DASHBOARD";else if (STADMIN) return "/employee/integration/dss/home";else return "/employee";
  }
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  return /*#__PURE__*/React__default.createElement("div", {
    className: "sidebar"
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.Link, {
    to: "/digit-ui/employee"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "actions active"
  }, /*#__PURE__*/React__default.createElement(ToolTipWrapper, {
    child: /*#__PURE__*/React__default.createElement("svg", {
      xmlns: "http://www.w3.org/2000/svg",
      height: "24",
      viewBox: "0 0 24 24",
      width: "24"
    }, /*#__PURE__*/React__default.createElement("path", {
      d: "M0 0h24v24H0z",
      fill: "none"
    }), /*#__PURE__*/React__default.createElement("path", {
      d: "M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z",
      fill: "white"
    })),
    t: t,
    label: "ACTION_TEST_HOME"
  }))), /*#__PURE__*/React__default.createElement("a", {
    href: getRedirectionUrl()
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "actions"
  }, /*#__PURE__*/React__default.createElement(ToolTipWrapper, {
    child: /*#__PURE__*/React__default.createElement("svg", {
      xmlns: "http://www.w3.org/2000/svg",
      height: "24",
      viewBox: "0 0 24 24",
      width: "24"
    }, /*#__PURE__*/React__default.createElement("path", {
      d: "M0 0h24v24H0z",
      fill: "none"
    }), /*#__PURE__*/React__default.createElement("path", {
      d: "M8.17 5.7L1 10.48V21h5v-8h4v8h5V10.25z",
      fill: "white"
    }), /*#__PURE__*/React__default.createElement("path", {
      d: "M17 7h2v2h-2z",
      fill: "none"
    }), /*#__PURE__*/React__default.createElement("path", {
      d: "M10 3v1.51l2 1.33L13.73 7H15v.85l2 1.34V11h2v2h-2v2h2v2h-2v4h6V3H10zm9 6h-2V7h2v2z",
      fill: "white"
    })),
    t: t,
    label: "CORE_CHANGE_TENANT_DESCRIPTION"
  }))));
};

var SideBar = function SideBar(_ref) {
  var CITIZEN = _ref.CITIZEN,
    isSidebarOpen = _ref.isSidebarOpen,
    toggleSidebar = _ref.toggleSidebar,
    handleLogout = _ref.handleLogout,
    mobileView = _ref.mobileView,
    userDetails = _ref.userDetails;
  if (CITIZEN) return /*#__PURE__*/React__default.createElement(CitizenSideBar, {
    isOpen: isSidebarOpen,
    isMobile: true,
    toggleSidebar: toggleSidebar,
    onLogout: handleLogout
  });else {
    if (!mobileView && userDetails !== null && userDetails !== void 0 && userDetails.access_token) return /*#__PURE__*/React__default.createElement(EmployeeSideBar, {
      mobileView: mobileView,
      userDetails: userDetails
    });else return /*#__PURE__*/React__default.createElement(CitizenSideBar, {
      isOpen: isSidebarOpen,
      isMobile: true,
      toggleSidebar: toggleSidebar,
      onLogout: handleLogout,
      isEmployee: true
    });
  }
};

var TopBarSideBar = function TopBarSideBar(_ref) {
  var t = _ref.t,
    stateInfo = _ref.stateInfo,
    userDetails = _ref.userDetails,
    CITIZEN = _ref.CITIZEN,
    cityDetails = _ref.cityDetails,
    mobileView = _ref.mobileView,
    handleUserDropdownSelection = _ref.handleUserDropdownSelection,
    logoUrl = _ref.logoUrl,
    _ref$showSidebar = _ref.showSidebar,
    showSidebar = _ref$showSidebar === void 0 ? true : _ref$showSidebar,
    showLanguageChange = _ref.showLanguageChange;
  var _useState = React.useState(false),
    isSidebarOpen = _useState[0],
    toggleSidebar = _useState[1];
  var history = reactRouterDom.useHistory();
  var handleLogout = function handleLogout() {
    toggleSidebar(false);
    Digit.UserService.logout();
  };
  var userProfile = function userProfile() {
    history.push("/digit-ui/employee/user/profile");
  };
  var userOptions = [{
    name: t("EDIT_PROFILE"),
    icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.EditPencilIcon, {
      className: "icon"
    }),
    func: userProfile
  }, {
    name: t("CORE_COMMON_LOGOUT"),
    icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.LogoutIcon, {
      className: "icon"
    }),
    func: handleLogout
  }];
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(TopBar, {
    t: t,
    stateInfo: stateInfo,
    toggleSidebar: toggleSidebar,
    isSidebarOpen: isSidebarOpen,
    handleLogout: handleLogout,
    userDetails: userDetails,
    CITIZEN: CITIZEN,
    cityDetails: cityDetails,
    mobileView: mobileView,
    userOptions: userOptions,
    handleUserDropdownSelection: handleUserDropdownSelection,
    logoUrl: logoUrl,
    showLanguageChange: showLanguageChange
  }), showSidebar && /*#__PURE__*/React__default.createElement(SideBar, {
    t: t,
    CITIZEN: CITIZEN,
    isSidebarOpen: isSidebarOpen,
    toggleSidebar: toggleSidebar,
    handleLogout: handleLogout,
    mobileView: mobileView,
    userDetails: userDetails
  }));
};

function UploadDrawer(_ref) {
  var setProfilePic = _ref.setProfilePic,
    closeDrawer = _ref.closeDrawer,
    userType = _ref.userType,
    removeProfilePic = _ref.removeProfilePic,
    showToast = _ref.showToast;
  var _useState = React.useState(null),
    setUploadedFile = _useState[1];
  var _useState2 = React.useState(""),
    file = _useState2[0],
    setFile = _useState2[1];
  var _useState3 = React.useState(null),
    setError = _useState3[1];
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var selectfile = function selectfile(e) {
    return setFile(e.target.files[0]);
  };
  var removeimg = function removeimg() {
    removeProfilePic();
    closeDrawer();
  };
  var onOverlayBodyClick = function onOverlayBodyClick() {
    return closeDrawer();
  };
  React.useEffect(function () {
    (function () {
      try {
        setError(null);
        var _temp4 = function () {
          if (file) {
            var _temp5 = function () {
              if (file.size >= 1000000) {
                showToast("error", t("CORE_COMMON_PROFILE_MAXIMUM_UPLOAD_SIZE_EXCEEDED"));
                setError(t("CORE_COMMON_PROFILE_MAXIMUM_UPLOAD_SIZE_EXCEEDED"));
              } else {
                var _temp6 = _catch(function () {
                  return Promise.resolve(Digit.UploadServices.Filestorage(userType + "-profile", file, Digit.ULBService.getStateId())).then(function (response) {
                    var _response$data, _response$data$files;
                    if ((response === null || response === void 0 ? void 0 : (_response$data = response.data) === null || _response$data === void 0 ? void 0 : (_response$data$files = _response$data.files) === null || _response$data$files === void 0 ? void 0 : _response$data$files.length) > 0) {
                      var _response$data2, _response$data2$files;
                      var fileStoreId = response === null || response === void 0 ? void 0 : (_response$data2 = response.data) === null || _response$data2 === void 0 ? void 0 : (_response$data2$files = _response$data2.files[0]) === null || _response$data2$files === void 0 ? void 0 : _response$data2$files.fileStoreId;
                      setUploadedFile(fileStoreId);
                      setProfilePic(fileStoreId);
                    } else {
                      showToast("error", t("CORE_COMMON_PROFILE_FILE_UPLOAD_ERROR"));
                      setError(t("CORE_COMMON_PROFILE_FILE_UPLOAD_ERROR"));
                    }
                  });
                }, function () {
                  showToast("error", t("CORE_COMMON_PROFILE_INVALID_FILE_INPUT"));
                });
                if (_temp6 && _temp6.then) return _temp6.then(function () {});
              }
            }();
            if (_temp5 && _temp5.then) return _temp5.then(function () {});
          }
        }();
        return _temp4 && _temp4.then ? _temp4.then(function () {}) : void 0;
      } catch (e) {
        Promise.reject(e);
      }
    })();
  }, [file]);
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    style: {
      position: "fixed",
      top: "0",
      left: "0",
      right: "0",
      bottom: "0",
      width: "100%",
      height: "100vh",
      backgroundColor: "rgba(0,0,0,.5)"
    },

    onClick: onOverlayBodyClick
  }), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%",
      justifyContent: "space-between",
      display: "flex",
      backgroundColor: "white",
      alignItems: "center",
      position: "fixed",
      left: "0",
      right: "0",
      height: "20%",
      bottom: userType === "citizen" ? "2.5rem" : "0",
      zIndex: "2"
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: "flex",
      flex: "1",
      flexDirection: "column",
      width: "100%",
      justifyContent: "center",
      alignItems: "center",
      gap: "8px 0"
    }
  }, /*#__PURE__*/React__default.createElement("label", {
    for: "file",
    style: {
      cursor: "pointer"
    }
  }, " ", /*#__PURE__*/React__default.createElement(digitUiReactComponents.GalleryIcon, null)), /*#__PURE__*/React__default.createElement("label", {
    style: {
      cursor: "pointer"
    }
  }, " Gallery"), /*#__PURE__*/React__default.createElement("input", {
    type: "file",
    id: "file",
    accept: "image/*, .png, .jpeg, .jpg",
    onChange: selectfile,
    style: {
      display: "none"
    }
  })), /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: "flex",
      flex: "1",
      width: "100%",
      justifyContent: "center",
      alignItems: "center",
      flexDirection: "column",
      gap: "8px 0"
    }
  }, /*#__PURE__*/React__default.createElement("button", {
    onClick: removeimg
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.RemoveIcon, null)), /*#__PURE__*/React__default.createElement("label", {
    style: {
      cursor: "pointer"
    }
  }, "Remove"))));
}

var defaultImage$1 = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAO4AAADUCAMAAACs0e/bAAAAM1BMVEXK0eL" + "/" + "/" + "/" + "/Dy97GzuD4+fvL0uPg5O7T2efb4OvR1+Xr7vTk5/Df4+37/P3v8fbO1eTt8PUsnq5FAAAGqElEQVR4nO2d25ajIBBFCajgvf/" + "/a0eMyZgEjcI5xgt7Hmatme507UaxuJXidiDqjmSgeVIMlB1ZR1WZAf2gbdu0QwixSYzjOJPmHurfEGEfY9XzjNGG9whQCeVAuv5xQEySLtR9hPuIcwj0EeroN5m3D1IbsbgHK0esiQ9MKs" + "qXVr8Hm/a/Pulk6wihpCIXBw3dh7bTvRBt9+dC5NfS1VH3xETdM3MxXRN1T0zUPTNR98xcS1dlV9NNfx3DhkTdM6PKqHteVBF1z0vU5f0sKdpc2zWLKutXrjJjdLvpesRmukqYonauPhXpds" + "Lb6CppmpnltsYIuY2yavi6Mi2/rzAWm1zUfF0limVLqkZyA+mDYevKBS37aGC+L1lX5e7uyU1Cv565uiua9k5LFqbqqrnu2I3m+jJ11ZoLeRtfmdB0Uw/ZDsP0VTxdn7a1VERfmq7Xl" + "Xyn5D2QWLoq8bZlPoBJumphJjVBw/Ll6CoTZGsTDs4NrGqKbqBth8ZHJUi6cn168QmleSm6GmB7Kxm+6obXlf7PoDHosCwM3QpiS2legi6ocSl3L0G3BdneDDgwQdENfeY+SfDJBkF37Z" + "B+GvwzA6/rMaafAn8143VhPZWdjMWG1oHXhdnemgPoAvLlB/iZyRTfVeF06wPoQhJmlm4bdcOAZRlRN5gcPc5SoPEQR1fDdbOo6wn+uYvXxY0QCLom6gYROKH+Aj5nvphuFXWDiLpRdxl" + "/19LFT95k6CHCrnW7pCDqBn1i1PUFvii2c11oZOJ6usWeH0RRNzC4Zs+6FTi2nevCVwCjbugnXklX5fkfTldL8PEilUB1kfNyN1u9MME2sATr4lbuB7AjfLAuvsRm1A0g6gYRdcPAjvBlje" + "2Z8brI8OC68AcRdlCkwLohx2mcZMjw9q+LzarQurjtnwPYAydX08WecECO/u6Ad0GBdYG7jO5gB4Ap+PwKcA9ZT43dn4/W9TyiPAn4OAJaF7h3uwe8StSCddFdM3jqFa2LvnnB5zzhuuBBAj" + "Y4gi50cg694gnXhTYvfMdrjtcFZhrwE9r41gUem8IXWMC3LrBzxh+a0gRd1N1LOK7M0IUUGuggvEmHoStA2/MJh7MpupiDU4TzjhxdzLAoO4ouZvqVURbFMHQlZD6SUeWHoguZsSLUGegreh" + "A+FZFowPdUWTi6iMoZlIpGGUUXkDbjj/9ZOLqAQS/+GIKl5BQOCn/ycqpzkXSDm5dU7ZWkG7wUyGlcmm7g5Ux56AqirgoaJ7BeokPTDbp9CbVunjFxPrl7+HqnkrSq1Da7JX20f3dV8yJi6v" + "oO81mX8vV0mx3qUsZCPRfTlVRdz2EvdufYGDvNQvvwqHtmXd+a1ITinwNcXc+lT6JuzdT1XDyBn/x7wtX1HCQQdW9MXc8xArGrirowfLeUEbMqqq6f7TF1lfRdOuGNiGi6SpT+WxY06xUfNN" + "2wBfyE9I4tlm7w5hvOPDNJN3yNiLMipji6gE3chKhouoCtN5x3QlF0EZt8OW/8ougitqJQlk1aii7iFC9l0MvRReyao7xNjKML2Z/PuHlzhi5mFxljiZeiC9rPTEisNEMX9KYAwo5Xhi7qaA" + "3hamboYm7dG+NVrXhdaYDv5zFaQZsYrCtbbAGnjkQDX2+J1FXCwOsqWOpKoIQNTFdqYBWydxqNqUoG0pVpCS+H8kaJaGKErlIaXj7CRRE+gRWuKwW9YZ80oVOUgbpdT0zpnSZJTIiwCtJVelv" + "Xntr4P5j6BWfPb5Wcx84C4cq3hb11lco2u2Mdwp6XdJ/Ne3wb8DWdfiRenZaXrhLwOj4e+GQeHroy3YOspS7TlU28Wle2m2QUS0mqdcbrdNW+ZHsSsyK7tBfm0q/dWcv+Z3mytVx3t7KWulq" + "Ue6ilunu8jF8pFwgv1FXp3mUt35OtRbr7eM4u4Gs6vUBXgeuHc5kfE/cbvWZtkROLm1DMtLCy80tzsu2PRj0hTI8fvrQuvsjlJkyutszq+m423wHaLTyniy/XuiGZ84LuT+m5ZfNfRxyGs7L" + "XZOvia7VujatUwVTrIt+Q/Csc7Tuhe+BOakT10b4TuoiiJjvgU9emTO42PwEfBa+cuodKkuf42DXr1D3JpXz73Hnn0j10evHKe+nufgfUm+7B84sX9FfdEzXux2DBpWuKokkCqN/5pa/8pmvn" + "L+RGKCddCGmatiPyPB/+ekO/M/q/7uvbt22kTt3zEnXPzCV13T3Gel4/6NduDu66xRvlPNkM1RjjxUdv+4WhGx6TftD19Q/dfzpwcHO+rE3fAAAAAElFTkSuQmCC";
var UserProfile = function UserProfile(_ref) {
  var _Digit$UserService$ge, _errors$userName, _errors$emailAddress, _errors$userName2, _errors$mobileNumber, _errors$emailAddress2, _ref3, _errors$currentPasswo, _errors$newPassword, _errors$confirmPasswo;
  var stateCode = _ref.stateCode,
    userType = _ref.userType,
    cityDetails = _ref.cityDetails;
  var history = reactRouterDom.useHistory();
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var url = window.location.href;
  var stateId = Digit.ULBService.getStateId();
  var tenant = Digit.ULBService.getCurrentTenantId();
  var userInfo = ((_Digit$UserService$ge = Digit.UserService.getUser()) === null || _Digit$UserService$ge === void 0 ? void 0 : _Digit$UserService$ge.info) || {};
  var _useState = React.useState(null),
    userDetails = _useState[0],
    setUserDetails = _useState[1];
  var _useState2 = React.useState(userInfo !== null && userInfo !== void 0 && userInfo.name ? userInfo.name : ""),
    name = _useState2[0],
    setName = _useState2[1];
  var _useState3 = React.useState(userInfo !== null && userInfo !== void 0 && userInfo.emailId ? userInfo.emailId : ""),
    email = _useState3[0],
    setEmail = _useState3[1];
  var _useState4 = React.useState(userDetails === null || userDetails === void 0 ? void 0 : userDetails.gender),
    gender = _useState4[0],
    setGender = _useState4[1];
  var _useState5 = React.useState(userInfo !== null && userInfo !== void 0 && userInfo.permanentCity ? userInfo.permanentCity : cityDetails.name),
    city = _useState5[0],
    setCity = _useState5[1];
  var _useState6 = React.useState(userInfo !== null && userInfo !== void 0 && userInfo.mobileNumber ? userInfo.mobileNumber : ""),
    mobileNumber = _useState6[0],
    setMobileNo = _useState6[1];
  var _useState7 = React.useState(null),
    profilePic = _useState7[0],
    setProfilePic = _useState7[1];
  var _useState8 = React.useState(""),
    profileImg = _useState8[0],
    setProfileImg = _useState8[1];
  var _useState9 = React.useState(false),
    openUploadSlide = _useState9[0],
    setOpenUploadSide = _useState9[1];
  var _useState10 = React.useState(false),
    changepassword = _useState10[0],
    setChangepassword = _useState10[1];
  var _useState11 = React.useState(""),
    currentPassword = _useState11[0],
    setCurrentPassword = _useState11[1];
  var _useState12 = React.useState(""),
    newPassword = _useState12[0],
    setNewPassword = _useState12[1];
  var _useState13 = React.useState(""),
    confirmPassword = _useState13[0],
    setConfirmPassword = _useState13[1];
  var _useState14 = React.useState(null),
    toast = _useState14[0],
    setToast = _useState14[1];
  var _useState15 = React.useState(false),
    loading = _useState15[0],
    setLoading = _useState15[1];
  var _React$useState = React__default.useState(window.innerWidth),
    windowWidth = _React$useState[0],
    setWindowWidth = _React$useState[1];
  var _React$useState2 = React__default.useState({}),
    errors = _React$useState2[0],
    setErrors = _React$useState2[1];
  var getUserInfo = function getUserInfo() {
    try {
      var uuid = userInfo === null || userInfo === void 0 ? void 0 : userInfo.uuid;
      var _temp2 = function () {
        if (uuid) {
          return Promise.resolve(Digit.UserService.userSearch(tenant, {
            uuid: [uuid]
          }, {})).then(function (usersResponse) {
            usersResponse && usersResponse.user && usersResponse.user.length && setUserDetails(usersResponse.user[0]);
          });
        }
      }();
      return Promise.resolve(_temp2 && _temp2.then ? _temp2.then(function () {}) : void 0);
    } catch (e) {
      return Promise.reject(e);
    }
  };
  React__default.useEffect(function () {
    window.addEventListener("resize", function () {
      return setWindowWidth(window.innerWidth);
    });
    return function () {
      window.removeEventListener("resize", function () {
        return setWindowWidth(window.innerWidth);
      });
    };
  });
  React.useEffect(function () {
    var _userDetails$photo;
    setLoading(true);
    getUserInfo();
    setGender({
      i18nKey: undefined,
      code: userDetails === null || userDetails === void 0 ? void 0 : userDetails.gender,
      value: userDetails === null || userDetails === void 0 ? void 0 : userDetails.gender
    });
    var thumbs = userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$photo = userDetails.photo) === null || _userDetails$photo === void 0 ? void 0 : _userDetails$photo.split(",");
    setProfileImg(thumbs === null || thumbs === void 0 ? void 0 : thumbs.at(0));
    setLoading(false);
  }, [userDetails !== null]);
  var validation = {};
  var editScreen = false;
  var onClickAddPic = function onClickAddPic() {
    return setOpenUploadSide(!openUploadSlide);
  };
  var TogleforPassword = function TogleforPassword() {
    return setChangepassword(!changepassword);
  };
  var setGenderName = function setGenderName(value) {
    return setGender(value);
  };
  var closeFileUploadDrawer = function closeFileUploadDrawer() {
    return setOpenUploadSide(false);
  };
  var setUserName = function setUserName(value) {
    setName(value);
    if (!new RegExp(/^[a-zA-Z ]+$/i).test(value) || value.length === 0 || value.length > 50) {
      setErrors(_extends({}, errors, {
        userName: {
          type: "pattern",
          message: "CORE_COMMON_PROFILE_NAME_INVALID"
        }
      }));
    } else {
      setErrors(_extends({}, errors, {
        userName: null
      }));
    }
  };
  var setUserEmailAddress = function setUserEmailAddress(value) {
    setEmail(value);
    if (value.length && !(value.includes("@") && value.includes("."))) {
      setErrors(_extends({}, errors, {
        emailAddress: {
          type: "pattern",
          message: "CORE_COMMON_PROFILE_EMAIL_INVALID"
        }
      }));
    } else {
      setErrors(_extends({}, errors, {
        emailAddress: null
      }));
    }
  };
  var setUserMobileNumber = function setUserMobileNumber(value) {
    setMobileNo(value);
    if (userType === "employee" && !new RegExp(/^[6-9]{1}[0-9]{9}$/).test(value)) {
      setErrors(_extends({}, errors, {
        mobileNumber: {
          type: 'pattern',
          message: "CORE_COMMON_PROFILE_MOBILE_NUMBER_INVALID"
        }
      }));
    } else {
      setErrors(_extends({}, errors, {
        mobileNumber: null
      }));
    }
  };
  var setUserCurrentPassword = function setUserCurrentPassword(value) {
    setCurrentPassword(value);
    if (value.length && !new RegExp(/^([a-zA-Z0-9@#$%]{8,15})$/i).test(value)) {
      setErrors(_extends({}, errors, {
        currentPassword: {
          type: "pattern",
          message: "CORE_COMMON_PROFILE_PASSWORD_INVALID"
        }
      }));
    } else {
      setErrors(_extends({}, errors, {
        currentPassword: null
      }));
    }
  };
  var setUserNewPassword = function setUserNewPassword(value) {
    setNewPassword(value);
    if (value.length && !new RegExp(/^([a-zA-Z0-9@#$%]{8,15})$/i).test(value)) {
      setErrors(_extends({}, errors, {
        newPassword: {
          type: "pattern",
          message: "CORE_COMMON_PROFILE_PASSWORD_INVALID"
        }
      }));
    } else {
      setErrors(_extends({}, errors, {
        newPassword: null
      }));
    }
  };
  var setUserConfirmPassword = function setUserConfirmPassword(value) {
    setConfirmPassword(value);
    if (value.length && !new RegExp(/^([a-zA-Z0-9@#$%]{8,15})$/i).test(value)) {
      setErrors(_extends({}, errors, {
        confirmPassword: {
          type: "pattern",
          message: "CORE_COMMON_PROFILE_PASSWORD_INVALID"
        }
      }));
    } else {
      setErrors(_extends({}, errors, {
        confirmPassword: null
      }));
    }
  };
  var removeProfilePic = function removeProfilePic() {
    setProfilePic(null);
    setProfileImg(null);
  };
  var showToast = function showToast(type, message, duration) {
    if (duration === void 0) {
      duration = 5000;
    }
    setToast({
      key: type,
      action: message
    });
    setTimeout(function () {
      setToast(null);
    }, duration);
  };
  var updateProfile = function updateProfile() {
    try {
      var _temp5 = function _temp5(_result) {
        if (_exit2) return _result;
        setLoading(false);
      };
      var _exit2 = false;
      setLoading(true);
      var _temp6 = _catch(function () {
        var requestData = _extends({}, userInfo, {
          name: name,
          gender: gender === null || gender === void 0 ? void 0 : gender.value,
          emailId: email,
          photo: profilePic
        });
        if (!new RegExp(/^([a-zA-Z ])*$/).test(name) || name === "" || name.length > 50 || name.length < 1) {
          throw JSON.stringify({
            type: "error",
            message: "CORE_COMMON_PROFILE_NAME_INVALID"
          });
        }
        if (userType === "employee" && !new RegExp(/^[6-9]{1}[0-9]{9}$/).test(mobileNumber)) {
          throw JSON.stringify({
            type: "error",
            message: "CORE_COMMON_PROFILE_MOBILE_NUMBER_INVALID"
          });
        }
        if (email.length && !(email.includes("@") && email.includes("."))) {
          throw JSON.stringify({
            type: "error",
            message: "CORE_COMMON_PROFILE_EMAIL_INVALID"
          });
        }
        if (currentPassword.length || newPassword.length || confirmPassword.length) {
          if (newPassword !== confirmPassword) {
            throw JSON.stringify({
              type: "error",
              message: "CORE_COMMON_PROFILE_PASSWORD_MISMATCH"
            });
          }
          if (!(currentPassword.length && newPassword.length && confirmPassword.length)) {
            throw JSON.stringify({
              type: "error",
              message: "CORE_COMMON_PROFILE_PASSWORD_INVALID"
            });
          }
          if (!new RegExp(/^([a-zA-Z0-9@#$%]{8,15})$/i).test(newPassword) && !new RegExp(/^([a-zA-Z0-9@#$%]{8,15})$/i).test(confirmPassword)) {
            throw JSON.stringify({
              type: "error",
              message: "CORE_COMMON_PROFILE_PASSWORD_INVALID"
            });
          }
        }
        return Promise.resolve(Digit.UserService.updateUser(requestData, stateCode)).then(function (_ref2) {
          var responseInfo = _ref2.responseInfo,
            user = _ref2.user;
          if (responseInfo && responseInfo.status === "200") {
            var _user = Digit.UserService.getUser();
            if (_user) {
              Digit.UserService.setUser(_extends({}, _user, {
                info: _extends({}, _user.info, {
                  name: name,
                  mobileNumber: mobileNumber,
                  emailId: email,
                  permanentCity: city
                })
              }));
            }
          }
          return function () {
            if (currentPassword.length && newPassword.length && confirmPassword.length) {
              var _requestData = {
                existingPassword: currentPassword,
                newPassword: newPassword,
                tenantId: tenant,
                type: "EMPLOYEE",
                username: userInfo === null || userInfo === void 0 ? void 0 : userInfo.userName,
                confirmPassword: confirmPassword
              };
              return function () {
                if (newPassword === confirmPassword) {
                  return _catch(function () {
                    return Promise.resolve(Digit.UserService.changePassword(_requestData, tenant)).then(function (res) {
                      var changePasswordResponseInfo = res.responseInfo;
                      if (changePasswordResponseInfo !== null && changePasswordResponseInfo !== void 0 && changePasswordResponseInfo.status && changePasswordResponseInfo.status === "200") {
                        showToast("success", t("CORE_COMMON_PROFILE_UPDATE_SUCCESS_WITH_PASSWORD"), 5000);
                      } else {
                        throw "";
                      }
                    });
                  }, function (error) {
                    var _error$Errors, _error$Errors$at;
                    throw JSON.stringify({
                      type: "error",
                      message: (_error$Errors = error.Errors) !== null && _error$Errors !== void 0 && (_error$Errors$at = _error$Errors.at(0)) !== null && _error$Errors$at !== void 0 && _error$Errors$at.description ? error.Errors.at(0).description : "CORE_COMMON_PROFILE_UPDATE_ERROR_WITH_PASSWORD"
                    });
                  });
                } else {
                  throw JSON.stringify({
                    type: "error",
                    message: "CORE_COMMON_PROFILE_ERROR_PASSWORD_NOT_MATCH"
                  });
                }
              }();
            } else if (responseInfo !== null && responseInfo !== void 0 && responseInfo.status && responseInfo.status === "200") {
              showToast("success", t("CORE_COMMON_PROFILE_UPDATE_SUCCESS"), 5000);
            }
          }();
        });
      }, function (error) {
        var errorObj = JSON.parse(error);
        showToast(errorObj.type, t(errorObj.message), 5000);
      });
      return Promise.resolve(_temp6 && _temp6.then ? _temp6.then(_temp5) : _temp5(_temp6));
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var menu = [];
  var _Digit$Hooks$pt$useGe = Digit.Hooks.pt.useGenderMDMS(stateId, "common-masters", "GenderType"),
    Menu = _Digit$Hooks$pt$useGe.data;
  Menu && Menu.map(function (genderDetails) {
    menu.push({
      i18nKey: "PT_COMMON_GENDER_" + genderDetails.code,
      code: "" + genderDetails.code,
      value: "" + genderDetails.code
    });
  });
  var setFileStoreId = function setFileStoreId(fileStoreId) {
    try {
      var _temp8 = function _temp8(thumbnails) {
        setProfileImg(thumbnails === null || thumbnails === void 0 ? void 0 : thumbnails.thumbs[0]);
        closeFileUploadDrawer();
      };
      setProfilePic(fileStoreId);
      return Promise.resolve(fileStoreId ? Promise.resolve(getThumbnails([fileStoreId], stateId)).then(_temp8) : _temp8(null));
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var getThumbnails = function getThumbnails(ids, tenantId) {
    try {
      return Promise.resolve(Digit.UploadServices.Filefetch(ids, tenantId)).then(function (res) {
        if (res.data.fileStoreIds && res.data.fileStoreIds.length !== 0) {
          return {
            thumbs: res.data.fileStoreIds.map(function (o) {
              return o.url.split(",")[3];
            }),
            images: res.data.fileStoreIds.map(function (o) {
              return Digit.Utils.getFileUrl(o.url);
            })
          };
        } else {
          return null;
        }
      });
    } catch (e) {
      return Promise.reject(e);
    }
  };
  if (loading) return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null);
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("section", {
    style: {
      margin: userType === "citizen" ? "8px" : "24px"
    }
  }, userType === "citizen" ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, null) : /*#__PURE__*/React__default.createElement(digitUiReactComponents.BreadCrumb, {
    crumbs: [{
      path: "/digit-ui/employee",
      content: t("ES_COMMON_HOME"),
      show: true
    }, {
      path: "/digit-ui/employee/user/profile",
      content: t("ES_COMMON_PAGE_1"),
      show: url.includes("/user/profile")
    }]
  })), /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: "flex",
      flex: 1,
      flexDirection: windowWidth < 768 || userType === "citizen" ? "column" : "row",
      margin: userType === "citizen" ? "8px" : "16px",
      gap: userType === "citizen" ? "" : "0 24px",
      boxShadow: userType === "citizen" ? "1px 1px 4px 0px rgba(0,0,0,0.2)" : "",
      background: userType === "citizen" ? "white" : "",
      borderRadius: userType === "citizen" ? "4px" : ""
    }
  }, /*#__PURE__*/React__default.createElement("section", {
    style: {
      position: "relative",
      display: "flex",
      flex: userType === "citizen" ? 1 : 2.5,
      justifyContent: "center",
      alignItems: "center",
      maxWidth: "100%",
      height: "376px",
      borderRadius: "4px",
      boxShadow: userType === "citizen" ? "" : "1px 1px 4px 0px rgba(0,0,0,0.2)",
      border: (userType === "citizen" ? "8px" : "24px") + " solid #fff",
      background: "#EEEEEE",
      padding: userType === "citizen" ? "8px" : "16px"
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      position: "relative",
      height: userType === "citizen" ? "114px" : "150px",
      width: userType === "citizen" ? "114px" : "150px",
      margin: "16px"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      margin: "auto",
      borderRadius: "300px",
      justifyContent: "center",
      height: "100%",
      width: "100%"
    },
    src: !profileImg || profileImg === "" ? defaultImage$1 : profileImg
  }), /*#__PURE__*/React__default.createElement("button", {
    style: {
      position: "absolute",
      left: "50%",
      bottom: "-24px",
      transform: "translateX(-50%)"
    },
    onClick: onClickAddPic
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CameraIcon, null)))), /*#__PURE__*/React__default.createElement("section", {
    style: {
      display: "flex",
      flexDirection: "column",
      flex: userType === "citizen" ? 1 : 7.5,
      width: "100%",
      borderRadius: "4px",
      height: "fit-content",
      boxShadow: userType === "citizen" ? "" : "1px 1px 4px 0px rgba(0,0,0,0.2)",
      background: "white",
      padding: userType === "citizen" ? "8px" : "24px",
      paddingBottom: "20px"
    }
  }, userType === "citizen" ? /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    style:  {}
  }, "" + t("CORE_COMMON_PROFILE_NAME"), "*"), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    t: t,
    style: {
      width: "100%"
    },
    type: "text",
    isMandatory: false,
    name: "name",
    value: name,
    onChange: function onChange(e) {
      return setUserName(e.target.value);
    }
  }, validation = {
    isRequired: true,
    pattern: "^[a-zA-Z-.`' ]*$",
    type: "tel",
    title: t("CORE_COMMON_PROFILE_NAME_ERROR_MESSAGE")
  }, {
    disable: editScreen
  })), (errors === null || errors === void 0 ? void 0 : errors.userName) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, " ", t(errors === null || errors === void 0 ? void 0 : (_errors$userName = errors.userName) === null || _errors$userName === void 0 ? void 0 : _errors$userName.message), " "))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    style:  {}
  }, "" + t("CORE_COMMON_PROFILE_GENDER")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    style: {
      width: "100%"
    },
    className: "form-field",
    selected: (gender === null || gender === void 0 ? void 0 : gender.length) === 1 ? gender[0] : gender,
    disable: (gender === null || gender === void 0 ? void 0 : gender.length) === 1 || editScreen,
    option: menu,
    select: setGenderName,
    value: gender,
    optionKey: "code",
    t: t,
    name: "gender"
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, null), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    style:  {}
  }, "" + t("CORE_COMMON_PROFILE_EMAIL")), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    style: {
      width: "100%"
    },
    type: "email",
    isMandatory: false,
    optionKey: "i18nKey",
    name: "email",
    value: email,
    onChange: function onChange(e) {
      return setUserEmailAddress(e.target.value);
    },
    disable: editScreen
  }), (errors === null || errors === void 0 ? void 0 : errors.emailAddress) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, " ", t(errors === null || errors === void 0 ? void 0 : (_errors$emailAddress = errors.emailAddress) === null || _errors$emailAddress === void 0 ? void 0 : _errors$emailAddress.message), " "))), /*#__PURE__*/React__default.createElement("button", {
    onClick: updateProfile,
    style: {
      marginTop: "24px",
      backgroundColor: "#F47738",
      width: "100%",
      height: "40px",
      color: "white",
      borderBottom: "1px solid black"
    }
  }, t("CORE_COMMON_SAVE"))) : /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_NAME"), "*"), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    t: t,
    type: "text",
    isMandatory: false,
    name: "name",
    value: name,
    onChange: function onChange(e) {
      return setUserName(e.target.value);
    },
    placeholder: "Enter Your Name"
  }, validation = {
    isRequired: true,
    pattern: "^[a-zA-Z-.`' ]*$",
    type: "text",
    title: t("CORE_COMMON_PROFILE_NAME_ERROR_MESSAGE")
  }, {
    disable: editScreen
  })), (errors === null || errors === void 0 ? void 0 : errors.userName) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, " ", t(errors === null || errors === void 0 ? void 0 : (_errors$userName2 = errors.userName) === null || _errors$userName2 === void 0 ? void 0 : _errors$userName2.message), " "))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_GENDER")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    style: {
      width: "100%"
    },
    selected: (gender === null || gender === void 0 ? void 0 : gender.length) === 1 ? gender[0] : gender,
    disable: (gender === null || gender === void 0 ? void 0 : gender.length) === 1 || editScreen,
    option: menu,
    select: setGenderName,
    value: gender,
    optionKey: "code",
    t: t,
    name: "gender"
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_CITY")), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    t: t,
    type: "text",
    isMandatory: false,
    name: "city",
    value: city,
    onChange: function onChange(e) {
      return setCity(e.target.value);
    },
    placeholder: "Enter Your City Name"
  }, validation = {
    isRequired: true,
    type: "text",
    title: t("CORE_COMMON_PROFILE_CITY_ERROR_MESSAGE")
  }, {
    disable: true
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, null))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style: {
      width: "300px"
    }
  }, t("CORE_COMMON_PROFILE_MOBILE_NUMBER") + "*"), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.MobileNumber, {
    value: mobileNumber,
    style: {
      width: "100%"
    },
    name: "mobileNumber",
    placeholder: "Enter a valid Mobile No.",
    onChange: function onChange(value) {
      return setUserMobileNumber(value);
    },
    disable: true,
    required: true,
    pattern: "[6-9]{1}[0-9]{9}",
    type: "tel",
    title: t("CORE_COMMON_PROFILE_MOBILE_NUMBER_INVALID")
  }), (errors === null || errors === void 0 ? void 0 : errors.mobileNumber) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, " ", t(errors === null || errors === void 0 ? void 0 : (_errors$mobileNumber = errors.mobileNumber) === null || _errors$mobileNumber === void 0 ? void 0 : _errors$mobileNumber.message), " "))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_EMAIL")), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "email",
    isMandatory: false,
    placeholder: "Enter a valid Email",
    optionKey: "i18nKey",
    name: "email",
    value: email,
    onChange: function onChange(e) {
      return setUserEmailAddress(e.target.value);
    },
    disable: editScreen
  }), (errors === null || errors === void 0 ? void 0 : errors.emailAddress) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, " ", t(errors === null || errors === void 0 ? void 0 : (_errors$emailAddress2 = errors.emailAddress) === null || _errors$emailAddress2 === void 0 ? void 0 : _errors$emailAddress2.message), " "))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, null, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("a", {
    style: (_ref3 = {
      color: "orange",
      cursor: "default",
      marginBottom: "5"
    }, _ref3["cursor"] = "pointer", _ref3),
    onClick: TogleforPassword
  }, t("CORE_COMMON_CHANGE_PASSWORD")), changepassword ? /*#__PURE__*/React__default.createElement("div", {
    style: {
      marginTop: "10px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_CURRENT_PASSWORD")), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "password",
    isMandatory: false,
    name: "name",
    pattern: "^([a-zA-Z0-9@#$%])+$",
    onChange: function onChange(e) {
      return setUserCurrentPassword(e.target.value);
    },
    disable: editScreen
  }), (errors === null || errors === void 0 ? void 0 : errors.currentPassword) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, t(errors === null || errors === void 0 ? void 0 : (_errors$currentPasswo = errors.currentPassword) === null || _errors$currentPasswo === void 0 ? void 0 : _errors$currentPasswo.message)))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_NEW_PASSWORD")), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "password",
    isMandatory: false,
    name: "name",
    pattern: "^([a-zA-Z0-9@#$%])+$",
    onChange: function onChange(e) {
      return setUserNewPassword(e.target.value);
    },
    disable: editScreen
  }), (errors === null || errors === void 0 ? void 0 : errors.newPassword) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, t(errors === null || errors === void 0 ? void 0 : (_errors$newPassword = errors.newPassword) === null || _errors$newPassword === void 0 ? void 0 : _errors$newPassword.message)))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LabelFieldPair, {
    style: {
      display: "flex"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    className: "profile-label-margin",
    style:  {
      width: "300px"
    }
  }, "" + t("CORE_COMMON_PROFILE_CONFIRM_PASSWORD")), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: "100%"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "password",
    isMandatory: false,
    name: "name",
    pattern: "^([a-zA-Z0-9@#$%])+$",
    onChange: function onChange(e) {
      return setUserConfirmPassword(e.target.value);
    },
    disable: editScreen
  }), (errors === null || errors === void 0 ? void 0 : errors.confirmPassword) && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, {
    style: {
      width: "70%",
      fontSize: "12px",
      marginTop: "-21px"
    }
  }, t(errors === null || errors === void 0 ? void 0 : (_errors$confirmPasswo = errors.confirmPassword) === null || _errors$confirmPasswo === void 0 ? void 0 : _errors$confirmPasswo.message))))) : ""))))), userType === "employee" ? /*#__PURE__*/React__default.createElement("div", {
    style: {
      height: "88px",
      backgroundColor: "#FFFFFF",
      display: "flex",
      justifyContent: "flex-end",
      marginTop: "64px",
      alignItems: "center"
    }
  }, /*#__PURE__*/React__default.createElement("button", {
    onClick: updateProfile,
    style: {
      marginTop: "24px",
      backgroundColor: "#F47738",
      width: windowWidth < 768 ? "100%" : "248px",
      height: "40px",
      float: "right",
      margin: windowWidth < 768 ? "0 16px" : "",
      marginRight: windowWidth < 768 ? "16px" : "31px",
      color: "white",
      borderBottom: "1px solid black"
    }
  }, t("CORE_COMMON_SAVE"))) : "", toast && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    error: toast.key === "error",
    label: t(toast.key === "success" ? "CORE_COMMON_PROFILE_UPDATE_SUCCESS" : toast.action),
    onClose: function onClose() {
      return setToast(null);
    },
    style: {
      maxWidth: "670px"
    }
  }), openUploadSlide == true ? /*#__PURE__*/React__default.createElement(UploadDrawer, {
    setProfilePic: setFileStoreId,
    closeDrawer: closeFileUploadDrawer,
    userType: userType,
    removeProfilePic: removeProfilePic,
    showToast: showToast
  }) : "");
};

var EmployeeApp = function EmployeeApp(_ref) {
  var _location$pathname, _location$pathname2;
  var stateInfo = _ref.stateInfo,
    userDetails = _ref.userDetails,
    CITIZEN = _ref.CITIZEN,
    cityDetails = _ref.cityDetails,
    mobileView = _ref.mobileView,
    handleUserDropdownSelection = _ref.handleUserDropdownSelection,
    logoUrl = _ref.logoUrl,
    DSO = _ref.DSO,
    stateCode = _ref.stateCode,
    modules = _ref.modules,
    appTenants = _ref.appTenants,
    sourceUrl = _ref.sourceUrl;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var location = reactRouterDom.useLocation();
  var showLanguageChange = location === null || location === void 0 ? void 0 : (_location$pathname = location.pathname) === null || _location$pathname === void 0 ? void 0 : _location$pathname.includes("language-selection");
  var isUserProfile = location === null || location === void 0 ? void 0 : (_location$pathname2 = location.pathname) === null || _location$pathname2 === void 0 ? void 0 : _location$pathname2.includes("user/profile");
  React.useEffect(function () {
    Digit.UserService.setType("employee");
  }, []);
  sourceUrl = "https://s3.ap-south-1.amazonaws.com/egov-qa-assets";
  var pdfUrl = "https://pg-egov-assets.s3.ap-south-1.amazonaws.com/Upyog+Code+and+Copyright+License_v1.pdf";
  return /*#__PURE__*/React__default.createElement("div", {
    className: "employee"
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user"
  }, isUserProfile && /*#__PURE__*/React__default.createElement(TopBarSideBar, {
    t: t,
    stateInfo: stateInfo,
    userDetails: userDetails,
    CITIZEN: CITIZEN,
    cityDetails: cityDetails,
    mobileView: mobileView,
    handleUserDropdownSelection: handleUserDropdownSelection,
    logoUrl: logoUrl,
    showSidebar: isUserProfile ? true : false,
    showLanguageChange: !showLanguageChange
  }), /*#__PURE__*/React__default.createElement("div", {
    className: isUserProfile ? "grounded-container" : "loginContainer",
    style: isUserProfile ? {
      padding: 0,
      paddingTop: "80px",
      marginLeft: mobileView ? "" : "64px"
    } : {
      "--banner-url": "url(" + (stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.bannerUrl) + ")",
      padding: "0px"
    }
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user/login"
  }, /*#__PURE__*/React__default.createElement(EmployeeLogin, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user/forgot-password"
  }, /*#__PURE__*/React__default.createElement(EmployeeForgotPassword, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user/change-password"
  }, /*#__PURE__*/React__default.createElement(EmployeeChangePassword, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user/profile"
  }, /*#__PURE__*/React__default.createElement(UserProfile, {
    stateCode: stateCode,
    userType: "employee",
    cityDetails: cityDetails
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user/language-selection"
  }, /*#__PURE__*/React__default.createElement(LanguageSelection, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
    to: path + "/user/language-selection"
  }))))), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, null, /*#__PURE__*/React__default.createElement(TopBarSideBar, {
    t: t,
    stateInfo: stateInfo,
    userDetails: userDetails,
    CITIZEN: CITIZEN,
    cityDetails: cityDetails,
    mobileView: mobileView,
    handleUserDropdownSelection: handleUserDropdownSelection,
    logoUrl: logoUrl
  }), /*#__PURE__*/React__default.createElement("div", {
    className: "main " + (DSO ? "m-auto" : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "employee-app-wrapper"
  }, /*#__PURE__*/React__default.createElement(ErrorBoundary, null, /*#__PURE__*/React__default.createElement(AppModules, {
    stateCode: stateCode,
    userType: "employee",
    modules: modules,
    appTenants: appTenants
  }))), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: '100%',
      bottom: 0
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center',
      color: "#22394d"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      cursor: "pointer",
      display: "inline-flex",
      height: '1.4em'
    },
    alt: "Powered by DIGIT",
    src: sourceUrl + "/digit-footer.png",
    onError: "this.src='./../digit-footer.png'",
    onClick: function onClick() {
      window.open('https://www.digit.org/', '_blank').focus();
    }
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    onClick: function onClick() {
      window.open('https://niua.in/', '_blank').focus();
    }
  }, "Copyright \xA9 2022 National Institute of Urban Affairs"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("a", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    href: pdfUrl,
    target: "_blank"
  }, "UPYOG License"))))), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
    to: path + "/user/language-selection"
  }))));
};

var Home = function Home() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var tenantId = Digit.ULBService.getCitizenCurrentTenant(true);
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    _Digit$Hooks$useStore2 = _Digit$Hooks$useStore.data;
  _Digit$Hooks$useStore2 = _Digit$Hooks$useStore2 === void 0 ? {} : _Digit$Hooks$useStore2;
  var stateInfo = _Digit$Hooks$useStore2.stateInfo,
    isLoading = _Digit$Hooks$useStore.isLoading;
  var conditionsToDisableNotificationCountTrigger = function conditionsToDisableNotificationCountTrigger() {
    var _Digit$UserService, _Digit$UserService$ge, _Digit$UserService$ge2, _Digit$UserService2, _Digit$UserService2$g;
    if (((_Digit$UserService = Digit.UserService) === null || _Digit$UserService === void 0 ? void 0 : (_Digit$UserService$ge = _Digit$UserService.getUser()) === null || _Digit$UserService$ge === void 0 ? void 0 : (_Digit$UserService$ge2 = _Digit$UserService$ge.info) === null || _Digit$UserService$ge2 === void 0 ? void 0 : _Digit$UserService$ge2.type) === "EMPLOYEE") return false;
    if (!((_Digit$UserService2 = Digit.UserService) !== null && _Digit$UserService2 !== void 0 && (_Digit$UserService2$g = _Digit$UserService2.getUser()) !== null && _Digit$UserService2$g !== void 0 && _Digit$UserService2$g.access_token)) return false;
    return true;
  };
  var _Digit$Hooks$useEvent = Digit.Hooks.useEvents({
      tenantId: tenantId,
      variant: "whats-new",
      config: {
        enabled: conditionsToDisableNotificationCountTrigger()
      }
    }),
    EventsData = _Digit$Hooks$useEvent.data,
    EventsDataLoading = _Digit$Hooks$useEvent.isLoading;
  if (!tenantId) {
    history.push("/digit-ui/citizen/select-language");
  }
  var allCitizenServicesProps = {
    header: t("DASHBOARD_CITIZEN_SERVICES_LABEL"),
    sideOption: {
      name: t("DASHBOARD_VIEW_ALL_LABEL"),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/all-services");
      }
    },
    options: [{
      name: t("ES_PGR_HEADER_COMPLAINT"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.ComplaintIcon, null),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/pgr-home");
      }
    }, {
      name: t("MODULE_PT"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.PTIcon, {
        className: "fill-path-primary-main"
      }),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/pt-home");
      }
    }, {
      name: t("MODULE_TL"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.CaseIcon, {
        className: "fill-path-primary-main"
      }),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/tl-home");
      }
    },
    {
      name: t("CS_COMMON_INBOX_BPA"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.OBPSIcon, null),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/obps-home");
      }
    }],
    styles: {
      display: "flex",
      flexWrap: "wrap",
      justifyContent: "flex-start",
      width: "100%"
    }
  };
  var allInfoAndUpdatesProps = {
    header: t("CS_COMMON_DASHBOARD_INFO_UPDATES"),
    sideOption: {
      name: t("DASHBOARD_VIEW_ALL_LABEL"),
      onClick: function onClick() {}
    },
    options: [{
      name: t("CS_HEADER_MYCITY"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.HomeIcon, null)
    }, {
      name: t("EVENTS_EVENTS_HEADER"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.Calender, null),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/engagement/events");
      }
    }, {
      name: t("CS_COMMON_DOCUMENTS"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.DocumentIcon, null),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/engagement/docs");
      }
    }, {
      name: t("CS_COMMON_SURVEYS"),
      Icon: /*#__PURE__*/React__default.createElement(digitUiReactComponents.DocumentIcon, null),
      onClick: function onClick() {
        return history.push("/digit-ui/citizen/engagement/surveys/list");
      }
    }
    ],

    styles: {
      display: "flex",
      flexWrap: "wrap",
      justifyContent: "flex-start",
      width: "100%"
    }
  };
  return isLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null) : /*#__PURE__*/React__default.createElement("div", {
    className: "HomePageWrapper"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "BannerWithSearch"
  }, /*#__PURE__*/React__default.createElement("img", {
    src: stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.bannerUrl
  }), /*#__PURE__*/React__default.createElement("div", {
    className: "Search"
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.StandaloneSearchBar, {
    placeholder: t("CS_COMMON_SEARCH_PLACEHOLDER")
  }))), /*#__PURE__*/React__default.createElement("div", {
    className: "ServicesSection"
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardBasedOptions, allCitizenServicesProps), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardBasedOptions, allInfoAndUpdatesProps)), conditionsToDisableNotificationCountTrigger() ? EventsDataLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null) : /*#__PURE__*/React__default.createElement("div", {
    className: "WhatsNewSection"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "headSection"
  }, /*#__PURE__*/React__default.createElement("h2", null, t("DASHBOARD_WHATS_NEW_LABEL")), /*#__PURE__*/React__default.createElement("p", {
    onClick: function onClick() {
      return history.push("/digit-ui/citizen/engagement/whats-new");
    }
  }, t("DASHBOARD_VIEW_ALL_LABEL"))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.WhatsNewCard, EventsData === null || EventsData === void 0 ? void 0 : EventsData[0])) : null);
};

var LanguageSelection$1 = function LanguageSelection() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    _Digit$Hooks$useStore2 = _Digit$Hooks$useStore.data;
  _Digit$Hooks$useStore2 = _Digit$Hooks$useStore2 === void 0 ? {} : _Digit$Hooks$useStore2;
  var languages = _Digit$Hooks$useStore2.languages,
    stateInfo = _Digit$Hooks$useStore2.stateInfo,
    isLoading = _Digit$Hooks$useStore.isLoading;
  var selectedLanguage = Digit.StoreData.getCurrentLanguage();
  var texts = React.useMemo(function () {
    return {
      header: t("CS_COMMON_CHOOSE_LANGUAGE"),
      submitBarLabel: t("CONTINUE")
    };
  }, [t]);
  var RadioButtonProps = React.useMemo(function () {
    return {
      options: languages,
      optionsKey: "label",
      additionalWrapperClass: "reverse-radio-selection-wrapper",
      onSelect: function onSelect(language) {
        return Digit.LocalizationService.changeLanguage(language.value, stateInfo.code);
      },
      selectedOption: languages === null || languages === void 0 ? void 0 : languages.filter(function (i) {
        return i.value === selectedLanguage;
      })[0]
    };
  }, [selectedLanguage, languages]);
  function onSubmit() {
    history.push("/digit-ui/citizen/select-location");
  }
  return isLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null) : /*#__PURE__*/React__default.createElement(digitUiReactComponents.PageBasedInput, {
    texts: texts,
    onSubmit: onSubmit
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("CHOOSE YOUR LANGUAGE")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioButtons, RadioButtonProps));
};

var LocationSelection = function LocationSelection() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var _Digit$Hooks$useTenan = Digit.Hooks.useTenants(),
    cities = _Digit$Hooks$useTenan.data,
    isLoading = _Digit$Hooks$useTenan.isLoading;
  var _useState = React.useState(function () {
      return {
        code: Digit.ULBService.getCitizenCurrentTenant(true)
      };
    }),
    selectedCity = _useState[0],
    setSelectedCity = _useState[1];
  var _useState2 = React.useState(false),
    showError = _useState2[0],
    setShowError = _useState2[1];
  var texts = React.useMemo(function () {
    return {
      header: t("CS_COMMON_CHOOSE_LOCATION"),
      submitBarLabel: t("CONTINUE")
    };
  }, [t]);
  function selectCity(city) {
    setSelectedCity(city);
    setShowError(false);
  }
  var RadioButtonProps = React.useMemo(function () {
    return {
      options: cities,
      optionsKey: "i18nKey",
      additionalWrapperClass: "reverse-radio-selection-wrapper",
      onSelect: selectCity,
      selectedOption: selectedCity
    };
  }, [cities, t, selectedCity]);
  function onSubmit() {
    if (selectedCity) {
      Digit.SessionStorage.set("CITIZEN.COMMON.HOME.CITY", selectedCity);
      history.push("/digit-ui/citizen");
    } else {
      setShowError(true);
    }
  }
  return isLoading ? /*#__PURE__*/React__default.createElement("loader", null) : /*#__PURE__*/React__default.createElement(React.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, null), /*#__PURE__*/React__default.createElement(digitUiReactComponents.PageBasedInput, {
    texts: texts,
    onSubmit: onSubmit
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("CHOOSE LOCATION")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.SearchOnRadioButtons, _extends({}, RadioButtonProps, {
    placeholder: t("SEARCH")
  })), showError ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabelError, null, t("LOCATION ERRROR")) : null));
};

var getTenants$1 = function getTenants(codes, tenants) {
  return tenants.filter(function (tenant) {
    return codes.map(function (item) {
      return item.code;
    }).includes(tenant.code);
  });
};
var Home$1 = function Home$1(_ref) {
  var stateInfo = _ref.stateInfo,
    userDetails = _ref.userDetails,
    CITIZEN = _ref.CITIZEN,
    cityDetails = _ref.cityDetails,
    mobileView = _ref.mobileView,
    handleUserDropdownSelection = _ref.handleUserDropdownSelection,
    logoUrl = _ref.logoUrl,
    stateCode = _ref.stateCode,
    modules = _ref.modules,
    appTenants = _ref.appTenants,
    sourceUrl = _ref.sourceUrl,
    pathname = _ref.pathname;
  var classname = Digit.Hooks.fsm.useRouteSubscription(pathname);
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  sourceUrl = "https://s3.ap-south-1.amazonaws.com/egov-qa-assets";
  var pdfUrl = "https://pg-egov-assets.s3.ap-south-1.amazonaws.com/Upyog+Code+and+Copyright+License_v1.pdf";
  var appRoutes = modules.map(function (_ref2, index) {
    var code = _ref2.code,
      tenants = _ref2.tenants;
    var Module = Digit.ComponentRegistryService.getComponent(code + "Module");
    return /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
      key: index,
      path: path + "/" + code.toLowerCase()
    }, /*#__PURE__*/React__default.createElement(Module, {
      stateCode: stateCode,
      moduleCode: code,
      userType: "citizen",
      tenants: getTenants$1(tenants, appTenants)
    }));
  });
  var ModuleLevelLinkHomePages = modules.map(function (_ref3, index) {
    var code = _ref3.code,
      bannerImage = _ref3.bannerImage;
    var Links = Digit.ComponentRegistryService.getComponent(code + "Links") || function () {
      return /*#__PURE__*/React__default.createElement(React__default.Fragment, null);
    };
    return /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
      key: index,
      path: path + "/" + code.toLowerCase() + "-home"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "moduleLinkHomePage"
    }, /*#__PURE__*/React__default.createElement("img", {
      src: bannerImage || (stateInfo === null || stateInfo === void 0 ? void 0 : stateInfo.bannerUrl),
      alt: "noimagefound"
    }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, {
      className: "moduleLinkHomePageBackButton"
    }), /*#__PURE__*/React__default.createElement("h1", null, t("MODULE_" + code.toUpperCase()))), /*#__PURE__*/React__default.createElement("div", {
      className: "moduleLinkHomePageModuleLinks"
    }, /*#__PURE__*/React__default.createElement(Links, {
      key: index,
      matchPath: "/digit-ui/citizen/" + code.toLowerCase(),
      userType: "citizen"
    })));
  });
  return /*#__PURE__*/React__default.createElement("div", {
    className: classname
  }, /*#__PURE__*/React__default.createElement(TopBarSideBar, {
    t: t,
    stateInfo: stateInfo,
    userDetails: userDetails,
    CITIZEN: CITIZEN,
    cityDetails: cityDetails,
    mobileView: mobileView,
    handleUserDropdownSelection: handleUserDropdownSelection,
    logoUrl: logoUrl,
    showSidebar: true
  }), /*#__PURE__*/React__default.createElement("div", {
    className: "main center-container mb-25"
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    exact: true,
    path: path
  }, /*#__PURE__*/React__default.createElement(Home, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    exact: true,
    path: path + "/select-language"
  }, /*#__PURE__*/React__default.createElement(LanguageSelection$1, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    exact: true,
    path: path + "/select-location"
  }, /*#__PURE__*/React__default.createElement(LocationSelection, null)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/all-services"
  }, /*#__PURE__*/React__default.createElement(AppHome, {
    userType: "citizen",
    modules: modules
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/login"
  }, /*#__PURE__*/React__default.createElement(Login, {
    stateCode: stateCode
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/register"
  }, /*#__PURE__*/React__default.createElement(Login, {
    stateCode: stateCode,
    isUserRegistered: false
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/user/profile"
  }, /*#__PURE__*/React__default.createElement(UserProfile, {
    stateCode: stateCode,
    userType: "citizen",
    cityDetails: cityDetails
  })), /*#__PURE__*/React__default.createElement(ErrorBoundary, null, appRoutes, ModuleLevelLinkHomePages))), /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: '100%',
      bottom: 0
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center',
      color: "#22394d"
    }
  }, /*#__PURE__*/React__default.createElement("img", {
    style: {
      cursor: "pointer",
      display: "inline-flex",
      height: '1.4em'
    },
    alt: "Powered by DIGIT",
    src: sourceUrl + "/digit-footer.png",
    onError: "this.src='./../digit-footer.png'",
    onClick: function onClick() {
      window.open('https://www.digit.org/', '_blank').focus();
    }
  }), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    onClick: function onClick() {
      window.open('https://niua.in/', '_blank').focus();
    }
  }, "Copyright \xA9 2022 National Institute of Urban Affairs"), /*#__PURE__*/React__default.createElement("span", {
    style: {
      margin: "0 10px"
    }
  }, "|"), /*#__PURE__*/React__default.createElement("a", {
    style: {
      cursor: "pointer",
      fontSize: "16px",
      fontWeight: "400"
    },
    href: pdfUrl,
    target: "_blank"
  }, "UPYOG License"))));
};

var DigitApp = function DigitApp(_ref) {
  var _userDetails$info;
  var stateCode = _ref.stateCode,
    modules = _ref.modules,
    appTenants = _ref.appTenants,
    logoUrl = _ref.logoUrl;
  var history = reactRouterDom.useHistory();
  var _useLocation = reactRouterDom.useLocation(),
    pathname = _useLocation.pathname;
  var innerWidth = window.innerWidth;
  var cityDetails = Digit.ULBService.getCurrentUlb();
  var userDetails = Digit.UserService.getUser();
  var _Digit$Hooks$useStore = Digit.Hooks.useStore.getInitData(),
    storeData = _Digit$Hooks$useStore.data;
  var _ref2 = storeData || {},
    stateInfo = _ref2.stateInfo;
  var DSO = Digit.UserService.hasAccess(["FSM_DSO"]);
  var CITIZEN = (userDetails === null || userDetails === void 0 ? void 0 : (_userDetails$info = userDetails.info) === null || _userDetails$info === void 0 ? void 0 : _userDetails$info.type) === "CITIZEN" || !window.location.pathname.split("/").includes("employee") ? true : false;
  if (window.location.pathname.split("/").includes("employee")) CITIZEN = false;
  React.useEffect(function () {
    if (!(pathname !== null && pathname !== void 0 && pathname.includes("application-details"))) {
      if (!(pathname !== null && pathname !== void 0 && pathname.includes("inbox"))) {
        Digit.SessionStorage.del("fsm/inbox/searchParams");
      }
      if (pathname !== null && pathname !== void 0 && pathname.includes("search")) {
        Digit.SessionStorage.del("fsm/search/searchParams");
      }
    }
    if (!(pathname !== null && pathname !== void 0 && pathname.includes("dss"))) {
      Digit.SessionStorage.del("DSS_FILTERS");
    }
  }, [pathname]);
  history.listen(function () {
    var _window;
    (_window = window) === null || _window === void 0 ? void 0 : _window.scrollTo({
      top: 0,
      left: 0,
      behavior: "smooth"
    });
  });
  var handleUserDropdownSelection = function handleUserDropdownSelection(option) {
    option.func();
  };
  var mobileView = innerWidth <= 640;
  var sourceUrl = window.location.origin + "/citizen";
  var commonProps = {
    stateInfo: stateInfo,
    userDetails: userDetails,
    CITIZEN: CITIZEN,
    cityDetails: cityDetails,
    mobileView: mobileView,
    handleUserDropdownSelection: handleUserDropdownSelection,
    logoUrl: logoUrl,
    DSO: DSO,
    stateCode: stateCode,
    modules: modules,
    appTenants: appTenants,
    sourceUrl: sourceUrl,
    pathname: pathname
  };
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "/digit-ui/employee"
  }, /*#__PURE__*/React__default.createElement(EmployeeApp, commonProps)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "/digit-ui/citizen"
  }, /*#__PURE__*/React__default.createElement(Home$1, commonProps)), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
    to: "/digit-ui/citizen"
  })));
};

var commonReducer = function commonReducer(defaultData) {
  return function (state, action) {
    if (state === void 0) {
      state = defaultData;
    }
    switch (action.type) {
      case "LANGUAGE_SELECT":
        return _extends({}, state, {
          selectedLanguage: action.payload
        });
      default:
        return state;
    }
  };
};

var getRootReducer = function getRootReducer(defaultStore, moduleReducers) {
  return redux.combineReducers(_extends({
    common: commonReducer(defaultStore)
  }, moduleReducers));
};
var middleware = [thunk];
var composeEnhancers = typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({}) : redux.compose;
var enhancer = composeEnhancers(redux.applyMiddleware.apply(void 0, middleware)
);

var getStore = function getStore(defaultStore, moduleReducers) {
  if (moduleReducers === void 0) {
    moduleReducers = {};
  }
  return redux.createStore(getRootReducer(defaultStore, moduleReducers), enhancer);
};

var DigitUIWrapper = function DigitUIWrapper(_ref) {
  var _initData$stateInfo;
  var stateCode = _ref.stateCode,
    enabledModules = _ref.enabledModules,
    moduleReducers = _ref.moduleReducers;
  var _Digit$Hooks$useInitS = Digit.Hooks.useInitStore(stateCode, enabledModules),
    isLoading = _Digit$Hooks$useInitS.isLoading,
    initData = _Digit$Hooks$useInitS.data;
  if (isLoading) {
    return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, {
      page: true
    });
  }
  var i18n = reactI18next.getI18n();
  return /*#__PURE__*/React__default.createElement(reactRedux.Provider, {
    store: getStore(initData, moduleReducers(initData))
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.BrowserRouter, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Body, null, /*#__PURE__*/React__default.createElement(DigitApp, {
    initData: initData,
    stateCode: stateCode,
    modules: initData === null || initData === void 0 ? void 0 : initData.modules,
    appTenants: initData.tenants,
    logoUrl: initData === null || initData === void 0 ? void 0 : (_initData$stateInfo = initData.stateInfo) === null || _initData$stateInfo === void 0 ? void 0 : _initData$stateInfo.logoUrl
  }))));
};
var DigitUI = function DigitUI(_ref2) {
  var stateCode = _ref2.stateCode,
    registry = _ref2.registry,
    enabledModules = _ref2.enabledModules,
    moduleReducers = _ref2.moduleReducers;
  var userType = Digit.UserService.getType();
  var queryClient = new reactQuery.QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 15 * 60 * 1000,
        cacheTime: 50 * 60 * 1000,
        retryDelay: function retryDelay(attemptIndex) {
          return Infinity;
        }
      }
    }
  });

  var ComponentProvider = Digit.Contexts.ComponentProvider;
  var DSO = Digit.UserService.hasAccess(["FSM_DSO"]);
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(ErrorBoundary, null, /*#__PURE__*/React__default.createElement(reactQuery.QueryClientProvider, {
    client: queryClient
  }, /*#__PURE__*/React__default.createElement(ComponentProvider.Provider, {
    value: registry
  }, /*#__PURE__*/React__default.createElement(DigitUIWrapper, {
    stateCode: stateCode,
    enabledModules: enabledModules,
    moduleReducers: moduleReducers
  })))));
};
var componentsToRegister = {
  SelectOtp: SelectOtp
};
var initCoreComponents = function initCoreComponents() {
  Object.entries(componentsToRegister).forEach(function (_ref3) {
    var key = _ref3[0],
      value = _ref3[1];
    Digit.ComponentRegistryService.setComponent(key, value);
  });
};

exports.DigitUI = DigitUI;
exports.initCoreComponents = initCoreComponents;
