// @egovernments/digit-ui-module-ws/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var reactI18next = require('react-i18next');
var reactRouterDom = require('react-router-dom');
var digitUiReactComponents = require('@egovernments/digit-ui-react-components');
var reactQuery = require('react-query');

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}

var newConfig = [{
  head: "ES_NEW_APPLICATION_PROPERTY_ASSESSMENT",
  body: [{
    route: "docs-required",
    component: "WSDocsRequired",
    key: "data",
    nextStep: "connection-holder"
  }, {
    route: "connection-holder",
    component: "ConnectionHolder",
    key: "ConnectionHolderDetails",
    nextStep: "service-name",
    texts: {
      headerCaption: "WS_COMMON_CONNECTION_DETAIL",
      header: "WS_COMMON_CONNECTION_HOLDER_DETAILS_HEADER",
      cardText: "WS_CONNECTION_HOLDER_TEXT",
      submitBarLabel: "CS_COMMON_NEXT"
    }
  }, {
    route: "service-name",
    component: "WSServiceName",
    key: "serviceName",
    nextStep: "water-connection-details",
    WATER: "water-connection-details",
    SEWERAGE: "sewerage-connection-details",
    texts: {
      headerCaption: "WS_COMMON_CONNECTION_DETAIL",
      header: "WS_SERVICE_NAME",
      cardText: "",
      submitBarLabel: "CS_COMMON_NEXT"
    }
  }, {
    route: "water-connection-details",
    component: "WSWaterConnectionDetails",
    key: "waterConectionDetails",
    nextStep: "sewerage-connection-details",
    WATER: "plumber-preference",
    texts: {
      headerCaption: "WS_COMMON_CONNECTION_DETAIL",
      header: "WS_WATER_CONNECTION_DETAILS",
      cardText: "WS_PROVIDE_PROPOSED_NO_OF_TAPS_SELECT_PIPE_SIZE",
      submitBarLabel: "CS_COMMON_NEXT"
    }
  }, {
    route: "sewerage-connection-details",
    component: "WSSewerageConnectionDetails",
    key: "sewerageConnectionDetails",
    nextStep: "plumber-preference",
    SEWERAGE: "plumber-preference",
    texts: {
      headerCaption: "WS_COMMON_CONNECTION_DETAIL",
      header: "PDF_STATIC_LABEL_SW_CONSOLIDATED_ACKNOWELDGMENT_LOGO_SUB_HEADER",
      cardText: "WS_PROVIDE_NO_WATER_CLOSETS_TOILETS",
      submitBarLabel: "CS_COMMON_NEXT"
    }
  }, {
    route: "plumber-preference",
    component: "WSPlumberPreference",
    key: "plumberPreference",
    nextStep: "document-details",
    texts: {
      headerCaption: "WS_COMMON_CONNECTION_DETAIL",
      header: "WS_PLUMBER_PREFERENCE",
      cardText: "",
      submitBarLabel: "CS_COMMON_NEXT"
    }
  }, {
    route: "document-details",
    component: "WSDocumentDetails",
    key: "documents",
    nextStep: null,
    texts: {
      headerCaption: "CE_DOCUMENT_DETAILS",
      header: "WS_DOCUMENT_DETAILS_HEADER",
      submitBarLabel: "CS_COMMON_NEXT"
    }
  }]
}];

var getPath = function getPath(path, params) {
  params && Object.keys(params).map(function (key) {
    path = path.replace(":" + key, params[key]);
  });
  return path;
};
var WSCreate = function WSCreate() {
  var _Digit, _Digit$ComponentRegis;
  var queryClient = reactQuery.useQueryClient();
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch();
  var _useLocation = reactRouterDom.useLocation(),
    pathname = _useLocation.pathname,
    state = _useLocation.state;
  var match = reactRouterDom.useRouteMatch();
  var history = reactRouterDom.useHistory();
  var location = reactRouterDom.useLocation();
  var tenantId = Digit.ULBService.getCurrentTenantId();
  var _Digit$Hooks$useSessi = Digit.Hooks.useSessionStorage("WS_CREATE", state !== null && state !== void 0 && state.edcrNumber ? {
      data: {
        scrutinyNumber: {
          edcrNumber: state === null || state === void 0 ? void 0 : state.edcrNumber
        }
      }
    } : {}),
    params = _Digit$Hooks$useSessi[0],
    setParams = _Digit$Hooks$useSessi[1];
  var CheckPage = (_Digit = Digit) === null || _Digit === void 0 ? void 0 : (_Digit$ComponentRegis = _Digit.ComponentRegistryService) === null || _Digit$ComponentRegis === void 0 ? void 0 : _Digit$ComponentRegis.getComponent('WSCheckPage');
  var stateId = Digit.ULBService.getStateId();
  var _Digit$Hooks$obps$Sea = Digit.Hooks.obps.SearchMdmsTypes.getFormConfig(stateId, []),
    newConfig$1 = _Digit$Hooks$obps$Sea.data;
  var goNext = function goNext(skipStep) {
    var currentPath = pathname.split("/").pop();
    var _config$find = config.find(function (routeObj) {
        return routeObj.route === currentPath;
      }),
      nextStep = _config$find.nextStep;
    var routeObject = config.find(function (routeObj) {
      return routeObj.route === currentPath && routeObj;
    });
    if (routeObject[sessionStorage.getItem("serviceName")]) nextStep = routeObject[sessionStorage.getItem("serviceName")];
    var redirectWithHistory = history.push;
    if (nextStep === null) {
      return redirectWithHistory(getPath(match.path, match.params) + "/check");
    }
    redirectWithHistory(getPath(match.path, match.params) + "/" + nextStep);
  };
  var createApplication = function createApplication() {
    try {
      history.push(getPath(match.path, match.params) + "/acknowledgement");
      return Promise.resolve();
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var handleSelect = function handleSelect(key, data, skipStep, isFromCreateApi) {
    var _extends2;
    if (isFromCreateApi) setParams(data);else if (key === "") setParams(_extends({}, data));else setParams(_extends({}, params, (_extends2 = {}, _extends2[key] = _extends({}, params[key], data), _extends2)));
    goNext();
  };
  var handleSkip = function handleSkip() {};
  var config = [];
  newConfig$1 = newConfig;
  newConfig$1.forEach(function (obj) {
    config = config.concat(obj.body.filter(function (a) {
      return !a.hideInCitizen;
    }));
  });
  config.indexRoute = "docs-required";

  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, config.map(function (routeObj, index) {
    var component = routeObj.component,
      texts = routeObj.texts,
      inputs = routeObj.inputs,
      key = routeObj.key;
    var Component = typeof component === "string" ? Digit.ComponentRegistryService.getComponent(component) : component;
    return /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
      path: getPath(match.path, match.params) + "/" + routeObj.route,
      key: index
    }, /*#__PURE__*/React__default.createElement(Component, {
      config: {
        texts: texts,
        inputs: inputs,
        key: key
      },
      onSelect: handleSelect,
      onSkip: handleSkip,
      t: t,
      formData: params
    }));
  }), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: getPath(match.path, match.params) + "/check"
  }, /*#__PURE__*/React__default.createElement(CheckPage, {
    onSubmit: createApplication,
    value: params
  })), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
    to: getPath(match.path, match.params) + "/" + config.indexRoute
  })));
};

var config = [{
  texts: {
    header: "WS_SEARCH_CONNECTION",
    submitBarLabel: "WS_SEARCH_LABEL",
    cardText: "WS_SEARCH_TEXT"
  },
  inputs: [{
    label: "CORE_COMMON_MOBILE_NUMBER",
    type: "mobileNumber",
    name: "mobileNumber",
    error: "CORE_COMMON_PHONENO_INVALIDMSG"
  }, {
    label: "WS_PROP_DETAIL_CITY",
    name: "city",
    error: "WS_WRONG_CITY"
  }, {
    label: "WS_PROP_DETAIL_LOCALITY_LABEL",
    type: "any",
    name: "locality",
    error: "UC_INVALID_LOCALITY"
  }]
}];

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var b = "function" === typeof Symbol && Symbol.for,
  c = b ? Symbol.for("react.element") : 60103,
  d = b ? Symbol.for("react.portal") : 60106,
  e = b ? Symbol.for("react.fragment") : 60107,
  f = b ? Symbol.for("react.strict_mode") : 60108,
  g = b ? Symbol.for("react.profiler") : 60114,
  h = b ? Symbol.for("react.provider") : 60109,
  k = b ? Symbol.for("react.context") : 60110,
  l = b ? Symbol.for("react.async_mode") : 60111,
  m = b ? Symbol.for("react.concurrent_mode") : 60111,
  n = b ? Symbol.for("react.forward_ref") : 60112,
  p = b ? Symbol.for("react.suspense") : 60113,
  q = b ? Symbol.for("react.suspense_list") : 60120,
  r = b ? Symbol.for("react.memo") : 60115,
  t = b ? Symbol.for("react.lazy") : 60116,
  v = b ? Symbol.for("react.block") : 60121,
  w = b ? Symbol.for("react.fundamental") : 60117,
  x = b ? Symbol.for("react.responder") : 60118,
  y = b ? Symbol.for("react.scope") : 60119;
function z(a) {
  if ("object" === typeof a && null !== a) {
    var u = a.$$typeof;
    switch (u) {
      case c:
        switch (a = a.type, a) {
          case l:
          case m:
          case e:
          case g:
          case f:
          case p:
            return a;
          default:
            switch (a = a && a.$$typeof, a) {
              case k:
              case n:
              case t:
              case r:
              case h:
                return a;
              default:
                return u;
            }
        }
      case d:
        return u;
    }
  }
}
function A(a) {
  return z(a) === m;
}
var AsyncMode = l;
var ConcurrentMode = m;
var ContextConsumer = k;
var ContextProvider = h;
var Element = c;
var ForwardRef = n;
var Fragment = e;
var Lazy = t;
var Memo = r;
var Portal = d;
var Profiler = g;
var StrictMode = f;
var Suspense = p;
var isAsyncMode = function isAsyncMode(a) {
  return A(a) || z(a) === l;
};
var isConcurrentMode = A;
var isContextConsumer = function isContextConsumer(a) {
  return z(a) === k;
};
var isContextProvider = function isContextProvider(a) {
  return z(a) === h;
};
var isElement = function isElement(a) {
  return "object" === typeof a && null !== a && a.$$typeof === c;
};
var isForwardRef = function isForwardRef(a) {
  return z(a) === n;
};
var isFragment = function isFragment(a) {
  return z(a) === e;
};
var isLazy = function isLazy(a) {
  return z(a) === t;
};
var isMemo = function isMemo(a) {
  return z(a) === r;
};
var isPortal = function isPortal(a) {
  return z(a) === d;
};
var isProfiler = function isProfiler(a) {
  return z(a) === g;
};
var isStrictMode = function isStrictMode(a) {
  return z(a) === f;
};
var isSuspense = function isSuspense(a) {
  return z(a) === p;
};
var isValidElementType = function isValidElementType(a) {
  return "string" === typeof a || "function" === typeof a || a === e || a === m || a === g || a === f || a === p || a === q || "object" === typeof a && null !== a && (a.$$typeof === t || a.$$typeof === r || a.$$typeof === h || a.$$typeof === k || a.$$typeof === n || a.$$typeof === w || a.$$typeof === x || a.$$typeof === y || a.$$typeof === v);
};
var typeOf = z;
var reactIs_production_min = {
  AsyncMode: AsyncMode,
  ConcurrentMode: ConcurrentMode,
  ContextConsumer: ContextConsumer,
  ContextProvider: ContextProvider,
  Element: Element,
  ForwardRef: ForwardRef,
  Fragment: Fragment,
  Lazy: Lazy,
  Memo: Memo,
  Portal: Portal,
  Profiler: Profiler,
  StrictMode: StrictMode,
  Suspense: Suspense,
  isAsyncMode: isAsyncMode,
  isConcurrentMode: isConcurrentMode,
  isContextConsumer: isContextConsumer,
  isContextProvider: isContextProvider,
  isElement: isElement,
  isForwardRef: isForwardRef,
  isFragment: isFragment,
  isLazy: isLazy,
  isMemo: isMemo,
  isPortal: isPortal,
  isProfiler: isProfiler,
  isStrictMode: isStrictMode,
  isSuspense: isSuspense,
  isValidElementType: isValidElementType,
  typeOf: typeOf
};

var reactIs_development = createCommonjsModule(function (module, exports) {

  if (process.env.NODE_ENV !== "production") {
    (function () {

      var hasSymbol = typeof Symbol === 'function' && Symbol.for;
      var REACT_ELEMENT_TYPE = hasSymbol ? Symbol.for('react.element') : 0xeac7;
      var REACT_PORTAL_TYPE = hasSymbol ? Symbol.for('react.portal') : 0xeaca;
      var REACT_FRAGMENT_TYPE = hasSymbol ? Symbol.for('react.fragment') : 0xeacb;
      var REACT_STRICT_MODE_TYPE = hasSymbol ? Symbol.for('react.strict_mode') : 0xeacc;
      var REACT_PROFILER_TYPE = hasSymbol ? Symbol.for('react.profiler') : 0xead2;
      var REACT_PROVIDER_TYPE = hasSymbol ? Symbol.for('react.provider') : 0xeacd;
      var REACT_CONTEXT_TYPE = hasSymbol ? Symbol.for('react.context') : 0xeace;

      var REACT_ASYNC_MODE_TYPE = hasSymbol ? Symbol.for('react.async_mode') : 0xeacf;
      var REACT_CONCURRENT_MODE_TYPE = hasSymbol ? Symbol.for('react.concurrent_mode') : 0xeacf;
      var REACT_FORWARD_REF_TYPE = hasSymbol ? Symbol.for('react.forward_ref') : 0xead0;
      var REACT_SUSPENSE_TYPE = hasSymbol ? Symbol.for('react.suspense') : 0xead1;
      var REACT_SUSPENSE_LIST_TYPE = hasSymbol ? Symbol.for('react.suspense_list') : 0xead8;
      var REACT_MEMO_TYPE = hasSymbol ? Symbol.for('react.memo') : 0xead3;
      var REACT_LAZY_TYPE = hasSymbol ? Symbol.for('react.lazy') : 0xead4;
      var REACT_BLOCK_TYPE = hasSymbol ? Symbol.for('react.block') : 0xead9;
      var REACT_FUNDAMENTAL_TYPE = hasSymbol ? Symbol.for('react.fundamental') : 0xead5;
      var REACT_RESPONDER_TYPE = hasSymbol ? Symbol.for('react.responder') : 0xead6;
      var REACT_SCOPE_TYPE = hasSymbol ? Symbol.for('react.scope') : 0xead7;
      function isValidElementType(type) {
        return typeof type === 'string' || typeof type === 'function' ||
        type === REACT_FRAGMENT_TYPE || type === REACT_CONCURRENT_MODE_TYPE || type === REACT_PROFILER_TYPE || type === REACT_STRICT_MODE_TYPE || type === REACT_SUSPENSE_TYPE || type === REACT_SUSPENSE_LIST_TYPE || typeof type === 'object' && type !== null && (type.$$typeof === REACT_LAZY_TYPE || type.$$typeof === REACT_MEMO_TYPE || type.$$typeof === REACT_PROVIDER_TYPE || type.$$typeof === REACT_CONTEXT_TYPE || type.$$typeof === REACT_FORWARD_REF_TYPE || type.$$typeof === REACT_FUNDAMENTAL_TYPE || type.$$typeof === REACT_RESPONDER_TYPE || type.$$typeof === REACT_SCOPE_TYPE || type.$$typeof === REACT_BLOCK_TYPE);
      }
      function typeOf(object) {
        if (typeof object === 'object' && object !== null) {
          var $$typeof = object.$$typeof;
          switch ($$typeof) {
            case REACT_ELEMENT_TYPE:
              var type = object.type;
              switch (type) {
                case REACT_ASYNC_MODE_TYPE:
                case REACT_CONCURRENT_MODE_TYPE:
                case REACT_FRAGMENT_TYPE:
                case REACT_PROFILER_TYPE:
                case REACT_STRICT_MODE_TYPE:
                case REACT_SUSPENSE_TYPE:
                  return type;
                default:
                  var $$typeofType = type && type.$$typeof;
                  switch ($$typeofType) {
                    case REACT_CONTEXT_TYPE:
                    case REACT_FORWARD_REF_TYPE:
                    case REACT_LAZY_TYPE:
                    case REACT_MEMO_TYPE:
                    case REACT_PROVIDER_TYPE:
                      return $$typeofType;
                    default:
                      return $$typeof;
                  }
              }
            case REACT_PORTAL_TYPE:
              return $$typeof;
          }
        }
        return undefined;
      }

      var AsyncMode = REACT_ASYNC_MODE_TYPE;
      var ConcurrentMode = REACT_CONCURRENT_MODE_TYPE;
      var ContextConsumer = REACT_CONTEXT_TYPE;
      var ContextProvider = REACT_PROVIDER_TYPE;
      var Element = REACT_ELEMENT_TYPE;
      var ForwardRef = REACT_FORWARD_REF_TYPE;
      var Fragment = REACT_FRAGMENT_TYPE;
      var Lazy = REACT_LAZY_TYPE;
      var Memo = REACT_MEMO_TYPE;
      var Portal = REACT_PORTAL_TYPE;
      var Profiler = REACT_PROFILER_TYPE;
      var StrictMode = REACT_STRICT_MODE_TYPE;
      var Suspense = REACT_SUSPENSE_TYPE;
      var hasWarnedAboutDeprecatedIsAsyncMode = false;

      function isAsyncMode(object) {
        {
          if (!hasWarnedAboutDeprecatedIsAsyncMode) {
            hasWarnedAboutDeprecatedIsAsyncMode = true;

            console['warn']('The ReactIs.isAsyncMode() alias has been deprecated, ' + 'and will be removed in React 17+. Update your code to use ' + 'ReactIs.isConcurrentMode() instead. It has the exact same API.');
          }
        }
        return isConcurrentMode(object) || typeOf(object) === REACT_ASYNC_MODE_TYPE;
      }
      function isConcurrentMode(object) {
        return typeOf(object) === REACT_CONCURRENT_MODE_TYPE;
      }
      function isContextConsumer(object) {
        return typeOf(object) === REACT_CONTEXT_TYPE;
      }
      function isContextProvider(object) {
        return typeOf(object) === REACT_PROVIDER_TYPE;
      }
      function isElement(object) {
        return typeof object === 'object' && object !== null && object.$$typeof === REACT_ELEMENT_TYPE;
      }
      function isForwardRef(object) {
        return typeOf(object) === REACT_FORWARD_REF_TYPE;
      }
      function isFragment(object) {
        return typeOf(object) === REACT_FRAGMENT_TYPE;
      }
      function isLazy(object) {
        return typeOf(object) === REACT_LAZY_TYPE;
      }
      function isMemo(object) {
        return typeOf(object) === REACT_MEMO_TYPE;
      }
      function isPortal(object) {
        return typeOf(object) === REACT_PORTAL_TYPE;
      }
      function isProfiler(object) {
        return typeOf(object) === REACT_PROFILER_TYPE;
      }
      function isStrictMode(object) {
        return typeOf(object) === REACT_STRICT_MODE_TYPE;
      }
      function isSuspense(object) {
        return typeOf(object) === REACT_SUSPENSE_TYPE;
      }
      exports.AsyncMode = AsyncMode;
      exports.ConcurrentMode = ConcurrentMode;
      exports.ContextConsumer = ContextConsumer;
      exports.ContextProvider = ContextProvider;
      exports.Element = Element;
      exports.ForwardRef = ForwardRef;
      exports.Fragment = Fragment;
      exports.Lazy = Lazy;
      exports.Memo = Memo;
      exports.Portal = Portal;
      exports.Profiler = Profiler;
      exports.StrictMode = StrictMode;
      exports.Suspense = Suspense;
      exports.isAsyncMode = isAsyncMode;
      exports.isConcurrentMode = isConcurrentMode;
      exports.isContextConsumer = isContextConsumer;
      exports.isContextProvider = isContextProvider;
      exports.isElement = isElement;
      exports.isForwardRef = isForwardRef;
      exports.isFragment = isFragment;
      exports.isLazy = isLazy;
      exports.isMemo = isMemo;
      exports.isPortal = isPortal;
      exports.isProfiler = isProfiler;
      exports.isStrictMode = isStrictMode;
      exports.isSuspense = isSuspense;
      exports.isValidElementType = isValidElementType;
      exports.typeOf = typeOf;
    })();
  }
});

var reactIs = createCommonjsModule(function (module) {

  if (process.env.NODE_ENV === 'production') {
    module.exports = reactIs_production_min;
  } else {
    module.exports = reactIs_development;
  }
});

var getOwnPropertySymbols = Object.getOwnPropertySymbols;
var hasOwnProperty = Object.prototype.hasOwnProperty;
var propIsEnumerable = Object.prototype.propertyIsEnumerable;
function toObject(val) {
  if (val === null || val === undefined) {
    throw new TypeError('Object.assign cannot be called with null or undefined');
  }
  return Object(val);
}
function shouldUseNative() {
  try {
    if (!Object.assign) {
      return false;
    }

    var test1 = new String('abc');
    test1[5] = 'de';
    if (Object.getOwnPropertyNames(test1)[0] === '5') {
      return false;
    }

    var test2 = {};
    for (var i = 0; i < 10; i++) {
      test2['_' + String.fromCharCode(i)] = i;
    }
    var order2 = Object.getOwnPropertyNames(test2).map(function (n) {
      return test2[n];
    });
    if (order2.join('') !== '0123456789') {
      return false;
    }

    var test3 = {};
    'abcdefghijklmnopqrst'.split('').forEach(function (letter) {
      test3[letter] = letter;
    });
    if (Object.keys(Object.assign({}, test3)).join('') !== 'abcdefghijklmnopqrst') {
      return false;
    }
    return true;
  } catch (err) {
    return false;
  }
}
var objectAssign = shouldUseNative() ? Object.assign : function (target, source) {
  var from;
  var to = toObject(target);
  var symbols;
  for (var s = 1; s < arguments.length; s++) {
    from = Object(arguments[s]);
    for (var key in from) {
      if (hasOwnProperty.call(from, key)) {
        to[key] = from[key];
      }
    }
    if (getOwnPropertySymbols) {
      symbols = getOwnPropertySymbols(from);
      for (var i = 0; i < symbols.length; i++) {
        if (propIsEnumerable.call(from, symbols[i])) {
          to[symbols[i]] = from[symbols[i]];
        }
      }
    }
  }
  return to;
};

var ReactPropTypesSecret = 'SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED';
var ReactPropTypesSecret_1 = ReactPropTypesSecret;

var has = Function.call.bind(Object.prototype.hasOwnProperty);

var printWarning = function printWarning() {};
if (process.env.NODE_ENV !== 'production') {
  var ReactPropTypesSecret$1 = ReactPropTypesSecret_1;
  var loggedTypeFailures = {};
  var has$1 = has;
  printWarning = function printWarning(text) {
    var message = 'Warning: ' + text;
    if (typeof console !== 'undefined') {
      console.error(message);
    }
    try {
      throw new Error(message);
    } catch (x) {}
  };
}

function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
  if (process.env.NODE_ENV !== 'production') {
    for (var typeSpecName in typeSpecs) {
      if (has$1(typeSpecs, typeSpecName)) {
        var error;
        try {
          if (typeof typeSpecs[typeSpecName] !== 'function') {
            var err = Error((componentName || 'React class') + ': ' + location + ' type `' + typeSpecName + '` is invalid; ' + 'it must be a function, usually from the `prop-types` package, but received `' + typeof typeSpecs[typeSpecName] + '`.' + 'This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.');
            err.name = 'Invariant Violation';
            throw err;
          }
          error = typeSpecs[typeSpecName](values, typeSpecName, componentName, location, null, ReactPropTypesSecret$1);
        } catch (ex) {
          error = ex;
        }
        if (error && !(error instanceof Error)) {
          printWarning((componentName || 'React class') + ': type specification of ' + location + ' `' + typeSpecName + '` is invalid; the type checker ' + 'function must return `null` or an `Error` but returned a ' + typeof error + '. ' + 'You may have forgotten to pass an argument to the type checker ' + 'creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and ' + 'shape all require an argument).');
        }
        if (error instanceof Error && !(error.message in loggedTypeFailures)) {
          loggedTypeFailures[error.message] = true;
          var stack = getStack ? getStack() : '';
          printWarning('Failed ' + location + ' type: ' + error.message + (stack != null ? stack : ''));
        }
      }
    }
  }
}

checkPropTypes.resetWarningCache = function () {
  if (process.env.NODE_ENV !== 'production') {
    loggedTypeFailures = {};
  }
};
var checkPropTypes_1 = checkPropTypes;

var printWarning$1 = function printWarning() {};
if (process.env.NODE_ENV !== 'production') {
  printWarning$1 = function printWarning(text) {
    var message = 'Warning: ' + text;
    if (typeof console !== 'undefined') {
      console.error(message);
    }
    try {
      throw new Error(message);
    } catch (x) {}
  };
}
function emptyFunctionThatReturnsNull() {
  return null;
}
var factoryWithTypeCheckers = function factoryWithTypeCheckers(isValidElement, throwOnDirectAccess) {
  var ITERATOR_SYMBOL = typeof Symbol === 'function' && Symbol.iterator;
  var FAUX_ITERATOR_SYMBOL = '@@iterator';

  function getIteratorFn(maybeIterable) {
    var iteratorFn = maybeIterable && (ITERATOR_SYMBOL && maybeIterable[ITERATOR_SYMBOL] || maybeIterable[FAUX_ITERATOR_SYMBOL]);
    if (typeof iteratorFn === 'function') {
      return iteratorFn;
    }
  }

  var ANONYMOUS = '<<anonymous>>';

  var ReactPropTypes = {
    array: createPrimitiveTypeChecker('array'),
    bigint: createPrimitiveTypeChecker('bigint'),
    bool: createPrimitiveTypeChecker('boolean'),
    func: createPrimitiveTypeChecker('function'),
    number: createPrimitiveTypeChecker('number'),
    object: createPrimitiveTypeChecker('object'),
    string: createPrimitiveTypeChecker('string'),
    symbol: createPrimitiveTypeChecker('symbol'),
    any: createAnyTypeChecker(),
    arrayOf: createArrayOfTypeChecker,
    element: createElementTypeChecker(),
    elementType: createElementTypeTypeChecker(),
    instanceOf: createInstanceTypeChecker,
    node: createNodeChecker(),
    objectOf: createObjectOfTypeChecker,
    oneOf: createEnumTypeChecker,
    oneOfType: createUnionTypeChecker,
    shape: createShapeTypeChecker,
    exact: createStrictShapeTypeChecker
  };

  function is(x, y) {
    if (x === y) {
      return x !== 0 || 1 / x === 1 / y;
    } else {
      return x !== x && y !== y;
    }
  }

  function PropTypeError(message, data) {
    this.message = message;
    this.data = data && typeof data === 'object' ? data : {};
    this.stack = '';
  }
  PropTypeError.prototype = Error.prototype;
  function createChainableTypeChecker(validate) {
    if (process.env.NODE_ENV !== 'production') {
      var manualPropTypeCallCache = {};
      var manualPropTypeWarningCount = 0;
    }
    function checkType(isRequired, props, propName, componentName, location, propFullName, secret) {
      componentName = componentName || ANONYMOUS;
      propFullName = propFullName || propName;
      if (secret !== ReactPropTypesSecret_1) {
        if (throwOnDirectAccess) {
          var err = new Error('Calling PropTypes validators directly is not supported by the `prop-types` package. ' + 'Use `PropTypes.checkPropTypes()` to call them. ' + 'Read more at http://fb.me/use-check-prop-types');
          err.name = 'Invariant Violation';
          throw err;
        } else if (process.env.NODE_ENV !== 'production' && typeof console !== 'undefined') {
          var cacheKey = componentName + ':' + propName;
          if (!manualPropTypeCallCache[cacheKey] &&
          manualPropTypeWarningCount < 3) {
            printWarning$1('You are manually calling a React.PropTypes validation ' + 'function for the `' + propFullName + '` prop on `' + componentName + '`. This is deprecated ' + 'and will throw in the standalone `prop-types` package. ' + 'You may be seeing this warning due to a third-party PropTypes ' + 'library. See https://fb.me/react-warning-dont-call-proptypes ' + 'for details.');
            manualPropTypeCallCache[cacheKey] = true;
            manualPropTypeWarningCount++;
          }
        }
      }
      if (props[propName] == null) {
        if (isRequired) {
          if (props[propName] === null) {
            return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required ' + ('in `' + componentName + '`, but its value is `null`.'));
          }
          return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required in ' + ('`' + componentName + '`, but its value is `undefined`.'));
        }
        return null;
      } else {
        return validate(props, propName, componentName, location, propFullName);
      }
    }
    var chainedCheckType = checkType.bind(null, false);
    chainedCheckType.isRequired = checkType.bind(null, true);
    return chainedCheckType;
  }
  function createPrimitiveTypeChecker(expectedType) {
    function validate(props, propName, componentName, location, propFullName, secret) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== expectedType) {
        var preciseType = getPreciseType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + preciseType + '` supplied to `' + componentName + '`, expected ') + ('`' + expectedType + '`.'), {
          expectedType: expectedType
        });
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createAnyTypeChecker() {
    return createChainableTypeChecker(emptyFunctionThatReturnsNull);
  }
  function createArrayOfTypeChecker(typeChecker) {
    function validate(props, propName, componentName, location, propFullName) {
      if (typeof typeChecker !== 'function') {
        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside arrayOf.');
      }
      var propValue = props[propName];
      if (!Array.isArray(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an array.'));
      }
      for (var i = 0; i < propValue.length; i++) {
        var error = typeChecker(propValue, i, componentName, location, propFullName + '[' + i + ']', ReactPropTypesSecret_1);
        if (error instanceof Error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createElementTypeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      if (!isValidElement(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createElementTypeTypeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      if (!reactIs.isValidElementType(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement type.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createInstanceTypeChecker(expectedClass) {
    function validate(props, propName, componentName, location, propFullName) {
      if (!(props[propName] instanceof expectedClass)) {
        var expectedClassName = expectedClass.name || ANONYMOUS;
        var actualClassName = getClassName(props[propName]);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + actualClassName + '` supplied to `' + componentName + '`, expected ') + ('instance of `' + expectedClassName + '`.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createEnumTypeChecker(expectedValues) {
    if (!Array.isArray(expectedValues)) {
      if (process.env.NODE_ENV !== 'production') {
        if (arguments.length > 1) {
          printWarning$1('Invalid arguments supplied to oneOf, expected an array, got ' + arguments.length + ' arguments. ' + 'A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z]).');
        } else {
          printWarning$1('Invalid argument supplied to oneOf, expected an array.');
        }
      }
      return emptyFunctionThatReturnsNull;
    }
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      for (var i = 0; i < expectedValues.length; i++) {
        if (is(propValue, expectedValues[i])) {
          return null;
        }
      }
      var valuesString = JSON.stringify(expectedValues, function replacer(key, value) {
        var type = getPreciseType(value);
        if (type === 'symbol') {
          return String(value);
        }
        return value;
      });
      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of value `' + String(propValue) + '` ' + ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.'));
    }
    return createChainableTypeChecker(validate);
  }
  function createObjectOfTypeChecker(typeChecker) {
    function validate(props, propName, componentName, location, propFullName) {
      if (typeof typeChecker !== 'function') {
        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside objectOf.');
      }
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an object.'));
      }
      for (var key in propValue) {
        if (has(propValue, key)) {
          var error = typeChecker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
          if (error instanceof Error) {
            return error;
          }
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createUnionTypeChecker(arrayOfTypeCheckers) {
    if (!Array.isArray(arrayOfTypeCheckers)) {
      process.env.NODE_ENV !== 'production' ? printWarning$1('Invalid argument supplied to oneOfType, expected an instance of array.') : void 0;
      return emptyFunctionThatReturnsNull;
    }
    for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
      var checker = arrayOfTypeCheckers[i];
      if (typeof checker !== 'function') {
        printWarning$1('Invalid argument supplied to oneOfType. Expected an array of check functions, but ' + 'received ' + getPostfixForTypeWarning(checker) + ' at index ' + i + '.');
        return emptyFunctionThatReturnsNull;
      }
    }
    function validate(props, propName, componentName, location, propFullName) {
      var expectedTypes = [];
      for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
        var checker = arrayOfTypeCheckers[i];
        var checkerResult = checker(props, propName, componentName, location, propFullName, ReactPropTypesSecret_1);
        if (checkerResult == null) {
          return null;
        }
        if (checkerResult.data && has(checkerResult.data, 'expectedType')) {
          expectedTypes.push(checkerResult.data.expectedType);
        }
      }
      var expectedTypesMessage = expectedTypes.length > 0 ? ', expected one of type [' + expectedTypes.join(', ') + ']' : '';
      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`' + expectedTypesMessage + '.'));
    }
    return createChainableTypeChecker(validate);
  }
  function createNodeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      if (!isNode(props[propName])) {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`, expected a ReactNode.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function invalidValidatorError(componentName, location, propFullName, key, type) {
    return new PropTypeError((componentName || 'React class') + ': ' + location + ' type `' + propFullName + '.' + key + '` is invalid; ' + 'it must be a function, usually from the `prop-types` package, but received `' + type + '`.');
  }
  function createShapeTypeChecker(shapeTypes) {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
      }
      for (var key in shapeTypes) {
        var checker = shapeTypes[key];
        if (typeof checker !== 'function') {
          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
        }
        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
        if (error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function createStrictShapeTypeChecker(shapeTypes) {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
      }
      var allKeys = objectAssign({}, props[propName], shapeTypes);
      for (var key in allKeys) {
        var checker = shapeTypes[key];
        if (has(shapeTypes, key) && typeof checker !== 'function') {
          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
        }
        if (!checker) {
          return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` key `' + key + '` supplied to `' + componentName + '`.' + '\nBad object: ' + JSON.stringify(props[propName], null, '  ') + '\nValid keys: ' + JSON.stringify(Object.keys(shapeTypes), null, '  '));
        }
        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
        if (error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }
  function isNode(propValue) {
    switch (typeof propValue) {
      case 'number':
      case 'string':
      case 'undefined':
        return true;
      case 'boolean':
        return !propValue;
      case 'object':
        if (Array.isArray(propValue)) {
          return propValue.every(isNode);
        }
        if (propValue === null || isValidElement(propValue)) {
          return true;
        }
        var iteratorFn = getIteratorFn(propValue);
        if (iteratorFn) {
          var iterator = iteratorFn.call(propValue);
          var step;
          if (iteratorFn !== propValue.entries) {
            while (!(step = iterator.next()).done) {
              if (!isNode(step.value)) {
                return false;
              }
            }
          } else {
            while (!(step = iterator.next()).done) {
              var entry = step.value;
              if (entry) {
                if (!isNode(entry[1])) {
                  return false;
                }
              }
            }
          }
        } else {
          return false;
        }
        return true;
      default:
        return false;
    }
  }
  function isSymbol(propType, propValue) {
    if (propType === 'symbol') {
      return true;
    }

    if (!propValue) {
      return false;
    }

    if (propValue['@@toStringTag'] === 'Symbol') {
      return true;
    }

    if (typeof Symbol === 'function' && propValue instanceof Symbol) {
      return true;
    }
    return false;
  }

  function getPropType(propValue) {
    var propType = typeof propValue;
    if (Array.isArray(propValue)) {
      return 'array';
    }
    if (propValue instanceof RegExp) {
      return 'object';
    }
    if (isSymbol(propType, propValue)) {
      return 'symbol';
    }
    return propType;
  }

  function getPreciseType(propValue) {
    if (typeof propValue === 'undefined' || propValue === null) {
      return '' + propValue;
    }
    var propType = getPropType(propValue);
    if (propType === 'object') {
      if (propValue instanceof Date) {
        return 'date';
      } else if (propValue instanceof RegExp) {
        return 'regexp';
      }
    }
    return propType;
  }

  function getPostfixForTypeWarning(value) {
    var type = getPreciseType(value);
    switch (type) {
      case 'array':
      case 'object':
        return 'an ' + type;
      case 'boolean':
      case 'date':
      case 'regexp':
        return 'a ' + type;
      default:
        return type;
    }
  }

  function getClassName(propValue) {
    if (!propValue.constructor || !propValue.constructor.name) {
      return ANONYMOUS;
    }
    return propValue.constructor.name;
  }
  ReactPropTypes.checkPropTypes = checkPropTypes_1;
  ReactPropTypes.resetWarningCache = checkPropTypes_1.resetWarningCache;
  ReactPropTypes.PropTypes = ReactPropTypes;
  return ReactPropTypes;
};

function emptyFunction() {}
function emptyFunctionWithReset() {}
emptyFunctionWithReset.resetWarningCache = emptyFunction;
var factoryWithThrowingShims = function factoryWithThrowingShims() {
  function shim(props, propName, componentName, location, propFullName, secret) {
    if (secret === ReactPropTypesSecret_1) {
      return;
    }
    var err = new Error('Calling PropTypes validators directly is not supported by the `prop-types` package. ' + 'Use PropTypes.checkPropTypes() to call them. ' + 'Read more at http://fb.me/use-check-prop-types');
    err.name = 'Invariant Violation';
    throw err;
  }
  shim.isRequired = shim;
  function getShim() {
    return shim;
  }
  var ReactPropTypes = {
    array: shim,
    bigint: shim,
    bool: shim,
    func: shim,
    number: shim,
    object: shim,
    string: shim,
    symbol: shim,
    any: shim,
    arrayOf: getShim,
    element: shim,
    elementType: shim,
    instanceOf: getShim,
    node: shim,
    objectOf: getShim,
    oneOf: getShim,
    oneOfType: getShim,
    shape: getShim,
    exact: getShim,
    checkPropTypes: emptyFunctionWithReset,
    resetWarningCache: emptyFunction
  };
  ReactPropTypes.PropTypes = ReactPropTypes;
  return ReactPropTypes;
};

var propTypes = createCommonjsModule(function (module) {

  if (process.env.NODE_ENV !== 'production') {
    var ReactIs = reactIs;

    var throwOnDirectAccess = true;
    module.exports = factoryWithTypeCheckers(ReactIs.isElement, throwOnDirectAccess);
  } else {
    module.exports = factoryWithThrowingShims();
  }
});

var SearchConnection = function SearchConnection(_ref) {
  var _Digit$Hooks$ws$usews;
  var propsConfig = _ref.config,
    formData = _ref.formData;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var validation = {};
  var history = reactRouterDom.useHistory();
  var tenantId = Digit.ULBService.getCurrentTenantId();
  var logginedUser = Digit.UserService.getUser();
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : formData.mobileNumber) || ""),
    mobileNumber = _useState[0],
    setMobileNumber = _useState[1];
  var _useState2 = React.useState((formData === null || formData === void 0 ? void 0 : formData.consumerNumber) || ""),
    consumerNumber = _useState2[0],
    setconsumerNumber = _useState2[1];
  var _useState3 = React.useState((formData === null || formData === void 0 ? void 0 : formData.oldconsumerNumber) || ""),
    oldconsumerNumber = _useState3[0],
    setoldconsumerNumber = _useState3[1];
  var _useState4 = React.useState((formData === null || formData === void 0 ? void 0 : formData.doorNumber) || ""),
    doorNumber = _useState4[0],
    setdoorNumber = _useState4[1];
  var _useState5 = React.useState((formData === null || formData === void 0 ? void 0 : formData.propertyId) || ""),
    propertyId = _useState5[0],
    setpropertyId = _useState5[1];
  var _useState6 = React.useState((formData === null || formData === void 0 ? void 0 : formData.consumerName) || ""),
    consumerName = _useState6[0],
    setconsumerName = _useState6[1];
  var _useState7 = React.useState((formData === null || formData === void 0 ? void 0 : formData.Servicecateogry) || "");
  var _useState8 = React.useState((formData === null || formData === void 0 ? void 0 : formData.city) || ""),
    city = _useState8[0],
    setcity = _useState8[1];
  var _useState9 = React.useState(null),
    showToast = _useState9[0],
    setShowToast = _useState9[1];
  var _useState10 = React.useState((formData === null || formData === void 0 ? void 0 : formData.locality) || ""),
    locality = _useState10[0],
    setLocality = _useState10[1];
  var _useState11 = React.useState((formData === null || formData === void 0 ? void 0 : formData.searchType) || {
      code: "CONSUMER_NUMBER",
      i18nKey: "WS_CONSUMER_NUMBER_SEARCH"
    }),
    searchType = _useState11[0],
    setSearchType = _useState11[1];
  var allCities = (_Digit$Hooks$ws$usews = Digit.Hooks.ws.usewsTenants()) === null || _Digit$Hooks$ws$usews === void 0 ? void 0 : _Digit$Hooks$ws$usews.sort(function (a, b) {
    var _a$i18nKey, _a$i18nKey$localeComp;
    return a === null || a === void 0 ? void 0 : (_a$i18nKey = a.i18nKey) === null || _a$i18nKey === void 0 ? void 0 : (_a$i18nKey$localeComp = _a$i18nKey.localeCompare) === null || _a$i18nKey$localeComp === void 0 ? void 0 : _a$i18nKey$localeComp.call(_a$i18nKey, b === null || b === void 0 ? void 0 : b.i18nKey);
  });
  var _useState12 = React.useState(null),
    mobileNumberError = _useState12[0],
    setmobileNumberError = _useState12[1];
  var _Digit$Hooks$mcollect = Digit.Hooks.mcollect.useMCollectMDMS(tenantId, "BillingService", "BusinessService"),
    isLoading = _Digit$Hooks$mcollect.isLoading;
  if (isLoading) {
    return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null);
  }
  var onConnectionSearch = function onConnectionSearch(data) {
    try {
      if (searchType.code === "CONSUMER_NUMBER") {
        if (!city.code) setShowToast({
          key: true,
          label: "WS_PLEASE_PROVIDE_CITY"
        });else if (logginedUser == null && !locality) setShowToast({
          key: true,
          label: "WS_PLEASE_PROVIDE_LOCALITY"
        });else if (!mobileNumber && !consumerNumber && !oldconsumerNumber && !propertyId) {
          setShowToast({
            key: true,
            label: "WS_HOME_SEARCH_CONN_RESULTS_DESC"
          });
        } else {
          history.push("/digit-ui/citizen/ws/search-results?mobileNumber=" + mobileNumber + "&consumerNumber=" + consumerNumber + "&oldconsumerNumber=" + oldconsumerNumber + "&propertyId=" + propertyId + "&tenantId=" + city.code + "&locality=" + locality.code);
        }
      } else {
        if (!city.code) setShowToast({
          key: true,
          label: "WS_PLEASE_PROVIDE_CITY"
        });else if (logginedUser == null && !locality) setShowToast({
          key: true,
          label: "WS_PLEASE_PROVIDE_LOCALITY"
        });else if (!doorNumber && !consumerName) setShowToast({
          key: true,
          label: "WS_HOME_SEARCH_CONN_RESULTS_DESC"
        });else {
          history.push("/digit-ui/citizen/ws/search-results?doorNumber=" + doorNumber + "&consumerName=" + consumerName + "&tenantId=" + city.code + "&locality=" + locality.code);
        }
      }
      return Promise.resolve();
    } catch (e) {
      return Promise.reject(e);
    }
  };
  var SearchTypes = [{
    code: "CONSUMER_NUMBER",
    i18nKey: "WS_CONSUMER_NUMBER_SEARCH"
  }, {
    code: "CONNECTION_DETAILS",
    i18nKey: "WS_CONNECTION_DETAILS_SEARCH"
  }];
  function setMobileNo(e) {
    setmobileNumberError(null);
    var validation = "^\\d{10}$";
    if (!e.target.value.match(validation)) {
      setmobileNumberError("CORE_COMMON_PHONENO_INVALIDMSG");
    }
    setMobileNumber(e.target.value);
  }
  function selectconsumerNumber(e) {
    setconsumerNumber(e.target.value);
  }
  function selectoldconsumerNumber(e) {
    setoldconsumerNumber(e.target.value);
  }
  function selectdoorNumber(e) {
    setdoorNumber(e.target.value);
  }
  function selectpropertyId(e) {
    setpropertyId(e.target.value);
  }
  function selectconsumerName(e) {
    setconsumerName(e.target.value);
  }
  function selectCity(value) {
    setcity(value);
    setLocality("");
  }
  function selectLocality(value) {
    setLocality(value);
  }
  function selectSearchType(value) {
    setSearchType(value);
  }
  return /*#__PURE__*/React__default.createElement("div", {
    style: {
      marginTop: "16px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    config: propsConfig,
    label: propsConfig.texts.submitButtonLabel,
    heading: propsConfig.texts.header,
    text: propsConfig.texts.text,
    cardStyle: {
      margin: "auto"
    },
    headingStyle: {
      fontSize: "32px",
      marginBottom: "16px"
    },
    onSelect: onConnectionSearch,
    componentInFront: /*#__PURE__*/React__default.createElement("div", {
      className: "employee-card-input employee-card-input--front"
    }, "+91"),
    isDisabled: false,
    forcedError: t(mobileNumberError)
    ,
    t: t
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioOrSelect, _extends({
    className: "form-field",
    isMandatory: true,
    t: t,
    optionKey: "code",
    name: "SearchType",
    options: SearchTypes,
    value: searchType,
    selectedOption: searchType,
    onSelect: selectSearchType
  }, validation = {
    isRequired: true,
    title: t("WS_SEARCH_TYPE_MANDATORY")
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_PROP_DETAIL_CITY") + "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioOrSelect, _extends({
    className: "form-field",
    isMandatory: true,
    t: t,
    optionKey: "code",
    name: "City",
    options: allCities,
    value: city,
    selectedOption: city,
    onSelect: selectCity
  }, validation = {
    isRequired: true,
    title: t("UC_CITY_MANDATORY")
  })), city && !logginedUser && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_PROP_DETAIL_LOCALITY_LABEL") + "*"), city && !logginedUser && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Localities, {
    selectLocality: selectLocality,
    tenantId: city === null || city === void 0 ? void 0 : city.code,
    boundaryType: "revenue",
    keepNull: false,
    optionCardStyles: {
      height: "600px",
      overflow: "auto",
      zIndex: "10"
    },
    selected: locality
    ,
    disableLoader: false
  }), searchType && (searchType === null || searchType === void 0 ? void 0 : searchType.code) == "CONSUMER_NUMBER" && /*#__PURE__*/React__default.createElement("div", {
    style: {
      border: "solid",
      borderRadius: "5px",
      padding: "10px",
      paddingTop: "20px",
      marginTop: "10px",
      borderColor: "#f3f3f3",
      background: "#FAFAFA",
      marginBottom: "20px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_CONSUMER_NUMBER_LABEL")), /*#__PURE__*/React__default.createElement("div", {
    className: "field-container"
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "employee-card-input employee-card-input--front",
    style: {
      marginTop: "-1px"
    }
  }, "+91"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    type: "mobileNumber",
    style: {
      background: "#FAFAFA"
    },
    t: t,
    isMandatory: false,
    optionKey: "i18nKey",
    name: "mobileNumber",
    value: mobileNumber,
    onChange: setMobileNo
  }, validation = {
    isRequired: false,
    pattern: "[6-9]{1}[0-9]{9}",
    type: "tel",
    title: t("CORE_COMMON_APPLICANT_MOBILE_NUMBER_INVALID")
  }))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    style: {
      textAlign: "center",
      color: "#505A5F"
    }
  }, "" + t("(or)")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_MYCONNECTIONS_CONSUMER_NO")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "any",
    isMandatory: false,
    style: {
      background: "#FAFAFA"
    },
    name: "consumerNumber",
    value: consumerNumber,
    onChange: selectconsumerNumber
  }), logginedUser && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    style: {
      textAlign: "center",
      color: "#505A5F"
    }
  }, "" + t("(or)")), logginedUser && /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_SEARCH_CONNNECTION_OLD_CONSUMER_LABEL")), logginedUser && /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "any",
    isMandatory: false,
    style: {
      background: "#FAFAFA"
    }
    ,
    name: "oldconsumerNumber",
    value: oldconsumerNumber,
    onChange: selectoldconsumerNumber
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, {
    style: {
      textAlign: "center",
      color: "#505A5F"
    }
  }, "" + t("(or)")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_PROPERTY_ID_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "any",
    isMandatory: false,
    style: {
      background: "#FAFAFA"
    }
    ,
    name: "propertyId",
    value: propertyId,
    onChange: selectpropertyId
  })), searchType && (searchType === null || searchType === void 0 ? void 0 : searchType.code) == "CONNECTION_DETAILS" && /*#__PURE__*/React__default.createElement("div", {
    style: {
      border: "solid",
      borderRadius: "5px",
      padding: "10px",
      paddingTop: "20px",
      marginTop: "10px",
      borderColor: "#f3f3f3",
      background: "#FAFAFA",
      marginBottom: "20px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_DOOR_NO_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "any",
    isMandatory: false,
    style: {
      background: "#FAFAFA"
    },
    name: "doorNumber",
    value: doorNumber,
    onChange: selectdoorNumber
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_CONSUMER_NAME_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "any",
    isMandatory: false,
    style: {
      background: "#FAFAFA"
    },
    name: "consumerName",
    value: consumerName,
    onChange: selectconsumerName
  }))), showToast && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    isDleteBtn: true,
    error: showToast.key,
    label: t(showToast.label),
    onClose: function onClose() {
      setShowToast(null);
    }
  }));
};
SearchConnection.propTypes = {
  loginParams: propTypes.any
};
SearchConnection.defaultProps = {
  loginParams: null
};

var SearchConnection$1 = function SearchConnection$1() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var params = React.useMemo(function () {
    return config.map(function (step) {
      var texts = {};
      for (var key in step.texts) {
        texts[key] = t(step.texts[key]);
      }
      return _extends({}, step, {
        texts: texts
      });
    }, [config]);
  });
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "" + path,
    exact: true
  }, /*#__PURE__*/React__default.createElement(SearchConnection, {
    config: params[0]
  })));
};

var config$1 = [{
  texts: {
    header: "CS_SEARCH_RESULTS",
    actionButtonLabel: "WS_VIEW_DETAILS_AND_PAY"
  },
  labels: [{
    label: "WS_COMMON_TABLE_COL_AMT_DUE_LABEL",
    key: "AmountDue",
    noteStyle: {
      fontSize: "24px",
      fontWeight: "bold"
    },
    notePrefix: " "
  }, {
    label: "WS_MYCONNECTIONS_CONSUMER_NO",
    key: "ConsumerNumber"
  }, {
    label: "WS_CONSUMER_NAME_LABEL",
    key: "ConsumerName"
  }, {
    label: "WS_MYCONNECTION_ADDRESS",
    key: "Address"
  }, {
    label: "WS_VIEW_BILL_DUE_DATE_LABEL",
    key: "DueDate"
  }]
}];

var bind = function bind(fn, thisArg) {
  return function wrap() {
    var args = new Array(arguments.length);
    for (var i = 0; i < args.length; i++) {
      args[i] = arguments[i];
    }
    return fn.apply(thisArg, args);
  };
};

var toString = Object.prototype.toString;

function isArray(val) {
  return toString.call(val) === '[object Array]';
}

function isUndefined(val) {
  return typeof val === 'undefined';
}

function isBuffer(val) {
  return val !== null && !isUndefined(val) && val.constructor !== null && !isUndefined(val.constructor) && typeof val.constructor.isBuffer === 'function' && val.constructor.isBuffer(val);
}

function isArrayBuffer(val) {
  return toString.call(val) === '[object ArrayBuffer]';
}

function isFormData(val) {
  return typeof FormData !== 'undefined' && val instanceof FormData;
}

function isArrayBufferView(val) {
  var result;
  if (typeof ArrayBuffer !== 'undefined' && ArrayBuffer.isView) {
    result = ArrayBuffer.isView(val);
  } else {
    result = val && val.buffer && val.buffer instanceof ArrayBuffer;
  }
  return result;
}

function isString(val) {
  return typeof val === 'string';
}

function isNumber(val) {
  return typeof val === 'number';
}

function isObject(val) {
  return val !== null && typeof val === 'object';
}

function isPlainObject(val) {
  if (toString.call(val) !== '[object Object]') {
    return false;
  }
  var prototype = Object.getPrototypeOf(val);
  return prototype === null || prototype === Object.prototype;
}

function isDate(val) {
  return toString.call(val) === '[object Date]';
}

function isFile(val) {
  return toString.call(val) === '[object File]';
}

function isBlob(val) {
  return toString.call(val) === '[object Blob]';
}

function isFunction(val) {
  return toString.call(val) === '[object Function]';
}

function isStream(val) {
  return isObject(val) && isFunction(val.pipe);
}

function isURLSearchParams(val) {
  return typeof URLSearchParams !== 'undefined' && val instanceof URLSearchParams;
}

function trim(str) {
  return str.replace(/^\s*/, '').replace(/\s*$/, '');
}

function isStandardBrowserEnv() {
  if (typeof navigator !== 'undefined' && (navigator.product === 'ReactNative' || navigator.product === 'NativeScript' || navigator.product === 'NS')) {
    return false;
  }
  return typeof window !== 'undefined' && typeof document !== 'undefined';
}

function forEach(obj, fn) {
  if (obj === null || typeof obj === 'undefined') {
    return;
  }

  if (typeof obj !== 'object') {
    obj = [obj];
  }
  if (isArray(obj)) {
    for (var i = 0, l = obj.length; i < l; i++) {
      fn.call(null, obj[i], i, obj);
    }
  } else {
    for (var key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        fn.call(null, obj[key], key, obj);
      }
    }
  }
}

function merge(
) {
  var result = {};
  function assignValue(val, key) {
    if (isPlainObject(result[key]) && isPlainObject(val)) {
      result[key] = merge(result[key], val);
    } else if (isPlainObject(val)) {
      result[key] = merge({}, val);
    } else if (isArray(val)) {
      result[key] = val.slice();
    } else {
      result[key] = val;
    }
  }
  for (var i = 0, l = arguments.length; i < l; i++) {
    forEach(arguments[i], assignValue);
  }
  return result;
}

function extend(a, b, thisArg) {
  forEach(b, function assignValue(val, key) {
    if (thisArg && typeof val === 'function') {
      a[key] = bind(val, thisArg);
    } else {
      a[key] = val;
    }
  });
  return a;
}

function stripBOM(content) {
  if (content.charCodeAt(0) === 0xFEFF) {
    content = content.slice(1);
  }
  return content;
}
var utils = {
  isArray: isArray,
  isArrayBuffer: isArrayBuffer,
  isBuffer: isBuffer,
  isFormData: isFormData,
  isArrayBufferView: isArrayBufferView,
  isString: isString,
  isNumber: isNumber,
  isObject: isObject,
  isPlainObject: isPlainObject,
  isUndefined: isUndefined,
  isDate: isDate,
  isFile: isFile,
  isBlob: isBlob,
  isFunction: isFunction,
  isStream: isStream,
  isURLSearchParams: isURLSearchParams,
  isStandardBrowserEnv: isStandardBrowserEnv,
  forEach: forEach,
  merge: merge,
  extend: extend,
  trim: trim,
  stripBOM: stripBOM
};

function encode(val) {
  return encodeURIComponent(val).replace(/%3A/gi, ':').replace(/%24/g, '$').replace(/%2C/gi, ',').replace(/%20/g, '+').replace(/%5B/gi, '[').replace(/%5D/gi, ']');
}

var buildURL = function buildURL(url, params, paramsSerializer) {
  if (!params) {
    return url;
  }
  var serializedParams;
  if (paramsSerializer) {
    serializedParams = paramsSerializer(params);
  } else if (utils.isURLSearchParams(params)) {
    serializedParams = params.toString();
  } else {
    var parts = [];
    utils.forEach(params, function serialize(val, key) {
      if (val === null || typeof val === 'undefined') {
        return;
      }
      if (utils.isArray(val)) {
        key = key + '[]';
      } else {
        val = [val];
      }
      utils.forEach(val, function parseValue(v) {
        if (utils.isDate(v)) {
          v = v.toISOString();
        } else if (utils.isObject(v)) {
          v = JSON.stringify(v);
        }
        parts.push(encode(key) + '=' + encode(v));
      });
    });
    serializedParams = parts.join('&');
  }
  if (serializedParams) {
    var hashmarkIndex = url.indexOf('#');
    if (hashmarkIndex !== -1) {
      url = url.slice(0, hashmarkIndex);
    }
    url += (url.indexOf('?') === -1 ? '?' : '&') + serializedParams;
  }
  return url;
};

function InterceptorManager() {
  this.handlers = [];
}

InterceptorManager.prototype.use = function use(fulfilled, rejected) {
  this.handlers.push({
    fulfilled: fulfilled,
    rejected: rejected
  });
  return this.handlers.length - 1;
};

InterceptorManager.prototype.eject = function eject(id) {
  if (this.handlers[id]) {
    this.handlers[id] = null;
  }
};

InterceptorManager.prototype.forEach = function forEach(fn) {
  utils.forEach(this.handlers, function forEachHandler(h) {
    if (h !== null) {
      fn(h);
    }
  });
};
var InterceptorManager_1 = InterceptorManager;

var transformData = function transformData(data, headers, fns) {
  utils.forEach(fns, function transform(fn) {
    data = fn(data, headers);
  });
  return data;
};

var isCancel = function isCancel(value) {
  return !!(value && value.__CANCEL__);
};

var normalizeHeaderName = function normalizeHeaderName(headers, normalizedName) {
  utils.forEach(headers, function processHeader(value, name) {
    if (name !== normalizedName && name.toUpperCase() === normalizedName.toUpperCase()) {
      headers[normalizedName] = value;
      delete headers[name];
    }
  });
};

var enhanceError = function enhanceError(error, config, code, request, response) {
  error.config = config;
  if (code) {
    error.code = code;
  }
  error.request = request;
  error.response = response;
  error.isAxiosError = true;
  error.toJSON = function toJSON() {
    return {
      message: this.message,
      name: this.name,
      description: this.description,
      number: this.number,
      fileName: this.fileName,
      lineNumber: this.lineNumber,
      columnNumber: this.columnNumber,
      stack: this.stack,
      config: this.config,
      code: this.code
    };
  };
  return error;
};

var createError = function createError(message, config, code, request, response) {
  var error = new Error(message);
  return enhanceError(error, config, code, request, response);
};

var settle = function settle(resolve, reject, response) {
  var validateStatus = response.config.validateStatus;
  if (!response.status || !validateStatus || validateStatus(response.status)) {
    resolve(response);
  } else {
    reject(createError('Request failed with status code ' + response.status, response.config, null, response.request, response));
  }
};

var cookies = utils.isStandardBrowserEnv() ?
function standardBrowserEnv() {
  return {
    write: function write(name, value, expires, path, domain, secure) {
      var cookie = [];
      cookie.push(name + '=' + encodeURIComponent(value));
      if (utils.isNumber(expires)) {
        cookie.push('expires=' + new Date(expires).toGMTString());
      }
      if (utils.isString(path)) {
        cookie.push('path=' + path);
      }
      if (utils.isString(domain)) {
        cookie.push('domain=' + domain);
      }
      if (secure === true) {
        cookie.push('secure');
      }
      document.cookie = cookie.join('; ');
    },
    read: function read(name) {
      var match = document.cookie.match(new RegExp('(^|;\\s*)(' + name + ')=([^;]*)'));
      return match ? decodeURIComponent(match[3]) : null;
    },
    remove: function remove(name) {
      this.write(name, '', Date.now() - 86400000);
    }
  };
}() :
function nonStandardBrowserEnv() {
  return {
    write: function write() {},
    read: function read() {
      return null;
    },
    remove: function remove() {}
  };
}();

var isAbsoluteURL = function isAbsoluteURL(url) {
  return /^([a-z][a-z\d\+\-\.]*:)?\/\//i.test(url);
};

var combineURLs = function combineURLs(baseURL, relativeURL) {
  return relativeURL ? baseURL.replace(/\/+$/, '') + '/' + relativeURL.replace(/^\/+/, '') : baseURL;
};

var buildFullPath = function buildFullPath(baseURL, requestedURL) {
  if (baseURL && !isAbsoluteURL(requestedURL)) {
    return combineURLs(baseURL, requestedURL);
  }
  return requestedURL;
};

var ignoreDuplicateOf = ['age', 'authorization', 'content-length', 'content-type', 'etag', 'expires', 'from', 'host', 'if-modified-since', 'if-unmodified-since', 'last-modified', 'location', 'max-forwards', 'proxy-authorization', 'referer', 'retry-after', 'user-agent'];

var parseHeaders = function parseHeaders(headers) {
  var parsed = {};
  var key;
  var val;
  var i;
  if (!headers) {
    return parsed;
  }
  utils.forEach(headers.split('\n'), function parser(line) {
    i = line.indexOf(':');
    key = utils.trim(line.substr(0, i)).toLowerCase();
    val = utils.trim(line.substr(i + 1));
    if (key) {
      if (parsed[key] && ignoreDuplicateOf.indexOf(key) >= 0) {
        return;
      }
      if (key === 'set-cookie') {
        parsed[key] = (parsed[key] ? parsed[key] : []).concat([val]);
      } else {
        parsed[key] = parsed[key] ? parsed[key] + ', ' + val : val;
      }
    }
  });
  return parsed;
};

var isURLSameOrigin = utils.isStandardBrowserEnv() ?
function standardBrowserEnv() {
  var msie = /(msie|trident)/i.test(navigator.userAgent);
  var urlParsingNode = document.createElement('a');
  var originURL;

  function resolveURL(url) {
    var href = url;
    if (msie) {
      urlParsingNode.setAttribute('href', href);
      href = urlParsingNode.href;
    }
    urlParsingNode.setAttribute('href', href);

    return {
      href: urlParsingNode.href,
      protocol: urlParsingNode.protocol ? urlParsingNode.protocol.replace(/:$/, '') : '',
      host: urlParsingNode.host,
      search: urlParsingNode.search ? urlParsingNode.search.replace(/^\?/, '') : '',
      hash: urlParsingNode.hash ? urlParsingNode.hash.replace(/^#/, '') : '',
      hostname: urlParsingNode.hostname,
      port: urlParsingNode.port,
      pathname: urlParsingNode.pathname.charAt(0) === '/' ? urlParsingNode.pathname : '/' + urlParsingNode.pathname
    };
  }
  originURL = resolveURL(window.location.href);

  return function isURLSameOrigin(requestURL) {
    var parsed = utils.isString(requestURL) ? resolveURL(requestURL) : requestURL;
    return parsed.protocol === originURL.protocol && parsed.host === originURL.host;
  };
}() :
function nonStandardBrowserEnv() {
  return function isURLSameOrigin() {
    return true;
  };
}();

var xhr = function xhrAdapter(config) {
  return new Promise(function dispatchXhrRequest(resolve, reject) {
    var requestData = config.data;
    var requestHeaders = config.headers;
    if (utils.isFormData(requestData)) {
      delete requestHeaders['Content-Type'];
    }

    var request = new XMLHttpRequest();

    if (config.auth) {
      var username = config.auth.username || '';
      var password = config.auth.password ? unescape(encodeURIComponent(config.auth.password)) : '';
      requestHeaders.Authorization = 'Basic ' + btoa(username + ':' + password);
    }
    var fullPath = buildFullPath(config.baseURL, config.url);
    request.open(config.method.toUpperCase(), buildURL(fullPath, config.params, config.paramsSerializer), true);

    request.timeout = config.timeout;

    request.onreadystatechange = function handleLoad() {
      if (!request || request.readyState !== 4) {
        return;
      }

      if (request.status === 0 && !(request.responseURL && request.responseURL.indexOf('file:') === 0)) {
        return;
      }

      var responseHeaders = 'getAllResponseHeaders' in request ? parseHeaders(request.getAllResponseHeaders()) : null;
      var responseData = !config.responseType || config.responseType === 'text' ? request.responseText : request.response;
      var response = {
        data: responseData,
        status: request.status,
        statusText: request.statusText,
        headers: responseHeaders,
        config: config,
        request: request
      };
      settle(resolve, reject, response);

      request = null;
    };

    request.onabort = function handleAbort() {
      if (!request) {
        return;
      }
      reject(createError('Request aborted', config, 'ECONNABORTED', request));

      request = null;
    };

    request.onerror = function handleError() {
      reject(createError('Network Error', config, null, request));

      request = null;
    };

    request.ontimeout = function handleTimeout() {
      var timeoutErrorMessage = 'timeout of ' + config.timeout + 'ms exceeded';
      if (config.timeoutErrorMessage) {
        timeoutErrorMessage = config.timeoutErrorMessage;
      }
      reject(createError(timeoutErrorMessage, config, 'ECONNABORTED', request));

      request = null;
    };

    if (utils.isStandardBrowserEnv()) {
      var xsrfValue = (config.withCredentials || isURLSameOrigin(fullPath)) && config.xsrfCookieName ? cookies.read(config.xsrfCookieName) : undefined;
      if (xsrfValue) {
        requestHeaders[config.xsrfHeaderName] = xsrfValue;
      }
    }

    if ('setRequestHeader' in request) {
      utils.forEach(requestHeaders, function setRequestHeader(val, key) {
        if (typeof requestData === 'undefined' && key.toLowerCase() === 'content-type') {
          delete requestHeaders[key];
        } else {
          request.setRequestHeader(key, val);
        }
      });
    }

    if (!utils.isUndefined(config.withCredentials)) {
      request.withCredentials = !!config.withCredentials;
    }

    if (config.responseType) {
      try {
        request.responseType = config.responseType;
      } catch (e) {
        if (config.responseType !== 'json') {
          throw e;
        }
      }
    }

    if (typeof config.onDownloadProgress === 'function') {
      request.addEventListener('progress', config.onDownloadProgress);
    }

    if (typeof config.onUploadProgress === 'function' && request.upload) {
      request.upload.addEventListener('progress', config.onUploadProgress);
    }
    if (config.cancelToken) {
      config.cancelToken.promise.then(function onCanceled(cancel) {
        if (!request) {
          return;
        }
        request.abort();
        reject(cancel);
        request = null;
      });
    }
    if (!requestData) {
      requestData = null;
    }

    request.send(requestData);
  });
};

var DEFAULT_CONTENT_TYPE = {
  'Content-Type': 'application/x-www-form-urlencoded'
};
function setContentTypeIfUnset(headers, value) {
  if (!utils.isUndefined(headers) && utils.isUndefined(headers['Content-Type'])) {
    headers['Content-Type'] = value;
  }
}
function getDefaultAdapter() {
  var adapter;
  if (typeof XMLHttpRequest !== 'undefined') {
    adapter = xhr;
  } else if (typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]') {
    adapter = xhr;
  }
  return adapter;
}
var defaults = {
  adapter: getDefaultAdapter(),
  transformRequest: [function transformRequest(data, headers) {
    normalizeHeaderName(headers, 'Accept');
    normalizeHeaderName(headers, 'Content-Type');
    if (utils.isFormData(data) || utils.isArrayBuffer(data) || utils.isBuffer(data) || utils.isStream(data) || utils.isFile(data) || utils.isBlob(data)) {
      return data;
    }
    if (utils.isArrayBufferView(data)) {
      return data.buffer;
    }
    if (utils.isURLSearchParams(data)) {
      setContentTypeIfUnset(headers, 'application/x-www-form-urlencoded;charset=utf-8');
      return data.toString();
    }
    if (utils.isObject(data)) {
      setContentTypeIfUnset(headers, 'application/json;charset=utf-8');
      return JSON.stringify(data);
    }
    return data;
  }],
  transformResponse: [function transformResponse(data) {
    if (typeof data === 'string') {
      try {
        data = JSON.parse(data);
      } catch (e) {}
    }
    return data;
  }],
  timeout: 0,
  xsrfCookieName: 'XSRF-TOKEN',
  xsrfHeaderName: 'X-XSRF-TOKEN',
  maxContentLength: -1,
  maxBodyLength: -1,
  validateStatus: function validateStatus(status) {
    return status >= 200 && status < 300;
  }
};
defaults.headers = {
  common: {
    'Accept': 'application/json, text/plain, */*'
  }
};
utils.forEach(['delete', 'get', 'head'], function forEachMethodNoData(method) {
  defaults.headers[method] = {};
});
utils.forEach(['post', 'put', 'patch'], function forEachMethodWithData(method) {
  defaults.headers[method] = utils.merge(DEFAULT_CONTENT_TYPE);
});
var defaults_1 = defaults;

function throwIfCancellationRequested(config) {
  if (config.cancelToken) {
    config.cancelToken.throwIfRequested();
  }
}

var dispatchRequest = function dispatchRequest(config) {
  throwIfCancellationRequested(config);

  config.headers = config.headers || {};

  config.data = transformData(config.data, config.headers, config.transformRequest);

  config.headers = utils.merge(config.headers.common || {}, config.headers[config.method] || {}, config.headers);
  utils.forEach(['delete', 'get', 'head', 'post', 'put', 'patch', 'common'], function cleanHeaderConfig(method) {
    delete config.headers[method];
  });
  var adapter = config.adapter || defaults_1.adapter;
  return adapter(config).then(function onAdapterResolution(response) {
    throwIfCancellationRequested(config);

    response.data = transformData(response.data, response.headers, config.transformResponse);
    return response;
  }, function onAdapterRejection(reason) {
    if (!isCancel(reason)) {
      throwIfCancellationRequested(config);

      if (reason && reason.response) {
        reason.response.data = transformData(reason.response.data, reason.response.headers, config.transformResponse);
      }
    }
    return Promise.reject(reason);
  });
};

var mergeConfig = function mergeConfig(config1, config2) {
  config2 = config2 || {};
  var config = {};
  var valueFromConfig2Keys = ['url', 'method', 'data'];
  var mergeDeepPropertiesKeys = ['headers', 'auth', 'proxy', 'params'];
  var defaultToConfig2Keys = ['baseURL', 'transformRequest', 'transformResponse', 'paramsSerializer', 'timeout', 'timeoutMessage', 'withCredentials', 'adapter', 'responseType', 'xsrfCookieName', 'xsrfHeaderName', 'onUploadProgress', 'onDownloadProgress', 'decompress', 'maxContentLength', 'maxBodyLength', 'maxRedirects', 'transport', 'httpAgent', 'httpsAgent', 'cancelToken', 'socketPath', 'responseEncoding'];
  var directMergeKeys = ['validateStatus'];
  function getMergedValue(target, source) {
    if (utils.isPlainObject(target) && utils.isPlainObject(source)) {
      return utils.merge(target, source);
    } else if (utils.isPlainObject(source)) {
      return utils.merge({}, source);
    } else if (utils.isArray(source)) {
      return source.slice();
    }
    return source;
  }
  function mergeDeepProperties(prop) {
    if (!utils.isUndefined(config2[prop])) {
      config[prop] = getMergedValue(config1[prop], config2[prop]);
    } else if (!utils.isUndefined(config1[prop])) {
      config[prop] = getMergedValue(undefined, config1[prop]);
    }
  }
  utils.forEach(valueFromConfig2Keys, function valueFromConfig2(prop) {
    if (!utils.isUndefined(config2[prop])) {
      config[prop] = getMergedValue(undefined, config2[prop]);
    }
  });
  utils.forEach(mergeDeepPropertiesKeys, mergeDeepProperties);
  utils.forEach(defaultToConfig2Keys, function defaultToConfig2(prop) {
    if (!utils.isUndefined(config2[prop])) {
      config[prop] = getMergedValue(undefined, config2[prop]);
    } else if (!utils.isUndefined(config1[prop])) {
      config[prop] = getMergedValue(undefined, config1[prop]);
    }
  });
  utils.forEach(directMergeKeys, function merge(prop) {
    if (prop in config2) {
      config[prop] = getMergedValue(config1[prop], config2[prop]);
    } else if (prop in config1) {
      config[prop] = getMergedValue(undefined, config1[prop]);
    }
  });
  var axiosKeys = valueFromConfig2Keys.concat(mergeDeepPropertiesKeys).concat(defaultToConfig2Keys).concat(directMergeKeys);
  var otherKeys = Object.keys(config1).concat(Object.keys(config2)).filter(function filterAxiosKeys(key) {
    return axiosKeys.indexOf(key) === -1;
  });
  utils.forEach(otherKeys, mergeDeepProperties);
  return config;
};

function Axios(instanceConfig) {
  this.defaults = instanceConfig;
  this.interceptors = {
    request: new InterceptorManager_1(),
    response: new InterceptorManager_1()
  };
}

Axios.prototype.request = function request(config) {
  if (typeof config === 'string') {
    config = arguments[1] || {};
    config.url = arguments[0];
  } else {
    config = config || {};
  }
  config = mergeConfig(this.defaults, config);

  if (config.method) {
    config.method = config.method.toLowerCase();
  } else if (this.defaults.method) {
    config.method = this.defaults.method.toLowerCase();
  } else {
    config.method = 'get';
  }

  var chain = [dispatchRequest, undefined];
  var promise = Promise.resolve(config);
  this.interceptors.request.forEach(function unshiftRequestInterceptors(interceptor) {
    chain.unshift(interceptor.fulfilled, interceptor.rejected);
  });
  this.interceptors.response.forEach(function pushResponseInterceptors(interceptor) {
    chain.push(interceptor.fulfilled, interceptor.rejected);
  });
  while (chain.length) {
    promise = promise.then(chain.shift(), chain.shift());
  }
  return promise;
};
Axios.prototype.getUri = function getUri(config) {
  config = mergeConfig(this.defaults, config);
  return buildURL(config.url, config.params, config.paramsSerializer).replace(/^\?/, '');
};

utils.forEach(['delete', 'get', 'head', 'options'], function forEachMethodNoData(method) {
  Axios.prototype[method] = function (url, config) {
    return this.request(mergeConfig(config || {}, {
      method: method,
      url: url,
      data: (config || {}).data
    }));
  };
});
utils.forEach(['post', 'put', 'patch'], function forEachMethodWithData(method) {
  Axios.prototype[method] = function (url, data, config) {
    return this.request(mergeConfig(config || {}, {
      method: method,
      url: url,
      data: data
    }));
  };
});
var Axios_1 = Axios;

function Cancel(message) {
  this.message = message;
}
Cancel.prototype.toString = function toString() {
  return 'Cancel' + (this.message ? ': ' + this.message : '');
};
Cancel.prototype.__CANCEL__ = true;
var Cancel_1 = Cancel;

function CancelToken(executor) {
  if (typeof executor !== 'function') {
    throw new TypeError('executor must be a function.');
  }
  var resolvePromise;
  this.promise = new Promise(function promiseExecutor(resolve) {
    resolvePromise = resolve;
  });
  var token = this;
  executor(function cancel(message) {
    if (token.reason) {
      return;
    }
    token.reason = new Cancel_1(message);
    resolvePromise(token.reason);
  });
}

CancelToken.prototype.throwIfRequested = function throwIfRequested() {
  if (this.reason) {
    throw this.reason;
  }
};

CancelToken.source = function source() {
  var cancel;
  var token = new CancelToken(function executor(c) {
    cancel = c;
  });
  return {
    token: token,
    cancel: cancel
  };
};
var CancelToken_1 = CancelToken;

var spread = function spread(callback) {
  return function wrap(arr) {
    return callback.apply(null, arr);
  };
};

var isAxiosError = function isAxiosError(payload) {
  return typeof payload === 'object' && payload.isAxiosError === true;
};

function createInstance(defaultConfig) {
  var context = new Axios_1(defaultConfig);
  var instance = bind(Axios_1.prototype.request, context);

  utils.extend(instance, Axios_1.prototype, context);

  utils.extend(instance, context);
  return instance;
}

var axios = createInstance(defaults_1);

axios.Axios = Axios_1;

axios.create = function create(instanceConfig) {
  return createInstance(mergeConfig(axios.defaults, instanceConfig));
};

axios.Cancel = Cancel_1;
axios.CancelToken = CancelToken_1;
axios.isCancel = isCancel;

axios.all = function all(promises) {
  return Promise.all(promises);
};
axios.spread = spread;

axios.isAxiosError = isAxiosError;
var axios_1 = axios;

var _default = axios;
axios_1.default = _default;

var stringReplaceAll = function stringReplaceAll(str, searcher, replaceWith) {
  if (str === void 0) {
    str = "";
  }
  if (searcher === void 0) {
    searcher = "";
  }
  if (replaceWith === void 0) {
    replaceWith = "";
  }
  if (searcher == "") return str;
  while (str.includes(searcher)) {
    str = str.replace(searcher, replaceWith);
  }
  return str;
};

var ChallanSearchResults = function ChallanSearchResults(_ref) {
  var template = _ref.template,
    header = _ref.header,
    actionButtonLabel = _ref.actionButtonLabel;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var _Digit$Hooks$useQuery = Digit.Hooks.useQueryParams(),
    mobileNumber = _Digit$Hooks$useQuery.mobileNumber,
    consumerNumber = _Digit$Hooks$useQuery.consumerNumber,
    oldconsumerNumber = _Digit$Hooks$useQuery.oldconsumerNumber,
    tenantId = _Digit$Hooks$useQuery.tenantId,
    propertyId = _Digit$Hooks$useQuery.propertyId,
    locality = _Digit$Hooks$useQuery.locality,
    doorNumber = _Digit$Hooks$useQuery.doorNumber,
    consumerName = _Digit$Hooks$useQuery.consumerName;
  var filters = {};
  if (mobileNumber) filters.mobileNumber = mobileNumber;
  if (consumerNumber) filters.connectionNumber = consumerNumber;
  if (oldconsumerNumber) filters.oldConnectionNumber = oldconsumerNumber;
  if (propertyId) filters.propertyId = propertyId;
  if (locality !== "undefined") filters.locality = locality;
  if (doorNumber) filters.doorNumber = doorNumber;
  if (consumerName) filters.consumerName = consumerName;
  filters = _extends({}, filters, {
    searchType: "CONNECTION"
  });
  var Waterresult = Digit.Hooks.ws.useWaterSearch({
    tenantId: tenantId,
    filters: _extends({}, filters),
    BusinessService: "WS",
    t: t
  });
  var Sewarageresult = Digit.Hooks.ws.useSewarageSearch({
    tenantId: tenantId,
    filters: _extends({}, filters),
    BusinessService: "SW",
    t: t
  });
  if (Waterresult !== null && Waterresult !== void 0 && Waterresult.isLoading || Sewarageresult !== null && Sewarageresult !== void 0 && Sewarageresult.isLoading || Waterresult == undefined || Sewarageresult == undefined) {
    return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null);
  }
  var onSubmit = function onSubmit(data) {
    history.push("/digit-ui/citizen/payment/my-bills/" + (data === null || data === void 0 ? void 0 : data.ConsumerNumber.split("/")[0]) + "/" + stringReplaceAll(data === null || data === void 0 ? void 0 : data.ConsumerNumber, "/", "+") + "?workflow=WNS&tenantId=" + tenantId);
  };
  var searchResults = Waterresult && Sewarageresult ? Waterresult.concat(Sewarageresult) : [];
  return /*#__PURE__*/React__default.createElement("div", {
    style: {
      marginTop: "16px"
    }
  }, /*#__PURE__*/React__default.createElement("div", null, header && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Header, {
    style: {
      marginLeft: "8px"
    }
  }, t(header), " (", searchResults === null || searchResults === void 0 ? void 0 : searchResults.length, ")"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.ResponseComposer, {
    data: searchResults,
    template: template,
    actionButtonLabel: actionButtonLabel,
    onSubmit: onSubmit
  })));
};
ChallanSearchResults.propTypes = {
  template: propTypes.any,
  header: propTypes.string,
  actionButtonLabel: propTypes.string
};
ChallanSearchResults.defaultProps = {
  template: [],
  header: null,
  actionButtonLabel: null
};

var CitizenSearchResults = function CitizenSearchResults() {
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var params = React.useMemo(function () {
    return config$1.map(function (step) {
      var texts = {};
      for (var key in step.texts) {
        texts[key] = t(step.texts[key]);
      }
      return _extends({}, step, {
        texts: texts
      });
    }, [config$1]);
  });
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: "" + path,
    exact: true
  }, /*#__PURE__*/React__default.createElement(ChallanSearchResults, {
    template: params[0].labels,
    header: params[0].texts.header,
    actionButtonLabel: params[0].texts.actionButtonLabel,
    t: t
  })));
};

var App = function App(_ref) {
  var path = _ref.path;
  var location = reactRouterDom.useLocation();
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.BackButton, {
    style: {
      border: "none"
    }
  }, t("CS_COMMON_BACK")), /*#__PURE__*/React__default.createElement(reactRouterDom.Switch, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.PrivateRoute, {
    path: path + "/create-application",
    component: WSCreate
  }), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/search",
    component: SearchConnection$1
  }), /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: path + "/search-results",
    component: CitizenSearchResults
  })));
};

var actions = [];
var getAction = function getAction(flow) {
  switch (flow) {
    default:
      actions = ['WS_COMMON_PROPERTY_DETAILS', 'WS_COMMON_CONNECTION_DETAIL', 'WS_COMMON_DOCUMENT_DETAILS', 'WS_COMMON_SUMMARY'];
  }
};
var Timeline = function Timeline(_ref) {
  var _ref$currentStep = _ref.currentStep,
    currentStep = _ref$currentStep === void 0 ? 1 : _ref$currentStep,
    _ref$flow = _ref.flow,
    flow = _ref$flow === void 0 ? "" : _ref$flow;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var isMobile = window.Digit.Utils.browser.isMobile();
  getAction(flow);
  return /*#__PURE__*/React__default.createElement("div", {
    className: "timeline-container",
    style: isMobile ? {} : {
      maxWidth: "960px",
      minWidth: "640px",
      marginRight: "auto"
    }
  }, actions.map(function (action, index, arr) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "timeline-checkpoint",
      key: index
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "timeline-content"
    }, /*#__PURE__*/React__default.createElement("span", {
      className: "circle " + (index <= currentStep - 1 && 'active')
    }, index < currentStep - 1 ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.TickMark, null) : index + 1), /*#__PURE__*/React__default.createElement("span", {
      className: "secondary-color"
    }, t(action))), index < arr.length - 1 && /*#__PURE__*/React__default.createElement("span", {
      className: "line " + (index < currentStep - 1 && 'active')
    }));
  }));
};

var WSServiceName = function WSServiceName(_ref) {
  var t = _ref.t,
    config = _ref.config,
    onSelect = _ref.onSelect,
    formData = _ref.formData;
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : formData.serviceName) || ""),
    serviceName = _useState[0],
    setServiceName = _useState[1];
  var _useState2 = React.useState(formData || {}),
    formDetails = _useState2[0];
  var serviceNameList = [{
    i18nKey: "WS_WATER_CONNECTION_ONLY",
    code: "WATER"
  }, {
    i18nKey: "WS_SEWERAGE_CONNECTION_ONLY",
    code: "SEWERAGE"
  }, {
    i18nKey: "WS_BOTH_WATER_AND_SEWERAGE",
    code: "BOTH"
  }];
  function onAdd() {}
  function onServiceNameSelect(value) {
    setServiceName(value);
  }
  var onSkip = function onSkip() {
    return onSelect();
  };
  var handleSubmit = function handleSubmit() {
    var _formDetails$serviceN;
    if ((serviceName === null || serviceName === void 0 ? void 0 : serviceName.code) == "WATER") sessionStorage.setItem("serviceName", "WATER");else if ((serviceName === null || serviceName === void 0 ? void 0 : serviceName.code) == "SEWERAGE") sessionStorage.setItem("serviceName", "SEWERAGE");else sessionStorage.setItem("serviceName", "");
    if ((serviceName === null || serviceName === void 0 ? void 0 : serviceName.code) != "BOTH" && (formDetails === null || formDetails === void 0 ? void 0 : (_formDetails$serviceN = formDetails.serviceName) === null || _formDetails$serviceN === void 0 ? void 0 : _formDetails$serviceN.code) != (serviceName === null || serviceName === void 0 ? void 0 : serviceName.code)) {
      var data = formDetails;
      data.waterConectionDetails = {};
      data.sewerageConnectionDetails = {};
      data.serviceName = serviceName;
      onSelect("", data);
    } else {
      onSelect(config.key, serviceName);
    }
  };
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 2
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    t: t,
    config: config,
    onSelect: handleSubmit,
    onSkip: onSkip,
    isDisabled: !serviceName,
    onAdd: onAdd
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_SELECT_SERVICE_TYPE_WANT_TO_APPLY"), "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioOrSelect, {
    name: "gender",
    options: serviceNameList,
    selectedOption: serviceName,
    optionKey: "i18nKey",
    onSelect: onServiceNameSelect,
    t: t
  })));
};

var WSWaterConnectionDetails = function WSWaterConnectionDetails(_ref) {
  var _formData$waterConect, _formData$waterConect2;
  var t = _ref.t,
    config = _ref.config,
    onSelect = _ref.onSelect,
    formData = _ref.formData;
  var tenantId = Digit.ULBService.getStateId();
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$waterConect = formData.waterConectionDetails) === null || _formData$waterConect === void 0 ? void 0 : _formData$waterConect.proposedTaps) || ""),
    proposedTaps = _useState[0],
    setProposedTaps = _useState[1];
  var _useState2 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$waterConect2 = formData.waterConectionDetails) === null || _formData$waterConect2 === void 0 ? void 0 : _formData$waterConect2.proposedPipeSize) || ""),
    proposedPipeSize = _useState2[0],
    setProposedPipeSize = _useState2[1];
  var _useState3 = React.useState([]),
    proposedPipeSizeList = _useState3[0],
    setProposedPipeSizeList = _useState3[1];
  var validation = {};
  var _Digit$Hooks$ws$WSSea = Digit.Hooks.ws.WSSearchMdmsTypes.useWSServicesCalculation(tenantId),
    wsServiceCalculationLoading = _Digit$Hooks$ws$WSSea.isLoading,
    wsServiceCalculation = _Digit$Hooks$ws$WSSea.data;
  React.useEffect(function () {
    var _wsServiceCalculation;
    if ((wsServiceCalculation === null || wsServiceCalculation === void 0 ? void 0 : (_wsServiceCalculation = wsServiceCalculation.PipeSize) === null || _wsServiceCalculation === void 0 ? void 0 : _wsServiceCalculation.length) > 0) {
      var pipeLists = [];
      wsServiceCalculation === null || wsServiceCalculation === void 0 ? void 0 : wsServiceCalculation.PipeSize.forEach(function (type) {
        pipeLists.push({
          i18nKey: type.size + " " + t("WS_INCHES_LABEL"),
          code: type.size,
          id: type.id,
          size: type.size
        });
      });
      setProposedPipeSizeList(pipeLists);
    }
  }, [wsServiceCalculation]);
  function onAdd() {}
  function setNumberOfProposedTaps(e) {
    setProposedTaps(e.target.value);
  }
  function setProposedPipeSizeSelect(value) {
    setProposedPipeSize(value);
  }
  var onSkip = function onSkip() {
    return onSelect();
  };
  var handleSubmit = function handleSubmit() {
    var details = {};
    details.proposedTaps = proposedTaps;
    details.proposedPipeSize = proposedPipeSize;
    onSelect(config.key, details);
  };
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 2
  }), !wsServiceCalculationLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    t: t,
    config: config,
    onSelect: handleSubmit,
    onSkip: onSkip,
    isDisabled: !proposedTaps || !proposedPipeSize,
    onAdd: onAdd
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_NO_OF_TAPS_PROPOSED"), "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    isMandatory: false,
    optionKey: "i18nKey",
    t: t,
    name: "proposedTaps",
    onChange: setNumberOfProposedTaps,
    value: proposedTaps
  }, validation = {
    isRequired: true,
    pattern: "^[0-9]*$",
    title: t("ERR_DEFAULT_INPUT_FIELD_MSG"),
    type: "text"
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_PROPOSED_PIPE_SIZE"), "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioOrSelect, {
    name: "proposedPipeSize",
    options: proposedPipeSizeList,
    selectedOption: proposedPipeSize,
    optionKey: "i18nKey",
    onSelect: setProposedPipeSizeSelect,
    t: t
  })) : /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null));
};

var WSDocsRequired = function WSDocsRequired(_ref) {
  var _wsDocs$Documents;
  var onSelect = _ref.onSelect;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var tenantId = Digit.ULBService.getStateId();
  var goNext = function goNext() {
    onSelect("DocsReq", "");
  };
  var _Digit$Hooks$ws$WSSea = Digit.Hooks.ws.WSSearchMdmsTypes.useWSServicesMasters(tenantId),
    wsDocsLoading = _Digit$Hooks$ws$WSSea.isLoading,
    wsDocs = _Digit$Hooks$ws$WSSea.data;
  return /*#__PURE__*/React__default.createElement(React.Fragment, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Card, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("WS_COMMON_APPL_NEW_CONNECTION")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CitizenInfoLabel, {
    style: {
      margin: "0px"
    },
    textStyle: {
      color: "#0B0C0C"
    },
    text: t("OBPS_DOCS_REQUIRED_TIME"),
    showInfo: false
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardText, {
    style: {
      color: "#0B0C0C",
      marginTop: "12px"
    }
  }, t("WS_NEW_CONNECTION_TEST_1")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardText, {
    style: {
      color: "#0B0C0C",
      marginTop: "12px"
    }
  }, t("WS_NEW_CONNECTION_TEST_2")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardSubHeader, null, t("WS_DOC_REQ_SCREEN_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardText, {
    style: {
      color: "#0B0C0C",
      marginTop: "12px"
    }
  }, t("WS_NEW_CONNECTION_TEST_3")), wsDocsLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null) : /*#__PURE__*/React__default.createElement(React.Fragment, null, wsDocs === null || wsDocs === void 0 ? void 0 : (_wsDocs$Documents = wsDocs.Documents) === null || _wsDocs$Documents === void 0 ? void 0 : _wsDocs$Documents.map(function (doc, index) {
    var _doc$dropdownData;
    return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("div", {
      style: {
        fontWeight: 700,
        marginBottom: "8px"
      },
      key: index
    }, /*#__PURE__*/React__default.createElement("div", {
      style: {
        display: "flex"
      }
    }, /*#__PURE__*/React__default.createElement("div", null, index + 1 + ".", "\xA0"), /*#__PURE__*/React__default.createElement("div", null, " " + t(doc === null || doc === void 0 ? void 0 : doc.code.replace('.', '_'))))), /*#__PURE__*/React__default.createElement("div", {
      style: {
        marginBottom: "16px",
        marginLeft: "18px"
      }
    }, doc === null || doc === void 0 ? void 0 : (_doc$dropdownData = doc.dropdownData) === null || _doc$dropdownData === void 0 ? void 0 : _doc$dropdownData.map(function (value, index) {
      var _doc$dropdownData2;
      return (doc === null || doc === void 0 ? void 0 : (_doc$dropdownData2 = doc.dropdownData) === null || _doc$dropdownData2 === void 0 ? void 0 : _doc$dropdownData2.length) !== index + 1 ? /*#__PURE__*/React__default.createElement("span", null, t(value === null || value === void 0 ? void 0 : value.i18nKey) + ", ") : /*#__PURE__*/React__default.createElement("span", null, "" + t(value === null || value === void 0 ? void 0 : value.i18nKey));
    })));
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.SubmitBar, {
    label: t("CS_COMMON_NEXT"),
    onSubmit: goNext
  })));
};

var _iteratorSymbol = /*#__PURE__*/typeof Symbol !== "undefined" ? Symbol.iterator || (Symbol.iterator = Symbol("Symbol.iterator")) : "@@iterator";
var _asyncIteratorSymbol = /*#__PURE__*/typeof Symbol !== "undefined" ? Symbol.asyncIterator || (Symbol.asyncIterator = Symbol("Symbol.asyncIterator")) : "@@asyncIterator";

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }
  if (result && result.then) {
    return result.then(void 0, recover);
  }
  return result;
}

var WSDocumentDetails = function WSDocumentDetails(_ref) {
  var _formData$documents, _wsDocs$Documents;
  var t = _ref.t,
    config = _ref.config,
    onSelect = _ref.onSelect,
    formData = _ref.formData;
  var tenantId = Digit.ULBService.getStateId();
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$documents = formData.documents) === null || _formData$documents === void 0 ? void 0 : _formData$documents.documents) || []),
    documents = _useState[0],
    setDocuments = _useState[1];
  var _useState2 = React.useState(null),
    error = _useState2[0],
    setError = _useState2[1];
  var _useState3 = React.useState(true),
    enableSubmit = _useState3[0],
    setEnableSubmit = _useState3[1];
  var _useState4 = React.useState(false),
    checkRequiredFields = _useState4[0],
    setCheckRequiredFields = _useState4[1];
  var _Digit$Hooks$ws$WSSea = Digit.Hooks.ws.WSSearchMdmsTypes.useWSServicesMasters(tenantId),
    wsDocsLoading = _Digit$Hooks$ws$WSSea.isLoading,
    wsDocs = _Digit$Hooks$ws$WSSea.data;
  var handleSubmit = function handleSubmit() {
    var document = formData.documents;
    var documentStep;
    documentStep = _extends({}, document, {
      documents: documents
    });
    onSelect(config.key, documentStep);
  };
  var onSkip = function onSkip() {
    return onSelect();
  };
  function onAdd() {}
  React.useEffect(function () {
    var count = 0;
    wsDocs === null || wsDocs === void 0 ? void 0 : wsDocs.Documents.map(function (doc) {
      var isRequired = false;
      documents.map(function (data) {
        if (doc.required && data !== null && data !== void 0 && data.documentType.includes(doc.code)) isRequired = true;
      });
      if (!isRequired && doc.required) count = count + 1;
    });
    if ((count == "0" || count == 0) && documents.length > 0) setEnableSubmit(false);else setEnableSubmit(true);
  }, [documents, checkRequiredFields]);
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 3
  }), !wsDocsLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    t: t,
    config: config,
    onSelect: handleSubmit,
    onSkip: onSkip,
    isDisabled: enableSubmit,
    onAdd: onAdd
  }, wsDocs === null || wsDocs === void 0 ? void 0 : (_wsDocs$Documents = wsDocs.Documents) === null || _wsDocs$Documents === void 0 ? void 0 : _wsDocs$Documents.map(function (document, index) {
    return /*#__PURE__*/React__default.createElement(SelectDocument, {
      key: index,
      document: document,
      t: t,
      error: error,
      setError: setError,
      setDocuments: setDocuments,
      documents: documents,
      setCheckRequiredFields: setCheckRequiredFields
    });
  }), error && /*#__PURE__*/React__default.createElement(digitUiReactComponents.Toast, {
    label: error,
    onClose: function onClose() {
      return setError(null);
    },
    error: true
  })) : /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null));
};
function SelectDocument(_ref2) {
  var _doc$dropdownData;
  var t = _ref2.t,
    key = _ref2.key,
    doc = _ref2.document,
    setDocuments = _ref2.setDocuments,
    error = _ref2.error,
    setError = _ref2.setError,
    documents = _ref2.documents,
    setCheckRequiredFields = _ref2.setCheckRequiredFields;
  var filteredDocument = documents === null || documents === void 0 ? void 0 : documents.filter(function (item) {
    var _item$documentType;
    return item === null || item === void 0 ? void 0 : (_item$documentType = item.documentType) === null || _item$documentType === void 0 ? void 0 : _item$documentType.includes(doc === null || doc === void 0 ? void 0 : doc.code);
  })[0];
  var tenantId = Digit.ULBService.getCurrentTenantId();
  var _useState5 = React.useState(filteredDocument ? _extends({}, filteredDocument, {
      active: true,
      code: filteredDocument === null || filteredDocument === void 0 ? void 0 : filteredDocument.documentType,
      i18nKey: filteredDocument === null || filteredDocument === void 0 ? void 0 : filteredDocument.documentType
    }) : (doc === null || doc === void 0 ? void 0 : (_doc$dropdownData = doc.dropdownData) === null || _doc$dropdownData === void 0 ? void 0 : _doc$dropdownData.length) === 1 ? doc === null || doc === void 0 ? void 0 : doc.dropdownData[0] : {}),
    selectedDocument = _useState5[0],
    setSelectedDocument = _useState5[1];
  var _useState6 = React.useState(null),
    file = _useState6[0],
    setFile = _useState6[1];
  var _useState7 = React.useState(function () {
      return (filteredDocument === null || filteredDocument === void 0 ? void 0 : filteredDocument.fileStoreId) || null;
    }),
    uploadedFile = _useState7[0],
    setUploadedFile = _useState7[1];
  var handleSelectDocument = function handleSelectDocument(value) {
    return setSelectedDocument(value);
  };
  function selectfile(e) {
    setFile(e.target.files[0]);
  }
  React.useEffect(function () {
    if (selectedDocument !== null && selectedDocument !== void 0 && selectedDocument.code) {
      setDocuments(function (prev) {
        var filteredDocumentsByDocumentType = prev === null || prev === void 0 ? void 0 : prev.filter(function (item) {
          return (item === null || item === void 0 ? void 0 : item.documentType) !== (selectedDocument === null || selectedDocument === void 0 ? void 0 : selectedDocument.code);
        });
        if ((uploadedFile === null || uploadedFile === void 0 ? void 0 : uploadedFile.length) === 0 || uploadedFile === null) return filteredDocumentsByDocumentType;
        var filteredDocumentsByFileStoreId = filteredDocumentsByDocumentType === null || filteredDocumentsByDocumentType === void 0 ? void 0 : filteredDocumentsByDocumentType.filter(function (item) {
          return (item === null || item === void 0 ? void 0 : item.fileStoreId) !== uploadedFile;
        });
        return [].concat(filteredDocumentsByFileStoreId, [{
          documentType: selectedDocument === null || selectedDocument === void 0 ? void 0 : selectedDocument.code,
          fileStoreId: uploadedFile,
          documentUid: uploadedFile,
          fileName: (file === null || file === void 0 ? void 0 : file.name) || ""
        }]);
      });
    }
  }, [uploadedFile, selectedDocument]);
  React.useEffect(function () {
    (function () {
      try {
        setError(null);
        var _temp4 = function () {
          if (file) {
            var _temp5 = function () {
              if (file.size >= 5242880) {
                setError(t("CS_MAXIMUM_UPLOAD_SIZE_EXCEEDED"));
              } else {
                var _temp6 = _catch(function () {
                  setUploadedFile(null);
                  return Promise.resolve(Digit.UploadServices.Filestorage("WS", file, tenantId === null || tenantId === void 0 ? void 0 : tenantId.split(".")[0])).then(function (response) {
                    var _response$data, _response$data$files;
                    if ((response === null || response === void 0 ? void 0 : (_response$data = response.data) === null || _response$data === void 0 ? void 0 : (_response$data$files = _response$data.files) === null || _response$data$files === void 0 ? void 0 : _response$data$files.length) > 0) {
                      var _response$data2, _response$data2$files;
                      setUploadedFile(response === null || response === void 0 ? void 0 : (_response$data2 = response.data) === null || _response$data2 === void 0 ? void 0 : (_response$data2$files = _response$data2.files[0]) === null || _response$data2$files === void 0 ? void 0 : _response$data2$files.fileStoreId);
                    } else {
                      setError(t("CS_FILE_UPLOAD_ERROR"));
                    }
                  });
                }, function () {
                  setError(t("CS_FILE_UPLOAD_ERROR"));
                });
                if (_temp6 && _temp6.then) return _temp6.then(function () {});
              }
            }();
            if (_temp5 && _temp5.then) return _temp5.then(function () {});
          }
        }();
        return _temp4 && _temp4.then ? _temp4.then(function () {}) : void 0;
      } catch (e) {
        Promise.reject(e);
      }
    })();
  }, [file]);
  return /*#__PURE__*/React__default.createElement("div", {
    style: {
      marginBottom: "24px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, doc !== null && doc !== void 0 && doc.required ? t(doc === null || doc === void 0 ? void 0 : doc.i18nKey) + " *" : "" + t(doc === null || doc === void 0 ? void 0 : doc.i18nKey)), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    t: t,
    isMandatory: false,
    option: doc === null || doc === void 0 ? void 0 : doc.dropdownData,
    selected: selectedDocument,
    optionKey: "i18nKey",
    select: handleSelectDocument
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.UploadFile, {
    id: "noc-doc-" + key,
    extraStyleName: "propertyCreate",
    accept: ".jpg,.png,.pdf",
    onUpload: selectfile,
    onDelete: function onDelete() {
      setUploadedFile(null);
      setCheckRequiredFields(true);
    },
    message: uploadedFile ? "1 " + t("CS_ACTION_FILEUPLOADED") : t("ES_NO_FILE_SELECTED_LABEL"),
    error: error
  }));
}

var WSSewerageConnectionDetails = function WSSewerageConnectionDetails(_ref) {
  var _formData$sewerageCon, _formData$sewerageCon2;
  var t = _ref.t,
    config = _ref.config,
    onSelect = _ref.onSelect,
    formData = _ref.formData;
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$sewerageCon = formData.sewerageConnectionDetails) === null || _formData$sewerageCon === void 0 ? void 0 : _formData$sewerageCon.proposedWaterClosets) || ""),
    proposedWaterClosets = _useState[0],
    setProposedWaterClosets = _useState[1];
  var _useState2 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$sewerageCon2 = formData.sewerageConnectionDetails) === null || _formData$sewerageCon2 === void 0 ? void 0 : _formData$sewerageCon2.proposedToilets) || ""),
    proposedToilets = _useState2[0],
    setProposedToilets = _useState2[1];
  var validation = {};
  function onAdd() {}
  function setNumberOfProposedWaterClosets(e) {
    setProposedWaterClosets(e.target.value);
  }
  function setNumberOfProposedToilets(e) {
    setProposedToilets(e.target.value);
  }
  var onSkip = function onSkip() {
    return onSelect();
  };
  var handleSubmit = function handleSubmit() {
    var details = {};
    details.proposedWaterClosets = proposedWaterClosets;
    details.proposedToilets = proposedToilets;
    onSelect(config.key, details);
  };
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 2
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    t: t,
    config: config,
    onSelect: handleSubmit,
    onSkip: onSkip,
    isDisabled: !proposedWaterClosets || !proposedToilets,
    onAdd: onAdd
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_NO_OF_WATER_CLOSETS"), "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    type: "number",
    isMandatory: false,
    optionKey: "i18nKey",
    t: t,
    name: "proposedWaterClosets",
    onChange: setNumberOfProposedWaterClosets,
    value: proposedWaterClosets
  }, validation = {
    isRequired: true,
    pattern: "^[0-9]*$",
    title: t("ERR_DEFAULT_INPUT_FIELD_MSG")
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_SERV_DETAIL_NO_OF_TOILETS"), "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    type: "number",
    isMandatory: false,
    optionKey: "i18nKey",
    t: t,
    name: "proposedToilets",
    onChange: setNumberOfProposedToilets,
    value: proposedToilets
  }, validation = {
    isRequired: true,
    pattern: "^[0-9]*$",
    title: t("ERR_DEFAULT_INPUT_FIELD_MSG")
  }))));
};

var WSPlumberPreference = function WSPlumberPreference(_ref) {
  var t = _ref.t,
    config = _ref.config,
    onSelect = _ref.onSelect,
    formData = _ref.formData;
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : formData.plumberPreference) || ""),
    plumberPreference = _useState[0],
    setPlumberPreference = _useState[1];
  var plumberPreferenceList = [{
    i18nKey: "WS_I_WOULD_PREFER_FROM_MUNICIPAL_OFFICE",
    code: "ULB"
  }, {
    i18nKey: "WS_I_WILL_ARRAGE_THE_PLUMBER_MYSELF",
    code: "Self"
  }];
  function onAdd() {}
  function onPlumberPreferenceSelect(value) {
    setPlumberPreference(value);
  }
  var onSkip = function onSkip() {
    return onSelect();
  };
  var handleSubmit = function handleSubmit() {
    onSelect(config.key, plumberPreference);
  };
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 2
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    t: t,
    config: config,
    onSelect: handleSubmit,
    onSkip: onSkip,
    isDisabled: !plumberPreference,
    onAdd: onAdd
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioOrSelect, {
    name: "plumberPreference",
    options: plumberPreferenceList,
    selectedOption: plumberPreference,
    optionKey: "i18nKey",
    onSelect: onPlumberPreferenceSelect,
    t: t
  })));
};

var ConnectionHolder = function ConnectionHolder(_ref) {
  var _formData$ConnectionH, _formData$formData, _formData$formData$Co, _formData$ConnectionH2, _formData$formData2, _formData$formData2$C, _formData$ConnectionH3, _formData$formData3, _formData$formData3$C, _formData$ConnectionH4, _formData$formData4, _formData$formData4$C, _formData$ConnectionH5, _formData$formData5, _formData$formData5$C, _formData$ConnectionH6, _formData$formData6, _formData$formData6$C, _formData$ConnectionH7, _formData$formData7, _formData$formData7$C, _formData$ConnectionH8, _formData$formData8, _formData$formData8$C, _formData$config$key, _formData$ConnectionH9, _formData$ConnectionH10, _Documentsob$Property;
  var t = _ref.t,
    config = _ref.config,
    onSelect = _ref.onSelect,
    formData = _ref.formData;
  var validation = {};
  var _useState = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH = formData.ConnectionHolderDetails) === null || _formData$ConnectionH === void 0 ? void 0 : _formData$ConnectionH.name) || (formData === null || formData === void 0 ? void 0 : (_formData$formData = formData.formData) === null || _formData$formData === void 0 ? void 0 : (_formData$formData$Co = _formData$formData.ConnectionHolderDetails) === null || _formData$formData$Co === void 0 ? void 0 : _formData$formData$Co.name) || ""),
    name = _useState[0],
    setName = _useState[1];
  var _useState2 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH2 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH2 === void 0 ? void 0 : _formData$ConnectionH2.guardian) || (formData === null || formData === void 0 ? void 0 : (_formData$formData2 = formData.formData) === null || _formData$formData2 === void 0 ? void 0 : (_formData$formData2$C = _formData$formData2.ConnectionHolderDetails) === null || _formData$formData2$C === void 0 ? void 0 : _formData$formData2$C.guardian) || ""),
    guardian = _useState2[0],
    setguardian = _useState2[1];
  var _useState3 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH3 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH3 === void 0 ? void 0 : _formData$ConnectionH3.gender) || (formData === null || formData === void 0 ? void 0 : (_formData$formData3 = formData.formData) === null || _formData$formData3 === void 0 ? void 0 : (_formData$formData3$C = _formData$formData3.ConnectionHolderDetails) === null || _formData$formData3$C === void 0 ? void 0 : _formData$formData3$C.gender)),
    gender = _useState3[0],
    setGender = _useState3[1];
  var _useState4 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH4 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH4 === void 0 ? void 0 : _formData$ConnectionH4.relationship) || (formData === null || formData === void 0 ? void 0 : (_formData$formData4 = formData.formData) === null || _formData$formData4 === void 0 ? void 0 : (_formData$formData4$C = _formData$formData4.ConnectionHolderDetails) === null || _formData$formData4$C === void 0 ? void 0 : _formData$formData4$C.relationship)),
    relationship = _useState4[0],
    setrelationship = _useState4[1];
  var _useState5 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH5 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH5 === void 0 ? void 0 : _formData$ConnectionH5.mobileNumber) || (formData === null || formData === void 0 ? void 0 : (_formData$formData5 = formData.formData) === null || _formData$formData5 === void 0 ? void 0 : (_formData$formData5$C = _formData$formData5.ConnectionHolderDetails) === null || _formData$formData5$C === void 0 ? void 0 : _formData$formData5$C.mobileNumber) || ""),
    mobileNumber = _useState5[0],
    setMobileNumber = _useState5[1];
  var _useState6 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH6 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH6 === void 0 ? void 0 : _formData$ConnectionH6.address) || (formData === null || formData === void 0 ? void 0 : (_formData$formData6 = formData.formData) === null || _formData$formData6 === void 0 ? void 0 : (_formData$formData6$C = _formData$formData6.ConnectionHolderDetails) === null || _formData$formData6$C === void 0 ? void 0 : _formData$formData6$C.address) || ""),
    address = _useState6[0],
    setaddress = _useState6[1];
  var _useState7 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH7 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH7 === void 0 ? void 0 : _formData$ConnectionH7.documentId) || (formData === null || formData === void 0 ? void 0 : (_formData$formData7 = formData.formData) === null || _formData$formData7 === void 0 ? void 0 : (_formData$formData7$C = _formData$formData7.ConnectionHolderDetails) === null || _formData$formData7$C === void 0 ? void 0 : _formData$formData7$C.documentId) || ""),
    documentId = _useState7[0],
    setdocumentId = _useState7[1];
  var _useState8 = React.useState(!(formData !== null && formData !== void 0 && (_formData$ConnectionH8 = formData.ConnectionHolderDetails) !== null && _formData$ConnectionH8 !== void 0 && _formData$ConnectionH8.isOwnerSame || formData !== null && formData !== void 0 && (_formData$formData8 = formData.formData) !== null && _formData$formData8 !== void 0 && (_formData$formData8$C = _formData$formData8.ConnectionHolderDetails) !== null && _formData$formData8$C !== void 0 && _formData$formData8$C.isOwnerSame) ? false : true),
    isOwnerSame = _useState8[0],
    setisOwnerSame = _useState8[1];
  var _useState9 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$config$key = formData[config.key]) === null || _formData$config$key === void 0 ? void 0 : _formData$config$key.fileStoreId) || null),
    uploadedFile = _useState9[0],
    setUploadedFile = _useState9[1];
  var _useState10 = React.useState(null),
    file = _useState10[0],
    setFile = _useState10[1];
  var _useState11 = React.useState(null),
    error = _useState11[0],
    setError = _useState11[1];
  var _useState12 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH9 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH9 === void 0 ? void 0 : _formData$ConnectionH9.documentType) || ""),
    dropdownValue = _useState12[0],
    setDropdownValue = _useState12[1];
  var _useState13 = React.useState((formData === null || formData === void 0 ? void 0 : (_formData$ConnectionH10 = formData.ConnectionHolderDetails) === null || _formData$ConnectionH10 === void 0 ? void 0 : _formData$ConnectionH10.specialCategoryType) || {}),
    ownerType = _useState13[0],
    setOwnerType = _useState13[1];
  var tenantId = Digit.ULBService.getCurrentTenantId();
  var stateId = Digit.ULBService.getStateId();
  var dropdownData = [];
  var _Digit$Hooks$pt$usePr = Digit.Hooks.pt.usePropertyMDMS(stateId, "PropertyTax", "Documents"),
    _Digit$Hooks$pt$usePr2 = _Digit$Hooks$pt$usePr.data,
    Documentsob = _Digit$Hooks$pt$usePr2 === void 0 ? {} : _Digit$Hooks$pt$usePr2;
  var docs = Documentsob === null || Documentsob === void 0 ? void 0 : (_Documentsob$Property = Documentsob.PropertyTax) === null || _Documentsob$Property === void 0 ? void 0 : _Documentsob$Property.Documents;
  var specialProofIdentity = Array.isArray(docs) && docs.filter(function (doc) {
    return doc.code.includes("SPECIALCATEGORYPROOF");
  });
  if (specialProofIdentity.length > 0) {
    var _specialProofIdentity, _dropdownData;
    dropdownData = (_specialProofIdentity = specialProofIdentity[0]) === null || _specialProofIdentity === void 0 ? void 0 : _specialProofIdentity.dropdownData;
    dropdownData.forEach(function (data) {
      data.i18nKey = stringReplaceAll(data.code, ".", "_");
    });
    dropdownData = (_dropdownData = dropdownData) === null || _dropdownData === void 0 ? void 0 : _dropdownData.filter(function (dropdown) {
      return dropdown.parentValue.includes(ownerType === null || ownerType === void 0 ? void 0 : ownerType.code);
    });
    if (dropdownData.length == 1 && dropdownValue != dropdownData[0]) {
      setTypeOfDropdownValue(dropdownData[0]);
    }
  }
  var GuardianOptions = [{
    name: "HUSBAND",
    code: "HUSBAND",
    i18nKey: "COMMON_MASTERS_OWNERTYPE_HUSBAND"
  }, {
    name: "Father",
    code: "FATHER",
    i18nKey: "COMMON_MASTERS_OWNERTYPE_FATHER"
  }];
  var _Digit$Hooks$obps$use = Digit.Hooks.obps.useMDMS(stateId, "common-masters", ["GenderType"]),
    isLoading = _Digit$Hooks$obps$use.isLoading,
    genderTypeData = _Digit$Hooks$obps$use.data;
  var _Digit$Hooks$pt$usePr3 = Digit.Hooks.pt.usePropertyMDMS(stateId, "PropertyTax", "OwnerType"),
    Menu = _Digit$Hooks$pt$usePr3.data;
  Menu ? Menu.sort(function (a, b) {
    return a.name.localeCompare(b.name);
  }) : "";
  var menu = [];
  genderTypeData && genderTypeData["common-masters"].GenderType.filter(function (data) {
    return data.active;
  }).map(function (genderDetails) {
    menu.push({
      i18nKey: "COMMON_GENDER_" + genderDetails.code,
      code: "" + genderDetails.code,
      value: "" + genderDetails.code
    });
  });
  React.useEffect(function () {
    (function () {
      try {
        setError(null);
        var _temp4 = function () {
          if (file) {
            var _temp5 = function () {
              if (file.size >= 2000000) {
                setError(t("PT_MAXIMUM_UPLOAD_SIZE_EXCEEDED"));
              } else {
                var _temp6 = _catch(function () {
                  return Promise.resolve(Digit.UploadServices.Filestorage("property-upload", file, Digit.ULBService.getStateId())).then(function (response) {
                    var _response$data, _response$data$files;
                    if ((response === null || response === void 0 ? void 0 : (_response$data = response.data) === null || _response$data === void 0 ? void 0 : (_response$data$files = _response$data.files) === null || _response$data$files === void 0 ? void 0 : _response$data$files.length) > 0) {
                      var _response$data2, _response$data2$files;
                      setUploadedFile(response === null || response === void 0 ? void 0 : (_response$data2 = response.data) === null || _response$data2 === void 0 ? void 0 : (_response$data2$files = _response$data2.files[0]) === null || _response$data2$files === void 0 ? void 0 : _response$data2$files.fileStoreId);
                    } else {
                      setError(t("PT_FILE_UPLOAD_ERROR"));
                    }
                  });
                }, function () {});
                if (_temp6 && _temp6.then) return _temp6.then(function () {});
              }
            }();
            if (_temp5 && _temp5.then) return _temp5.then(function () {});
          }
        }();
        return _temp4 && _temp4.then ? _temp4.then(function () {}) : void 0;
      } catch (e) {
        Promise.reject(e);
      }
    })();
  }, [file]);
  function selectChecked(e) {
    setisOwnerSame(!isOwnerSame);
  }
  function setTypeOfDropdownValue(dropdownValue) {
    setDropdownValue(dropdownValue);
  }
  function SelectName(e) {
    setName(e.target.value);
  }
  function selectguardian(e) {
    setguardian(e.target.value);
  }
  function setGenderName(value) {
    setGender(value);
  }
  function setRelationshipName(value) {
    setrelationship(value);
  }
  function setMobileNo(e) {
    setMobileNumber(e.target.value);
  }
  function selectaddress(e) {
    setaddress(e.target.value);
  }
  function selectdocumentId(e) {
    setdocumentId(e.target.value);
  }
  function setTypeOfOwner(value) {
    setOwnerType(value);
  }
  function selectfile(e) {
    setFile(e.target.files[0]);
  }
  var goNext = function goNext() {
    if (isOwnerSame == true) {
      onSelect(config.key, {
        "property ID": "PT-23-45-677903",
        isOwnerSame: isOwnerSame
      });
    } else {
      var ConnectionDet = {
        isOwnerSame: isOwnerSame,
        name: name,
        mobileNumber: mobileNumber,
        gender: gender,
        guardian: guardian,
        address: address,
        relationship: relationship,
        specialCategoryType: ownerType,
        documentId: documentId,
        fileStoreId: uploadedFile,
        documentType: dropdownValue
      };
      onSelect(config.key, ConnectionDet);
    }
  };
  var onSkip = function onSkip() {
    return onSelect();
  };
  return /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 2
  }), !isLoading ? /*#__PURE__*/React__default.createElement(digitUiReactComponents.FormStep, {
    config: config,
    onSelect: goNext,
    onSkip: onSkip,
    t: t,
    isDisabled: !isOwnerSame && (!name || !mobileNumber || !gender)
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CheckBox, {
    label: t("WS_CONN_HOLDER_SAME_AS_OWNER_DETAILS"),
    onChange: function onChange(e) {
      return selectChecked();
    }
    ,
    checked: isOwnerSame,
    style: {
      paddingBottom: "10px",
      paddingTop: "10px"
    }
  })), !isOwnerSame && /*#__PURE__*/React__default.createElement("div", {
    style: {
      border: "solid",
      borderRadius: "5px",
      padding: "10px",
      paddingTop: "20px",
      marginTop: "10px",
      borderColor: "#f3f3f3",
      background: "#FAFAFA"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_OWN_DETAIL_NAME") + "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    t: t,
    type: "text",
    style: {
      background: "#FAFAFA"
    },
    isMandatory: false,
    optionKey: "i18nKey",
    name: "name",
    value: name,
    onChange: SelectName
  }, validation = {
    isRequired: true,
    pattern: "^[a-zA-Z-.`' ]*$",
    type: "text",
    title: t("WS_NAME_ERROR_MESSAGE")
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_OWN_DETAIL_GENDER_LABEL") + "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioButtons, {
    t: t,
    options: menu,
    optionsKey: "code",
    name: "gender",
    value: gender,
    selectedOption: gender,
    onSelect: setGenderName,
    isDependent: true,
    labelKey: "COMMON_GENDER"
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_OWN_MOBILE_NO") + "*"), /*#__PURE__*/React__default.createElement(digitUiReactComponents.MobileNumber, {
    value: mobileNumber,
    name: "mobileNumber",
    onChange: function onChange(value) {
      return setMobileNo({
        target: {
          value: value
        }
      });
    },
    style: {
      background: "#FAFAFA"
    }
    ,
    required: true,
    pattern: "[6-9]{1}[0-9]{9}",
    type: "tel",
    title: t("CORE_COMMON_APPLICANT_MOBILE_NUMBER_INVALID")
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_OWN_DETAIL_GUARDIAN_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, _extends({
    t: t,
    type: "text",
    isMandatory: false,
    optionKey: "i18nKey",
    name: "guardian",
    value: guardian,
    style: {
      background: "#FAFAFA"
    },
    onChange: selectguardian
  }, validation = {
    isRequired: true,
    pattern: "^[a-zA-Z-.`' ]*$",
    type: "text",
    title: t("WS_NAME_ERROR_MESSAGE")
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.RadioButtons, {
    t: t,
    optionsKey: "i18nKey",
    name: "relationship",
    options: GuardianOptions,
    value: relationship,
    selectedOption: relationship,
    onSelect: setRelationshipName,
    isDependent: true,
    labelKey: "COMMON_MASTERS_OWNERTYPE"
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_COMMON_TABLE_COL_ADDRESS")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "text",
    style: {
      background: "#FAFAFA"
    },
    isMandatory: false,
    optionKey: "i18nKey",
    name: "address",
    value: address,
    onChange: selectaddress
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, t("WS_OWN_SPECIAL_CAT_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    className: "form-field",
    selected: ownerType
    ,
    option: Menu,
    select: setTypeOfOwner,
    optionKey: "i18nKey",
    t: t
  }), ownerType && Object.entries(ownerType).length > 0 && (ownerType === null || ownerType === void 0 ? void 0 : ownerType.code) !== "NONE" && /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_DOCUMENT_ID_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.TextInput, {
    t: t,
    type: "text",
    style: {
      background: "#FAFAFA"
    },
    isMandatory: false,
    optionKey: "i18nKey",
    name: "documentId",
    value: documentId,
    onChange: selectdocumentId
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardLabel, null, "" + t("WS_DOCUMENT_TYPE_LABEL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Dropdown, {
    t: t,
    isMandatory: false,
    option: dropdownData,
    selected: dropdownValue,
    optionKey: "i18nKey",
    select: setTypeOfDropdownValue
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.UploadFile, {
    id: "ptm-doc"
    ,
    accept: ".jpg,.png,.pdf",
    onUpload: selectfile,
    onDelete: function onDelete() {
      setUploadedFile(null);
    },
    message: uploadedFile ? "1 " + t("WS_ACTION_FILEUPLOADED") : t("WS_ACTION_NO_FILEUPLOADED"),
    error: error
  }), error ? /*#__PURE__*/React__default.createElement("div", {
    style: {
      height: "20px",
      width: "100%",
      fontSize: "20px",
      color: "red",
      marginTop: "5px"
    }
  }, error) : ""))) : /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null));
};

var CheckPage = function CheckPage(_ref) {
  var _ConnectionHolderDeta, _ConnectionHolderDeta2, _ConnectionHolderDeta3, _waterConectionDetail;
  var onSubmit = _ref.onSubmit,
    value = _ref.value;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var history = reactRouterDom.useHistory();
  var match = reactRouterDom.useRouteMatch();
  var ConnectionHolderDetails = value.ConnectionHolderDetails,
    serviceName = value.serviceName,
    waterConectionDetails = value.waterConectionDetails,
    sewerageConnectionDetails = value.sewerageConnectionDetails,
    documents = value.documents;
  var routeLink = "/digit-ui/citizen/ws/create-application";
  function routeTo(jumpTo) {
    location.href = jumpTo;
  }
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(Timeline, {
    currentStep: 4
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Header, null, t("WS_COMMON_SUMMARY")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Card, {
    style: {
      paddingRight: "16px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("WS_BASIC_DETAILS_HEADER")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.StatusTable, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_PROPERTY_ID_LABEL"),
    text: "PT-767-23-0654553"
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_OWNERS_NAME_LABEL"),
    text: t("NS Prasad")
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_COMMON_TABLE_COL_ADDRESS"),
    text: "NA"
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_CONNECTION_DETAILS_STATUS_LABEL"),
    text: "NA"
  }))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Card, {
    style: {
      paddingRight: "16px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("WS_COMMON_CONNECTION_HOLDER_DETAILS_HEADER")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LinkButton, {
    label: /*#__PURE__*/React__default.createElement(digitUiReactComponents.EditIcon, {
      style: {
        marginTop: "-10px",
        float: "right",
        position: "relative",
        bottom: "32px"
      }
    }),
    style: {
      width: "100px",
      display: "inline"
    },
    onClick: function onClick() {
      return routeTo(routeLink + "/connection-holder");
    }
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.StatusTable, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_OWN_MOBILE_NO"),
    text: ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : ConnectionHolderDetails.mobileNumber
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_OWN_DETAIL_NAME"),
    text: ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : ConnectionHolderDetails.name
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_OWN_DETAIL_GENDER_LABEL"),
    text: t(ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : (_ConnectionHolderDeta = ConnectionHolderDetails.gender) === null || _ConnectionHolderDeta === void 0 ? void 0 : _ConnectionHolderDeta.i18nKey)
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_FATHERS_HUSBAND_NAME"),
    text: ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : ConnectionHolderDetails.guardian
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_CONN_HOLDER_OWN_DETAIL_RELATION_LABEL"),
    text: t(ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : (_ConnectionHolderDeta2 = ConnectionHolderDetails.relationship) === null || _ConnectionHolderDeta2 === void 0 ? void 0 : _ConnectionHolderDeta2.i18nKey)
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_OWN_DETAIL_CROSADD"),
    text: ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : ConnectionHolderDetails.address
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_OWN_DETAIL_SPECIAL_APPLICANT_LABEL"),
    text: t(ConnectionHolderDetails === null || ConnectionHolderDetails === void 0 ? void 0 : (_ConnectionHolderDeta3 = ConnectionHolderDetails.specialCategoryType) === null || _ConnectionHolderDeta3 === void 0 ? void 0 : _ConnectionHolderDeta3.i18nKey)
  }))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Card, {
    style: {
      paddingRight: "16px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("WS_COMMON_CONNECTION_DETAIL")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LinkButton, {
    label: /*#__PURE__*/React__default.createElement(digitUiReactComponents.EditIcon, {
      style: {
        marginTop: "-10px",
        float: "right",
        position: "relative",
        bottom: "32px"
      }
    }),
    style: {
      width: "100px",
      display: "inline"
    },
    onClick: function onClick() {
      return routeTo(routeLink + "/service-name");
    }
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.StatusTable, null, /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_SERVICE_NAME_LABEL"),
    text: t(serviceName === null || serviceName === void 0 ? void 0 : serviceName.i18nKey)
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_SERV_DETAIL_CONN_TYPE"),
    text: "Metered"
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_NO_OF_TAPS"),
    text: waterConectionDetails === null || waterConectionDetails === void 0 ? void 0 : waterConectionDetails.proposedTaps
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_SERV_DETAIL_PIPE_SIZE"),
    text: t(waterConectionDetails === null || waterConectionDetails === void 0 ? void 0 : (_waterConectionDetail = waterConectionDetails.proposedPipeSize) === null || _waterConectionDetail === void 0 ? void 0 : _waterConectionDetail.i18nKey)
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_NO_OF_WATER_CLOSETS_LABEL"),
    text: sewerageConnectionDetails === null || sewerageConnectionDetails === void 0 ? void 0 : sewerageConnectionDetails.proposedWaterClosets
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_SERV_DETAIL_NO_OF_TOILETS"),
    text: sewerageConnectionDetails === null || sewerageConnectionDetails === void 0 ? void 0 : sewerageConnectionDetails.proposedToilets
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_SERV_DETAIL_WATER_SOURCE"),
    text: "NA"
  }), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Row, {
    className: "border-none",
    label: t("WS_SERV_DETAIL_WATER_SUB_SOURCE"),
    text: "NA"
  }))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.Card, {
    style: {
      paddingRight: "16px"
    }
  }, /*#__PURE__*/React__default.createElement(digitUiReactComponents.CardHeader, null, t("WS_COMMON_DOCUMENT_DETAILS")), /*#__PURE__*/React__default.createElement(digitUiReactComponents.LinkButton, {
    label: /*#__PURE__*/React__default.createElement(digitUiReactComponents.EditIcon, {
      style: {
        marginTop: "-10px",
        float: "right",
        position: "relative",
        bottom: "32px"
      }
    }),
    style: {
      width: "100px",
      display: "inline"
    },
    onClick: function onClick() {
      return routeTo(routeLink + "/document-details");
    }
  }), documents && documents.map(function (doc, index) {
    var _doc$documentType;
    return /*#__PURE__*/React__default.createElement("div", {
      key: "doc-" + index
    }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement(CardSectionHeader, null, t(doc === null || doc === void 0 ? void 0 : (_doc$documentType = doc.documentType) === null || _doc$documentType === void 0 ? void 0 : _doc$documentType.split('.').slice(0, 2).join('_'))), /*#__PURE__*/React__default.createElement(digitUiReactComponents.StatusTable, null, (documents === null || documents === void 0 ? void 0 : documents.length) != index + 1 ? /*#__PURE__*/React__default.createElement("hr", {
      style: {
        color: "#cccccc",
        backgroundColor: "#cccccc",
        height: "2px",
        marginTop: "20px",
        marginBottom: "20px"
      }
    }) : null)));
  })), /*#__PURE__*/React__default.createElement(digitUiReactComponents.SubmitBar, {
    label: t("CS_COMMON_SUBMIT"),
    onSubmit: onSubmit
  }));
};

var WSModule = function WSModule(_ref) {
  var stateCode = _ref.stateCode,
    userType = _ref.userType,
    tenants = _ref.tenants;
  var moduleCode = "ws";
  var _useRouteMatch = reactRouterDom.useRouteMatch(),
    path = _useRouteMatch.path;
  var language = Digit.StoreData.getCurrentLanguage();
  var _Digit$Services$useSt = Digit.Services.useStore({
      stateCode: stateCode,
      moduleCode: moduleCode,
      language: language
    }),
    isLoading = _Digit$Services$useSt.isLoading;
  Digit.SessionStorage.set("WS_TENANTS", tenants);
  if (isLoading) {
    return /*#__PURE__*/React__default.createElement(digitUiReactComponents.Loader, null);
  }
  if (userType === "citizen") {
    return /*#__PURE__*/React__default.createElement(App, {
      path: path,
      stateCode: stateCode
    });
  }

};

var WSLinks = function WSLinks(_ref2) {
  var matchPath = _ref2.matchPath;
  var _useTranslation = reactI18next.useTranslation(),
    t = _useTranslation.t;
  var links = [{
    link: "",
    i18nKey: t("ACTION_TEST_WATER_AND_SEWERAGE_BILL")
  }, {
    link: "",
    i18nKey: t("ACTION_TEST_MY_CONNECTIONS")
  }, {
    link: matchPath + "/create-application",
    i18nKey: t("ACTION_TEST_APPLY_NEW_CONNECTION")
  }, {
    link: matchPath + "/search",
    i18nKey: t("ACTION_TEXT_WS_SEARCH_AND_PAY")
  }];
  return /*#__PURE__*/React__default.createElement(digitUiReactComponents.CitizenHomeCard, {
    header: t("ACTION_TEST_WATER_AND_SEWERAGE"),
    links: links,
    Icon: function Icon() {
      return /*#__PURE__*/React__default.createElement(digitUiReactComponents.WSICon, null);
    }
  });
};
var componentsToRegister = {
  WSModule: WSModule,
  WSLinks: WSLinks,
  WSDocsRequired: WSDocsRequired,
  WSDocumentDetails: WSDocumentDetails,
  WSServiceName: WSServiceName,
  WSWaterConnectionDetails: WSWaterConnectionDetails,
  WSSewerageConnectionDetails: WSSewerageConnectionDetails,
  WSPlumberPreference: WSPlumberPreference,
  ConnectionHolder: ConnectionHolder,
  WSCheckPage: CheckPage
};
var initWSComponents = function initWSComponents() {
  Object.entries(componentsToRegister).forEach(function (_ref3) {
    var key = _ref3[0],
      value = _ref3[1];
    Digit.ComponentRegistryService.setComponent(key, value);
  });
};

exports.initWSComponents = initWSComponents;
